# Copy Number Signature Identification {#cnsig}

## Introduction 

Unlike several mutation types presented in current COSMIC database for generating mutational signatures, it is hard to represent copy number features and generate the matrix for NMF decomposition.

@macintyre2018copy created a new method to generate the matrix for extracting signature by non-negative matrix factorization (NMF) algorithm. The steps are:

- derive 6 copy number features from absolute copy number profile
- apply mixture modelling to breakdown each feature distribution into mixtures of Gaussian or mixtures of Poisson distributions
- generate a sample-by-component matrix representing the sum of posterior probabilities of each copy-number event being assigned to each component.

Based on previous work, our group devised a new method which discards the statistical modeling and create a fixed number of predefined components from 8 copy number features to generate the matrix as the input of NMF [@WangSX2020],
it is easier to reproduce the result, apply to different cancer types and compare results.
To test if the method would works, we applied it to prostate cancer and successfully identified 5 copy number signatures.

Currently, there are few studies focus on copy number signatures and no reference signature database for matching and explaining the etiologies. If you study them, you should do extra work to explore and validate them. Furthermore, the input absolute copy number data may be generated by different methods and platforms, it is normal that the contribution of some copy number feature components varies a little and result in relatively lower signature similarity when comparing different cohorts or different copy number profile generation methods.

## Read Data 

The input requires absolute copy number profile with following information:

- Segment chromosome.
- Segment start.
- Segment end.
- **Absolute copy number value for this segment**: must be integer (i.e. total copy number value in results from Sequenza or FACETS).
- Sample ID.

The input data can be result from any softwares which provides information above.

- For chip data/segmentation data (copy ratio is available), [ABSOLUTE](https://software.broadinstitute.org/cancer/cga/absolute) is recommended.
- For raw sequencing data, [Sequenza](https://cran.r-project.org/web/packages/sequenza/index.html) and [FACETS](https://github.com/mskcc/facets) are recommended.

The import work is done by `read_copynumber()`, this function supports `data.frame` or file, and even result directory from [ABSOLUTE](https://software.broadinstitute.org/cancer/cga/absolute).

Option `sigminer.sex` is used to control the processing of sex. If you don't care the sex chromosomes (i.e. X and Y),
you can ignore this setting after removing the X/Y segments, otherwise the `summary` in the result `cn` and tally process may be biased.

```{r}
## Default is "female"
## You can ignore the setting if all samples are females
## But we recommend you set it
options(sigminer.sex = "male")
## For cohort contains both males and females,
## set a data.frame with two columns, i.e.
## options(sigminer.sex = sex_df),
## which
## sex_df = data.frame(sample = c("sample1", "sample2",
##                     sex = "female", "male"))

# Load toy dataset of absolute copynumber profile
load(system.file("extdata", "toy_segTab.RData",
  package = "sigminer", mustWork = TRUE
))
cn <- read_copynumber(segTabs,
  seg_cols = c("chromosome", "start", "end", "segVal"),
  genome_build = "hg19", complement = FALSE, verbose = TRUE
)
cn
```

> Currently, you can refer to `extract_facets_cnv()` and `extract_seqz_cnv()` in <https://github.com/ShixiangWang/prad_signature/blob/master/analysis/src/99-functions.R> to
> see how to get tidy data from a result directory of FACETS or Sequenza.

## Tally Components 

Currently, there are two methods for generating sample-by-component matrix, i.e. "W" or "M".

Option `sigminer.copynumber.max` is used to control the processing of max copy number values. Run `?sig_tally` to see more.

```{r, eval=FALSE}
## Even you set max_copynumber = 20 in read_copynumber(),
## the segmental copy number may be greater than 20
## because for male samples, the X/Y segmental copy number
## values will be doubled in tally process.
## This setting will make copy number values of all segments
## not greater than 20.
options(sigminer.copynumber.max = 20)

# Load copy number object
load(system.file("extdata", "toy_copynumber.RData",
  package = "sigminer", mustWork = TRUE
))

# Use method designed by Wang, Shixiang et al.
cn_tally_W <- sig_tally(cn, method = "W")
# Use method designed by Macintyre et al.
cn_tally_M <- sig_tally(cn, method = "M")
```

> You can set `options(sigminer.sex = "male", sigminer.copynumber.max = 20)` at the top of your code
> to avoid setting them in two places.
>
> **Of note**, the `sigminer.copynumber.max` option only has effect on `sig_tally()` with method "W" or "M",
> the `sigminer.sex` option has effects on `read_copynumber()` and `sig_tally()` with method "W" or "M".

```{r, include=FALSE}
load(system.file("extdata", "toy_copynumber_tally_W.RData",
  package = "sigminer", mustWork = TRUE
))

load(system.file("extdata", "toy_copynumber_tally_M.RData",
  package = "sigminer", mustWork = TRUE
))
```

This step return a `list` containing information about copy number features, components and matrix for NMF etc.

## Extract Signatures 


When you get the matrix, you can just do the signature extractin as SBS signatures (see chapter Chapter \@ref(sbssig)). So here we won't talk much.

```{r}
cn_tally_W$nmf_matrix[1:5, 1:5]
```


```{r, eval=FALSE}
# library(NMF)
sig_w = sig_extract(cn_tally_W$nmf_matrix, n_sig = 2, pConstant = 1e-13)
```


```{r, include=FALSE}
load("data/sig_w.RData")
load("data/sig_m.RData")
```




