<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Copy Number Signature Identification | sigminer: Genomic Alteration Signature Analysis in R</title>
  <meta name="description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Copy Number Signature Identification | sigminer: Genomic Alteration Signature Analysis in R" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  <meta name="github-repo" content="ShixiangWang/sigminer-doc" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Copy Number Signature Identification | sigminer: Genomic Alteration Signature Analysis in R" />
  
  <meta name="twitter:description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  

<meta name="author" content="Shixiang Wang, ShanghaiTech University" />
<meta name="author" content="Xue-Song Liu, ShanghaiTech University" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="sbssig.html"/>
<link rel="next" href="part-object-visualization.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">sigminer: Signature Miner</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i>Installation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#issues-or-suggestions"><i class="fa fa-check"></i>Issues or Suggestions</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#preparation"><i class="fa fa-check"></i>Preparation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#citation-and-license"><i class="fa fa-check"></i>Citation and LICENSE</a></li>
</ul></li>
<li class="part"><span><b>I Signature Identification</b></span></li>
<li class="chapter" data-level="1" data-path="sbssig.html"><a href="sbssig.html"><i class="fa fa-check"></i><b>1</b> SBS Signature Identification</a>
<ul>
<li class="chapter" data-level="" data-path="sbssig.html"><a href="sbssig.html#read-data"><i class="fa fa-check"></i>Read Data</a></li>
<li class="chapter" data-level="" data-path="sbssig.html"><a href="sbssig.html#tally-components"><i class="fa fa-check"></i>Tally Components</a></li>
<li class="chapter" data-level="" data-path="sbssig.html"><a href="sbssig.html#extract-signatures"><i class="fa fa-check"></i>Extract Signatures</a>
<ul>
<li class="chapter" data-level="" data-path="sbssig.html"><a href="sbssig.html#method-1-estimate-signature-number"><i class="fa fa-check"></i>Method 1: Estimate Signature Number</a></li>
<li class="chapter" data-level="" data-path="sbssig.html"><a href="sbssig.html#method-1-extract-signatures"><i class="fa fa-check"></i>Method 1: Extract Signatures</a></li>
<li class="chapter" data-level="" data-path="sbssig.html"><a href="sbssig.html#method-2-auto-extract"><i class="fa fa-check"></i>Method 2: Auto-extract</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="sbssig.html"><a href="sbssig.html#match-signatures"><i class="fa fa-check"></i>Match Signatures</a></li>
<li><a href="sbssig.html#operate-signature">Operate <code>Signature</code></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cnsig.html"><a href="cnsig.html"><i class="fa fa-check"></i><b>2</b> Copy Number Signature Identification</a>
<ul>
<li class="chapter" data-level="" data-path="cnsig.html"><a href="cnsig.html#introduction-1"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="cnsig.html"><a href="cnsig.html#read-data-1"><i class="fa fa-check"></i>Read Data</a></li>
<li class="chapter" data-level="" data-path="cnsig.html"><a href="cnsig.html#tally-components-1"><i class="fa fa-check"></i>Tally Components</a></li>
<li class="chapter" data-level="" data-path="cnsig.html"><a href="cnsig.html#extract-signatures-1"><i class="fa fa-check"></i>Extract Signatures</a></li>
</ul></li>
<li class="part"><span><b>II Object Visualization</b></span></li>
<li class="chapter" data-level="3" data-path="cnobject.html"><a href="cnobject.html"><i class="fa fa-check"></i><b>3</b> <code>CopyNumber</code> Object</a>
<ul>
<li class="chapter" data-level="" data-path="cnobject.html"><a href="cnobject.html#profile"><i class="fa fa-check"></i>Profile</a></li>
<li class="chapter" data-level="" data-path="cnobject.html"><a href="cnobject.html#distribution"><i class="fa fa-check"></i>Distribution</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sigobject.html"><a href="sigobject.html"><i class="fa fa-check"></i><b>4</b> <code>Signature</code> Object</a>
<ul>
<li class="chapter" data-level="" data-path="sigobject.html"><a href="sigobject.html#signature-profile"><i class="fa fa-check"></i>Signature Profile</a>
<ul>
<li class="chapter" data-level="" data-path="sigobject.html"><a href="sigobject.html#sbs-signature-profile"><i class="fa fa-check"></i>SBS Signature profile</a></li>
<li class="chapter" data-level="" data-path="sigobject.html"><a href="sigobject.html#copy-number-signature-porfile"><i class="fa fa-check"></i>Copy Number Signature Porfile</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="sigobject.html"><a href="sigobject.html#exposure-profile"><i class="fa fa-check"></i>Exposure Profile</a></li>
<li class="chapter" data-level="" data-path="sigobject.html"><a href="sigobject.html#consensus-map"><i class="fa fa-check"></i>Consensus Map</a></li>
</ul></li>
<li class="part"><span><b>III Signature Analysis and Visualization</b></span></li>
<li class="chapter" data-level="5" data-path="association-analysis.html"><a href="association-analysis.html"><i class="fa fa-check"></i><b>5</b> Association Analysis and Visualization</a></li>
<li class="chapter" data-level="6" data-path="group-analysis.html"><a href="group-analysis.html"><i class="fa fa-check"></i><b>6</b> Group Analysis and Visualization</a>
<ul>
<li class="chapter" data-level="" data-path="group-analysis.html"><a href="group-analysis.html#group-generation"><i class="fa fa-check"></i>Group Generation</a></li>
<li class="chapter" data-level="" data-path="group-analysis.html"><a href="group-analysis.html#group-comparison-analysis"><i class="fa fa-check"></i>Group Comparison Analysis</a></li>
<li class="chapter" data-level="" data-path="group-analysis.html"><a href="group-analysis.html#group-visualization"><i class="fa fa-check"></i>Group Visualization</a></li>
</ul></li>
<li class="part"><span><b>IV Others</b></span></li>
<li class="chapter" data-level="7" data-path="datasets.html"><a href="datasets.html"><i class="fa fa-check"></i><b>7</b> Datasets</a>
<ul>
<li class="chapter" data-level="" data-path="datasets.html"><a href="datasets.html#reference-annotation"><i class="fa fa-check"></i>Reference Annotation</a></li>
<li class="chapter" data-level="" data-path="datasets.html"><a href="datasets.html#copy-number-components-setting"><i class="fa fa-check"></i>Copy Number components setting</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">sigminer: Genomic Alteration Signature Analysis in R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="cnsig" class="section level1">
<h1><span class="header-section-number">Chapter 2</span> Copy Number Signature Identification</h1>
<div id="introduction-1" class="section level2">
<h2>Introduction</h2>
<p>Unlike several mutation types presented in current COSMIC database for generating mutational signatures, it is hard to represent copy number features and generate the matrix for NMF decomposition.</p>
<p><span class="citation">Macintyre et al. (<a href="#ref-macintyre2018copy" role="doc-biblioref">2018</a>)</span> created a new method to generate the matrix for extracting signature by non-negative matrix factorization (NMF) algorithm. The steps are:</p>
<ul>
<li>derive 6 copy number features from absolute copy number profile</li>
<li>apply mixture modelling to breakdown each feature distribution into mixtures of Gaussian or mixtures of Poisson distributions</li>
<li>generate a sample-by-component matrix representing the sum of posterior probabilities of each copy-number event being assigned to each component.</li>
</ul>
<p>Based on previous work, our group devised a new method which discards the statistical modeling and create a fixed number of predefined components from 8 copy number features to generate the matrix as the input of NMF,
it is easier to reproduce the result, apply to different cancer types and compare results.
To test if the method would works, we applied it to prostate cancer and successfully identified 5 copy number signatures.</p>
<p>Currently, there are few studies focus on copy number signatures and no reference signature database for matching and explaining the etiologies. If you study them, you should do extra work to explore and validate them. Furthermore, the input absolute copy number data may be generated by different methods and platforms, it is normal that the contribution of some copy number feature components varies a little and result in relatively lower signature similarity when comparing different cohorts or different copy number profile generation methods.</p>
</div>
<div id="read-data-1" class="section level2">
<h2>Read Data</h2>
<p>The input requires absolute copy number profile with following information:</p>
<ul>
<li>Segment chromosome.</li>
<li>Segment start.</li>
<li>Segment end.</li>
<li><strong>Segment value</strong>: must be integer.</li>
<li>Sample ID.</li>
</ul>
<p>The input data can be result from any softwares which provides information above.</p>
<ul>
<li>For chip data/segmentation data (copy ratio is available), <a href="https://software.broadinstitute.org/cancer/cga/absolute">ABSOLUTE</a> is recommended.</li>
<li>For raw sequencing data, <a href="https://cran.r-project.org/web/packages/sequenza/index.html">Sequenza</a> and <a href="https://github.com/mskcc/facets">FACETS</a> are recommended.</li>
</ul>
<p>The import work is done by <code>read_copynumber</code>, this function supports <code>data.frame</code> or file, and even result directory from <a href="https://software.broadinstitute.org/cancer/cga/absolute">ABSOLUTE</a>.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="cnsig.html#cb22-1"></a><span class="co"># Load toy dataset of absolute copynumber profile</span></span>
<span id="cb22-2"><a href="cnsig.html#cb22-2"></a><span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;toy_segTab.RData&quot;</span>,</span>
<span id="cb22-3"><a href="cnsig.html#cb22-3"></a>  <span class="dt">package =</span> <span class="st">&quot;sigminer&quot;</span>, <span class="dt">mustWork =</span> <span class="ot">TRUE</span></span>
<span id="cb22-4"><a href="cnsig.html#cb22-4"></a>))</span>
<span id="cb22-5"><a href="cnsig.html#cb22-5"></a>cn &lt;-<span class="st"> </span><span class="kw">read_copynumber</span>(segTabs,</span>
<span id="cb22-6"><a href="cnsig.html#cb22-6"></a>  <span class="dt">seg_cols =</span> <span class="kw">c</span>(<span class="st">&quot;chromosome&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>, <span class="st">&quot;segVal&quot;</span>),</span>
<span id="cb22-7"><a href="cnsig.html#cb22-7"></a>  <span class="dt">genome_build =</span> <span class="st">&quot;hg19&quot;</span>, <span class="dt">complement =</span> <span class="ot">FALSE</span>, <span class="dt">verbose =</span> <span class="ot">TRUE</span></span>
<span id="cb22-8"><a href="cnsig.html#cb22-8"></a>)</span>
<span id="cb22-9"><a href="cnsig.html#cb22-9"></a><span class="co">#&gt; Treating input as a data frame...</span></span>
<span id="cb22-10"><a href="cnsig.html#cb22-10"></a><span class="co">#&gt; Checking chromosome names...</span></span>
<span id="cb22-11"><a href="cnsig.html#cb22-11"></a><span class="co">#&gt; Segments info:</span></span>
<span id="cb22-12"><a href="cnsig.html#cb22-12"></a><span class="co">#&gt;     Keep - 467</span></span>
<span id="cb22-13"><a href="cnsig.html#cb22-13"></a><span class="co">#&gt;   Filter - 0</span></span>
<span id="cb22-14"><a href="cnsig.html#cb22-14"></a><span class="co">#&gt; Joining adjacent segments with same copy number value...</span></span>
<span id="cb22-15"><a href="cnsig.html#cb22-15"></a><span class="co">#&gt; Anotating...</span></span>
<span id="cb22-16"><a href="cnsig.html#cb22-16"></a><span class="co">#&gt; </span></span>
<span id="cb22-17"><a href="cnsig.html#cb22-17"></a><span class="co">#&gt; Summarizing per sample...</span></span>
<span id="cb22-18"><a href="cnsig.html#cb22-18"></a><span class="co">#&gt; Done!</span></span>
<span id="cb22-19"><a href="cnsig.html#cb22-19"></a>cn</span>
<span id="cb22-20"><a href="cnsig.html#cb22-20"></a><span class="co">#&gt; An object of class CopyNumber </span></span>
<span id="cb22-21"><a href="cnsig.html#cb22-21"></a><span class="co">#&gt; =============================</span></span>
<span id="cb22-22"><a href="cnsig.html#cb22-22"></a><span class="co">#&gt;                           sample n_of_seg n_of_cnv n_of_amp n_of_del n_of_vchr cna_burden</span></span>
<span id="cb22-23"><a href="cnsig.html#cb22-23"></a><span class="co">#&gt;  1: TCGA-DF-A2KN-01A-11D-A17U-01       33        6        5        1         4      0.000</span></span>
<span id="cb22-24"><a href="cnsig.html#cb22-24"></a><span class="co">#&gt;  2: TCGA-19-2621-01B-01D-0911-01       33        8        5        3         5      0.099</span></span>
<span id="cb22-25"><a href="cnsig.html#cb22-25"></a><span class="co">#&gt;  3: TCGA-B6-A0X5-01A-21D-A107-01       28        8        4        4         2      0.087</span></span>
<span id="cb22-26"><a href="cnsig.html#cb22-26"></a><span class="co">#&gt;  4: TCGA-A8-A07S-01A-11D-A036-01       38       11        2        9         4      0.112</span></span>
<span id="cb22-27"><a href="cnsig.html#cb22-27"></a><span class="co">#&gt;  5: TCGA-26-6174-01A-21D-1842-01       43       13        8        5         8      0.119</span></span>
<span id="cb22-28"><a href="cnsig.html#cb22-28"></a><span class="co">#&gt;  6: TCGA-CV-7432-01A-11D-2128-01       40       16        7        9         9      0.198</span></span>
<span id="cb22-29"><a href="cnsig.html#cb22-29"></a><span class="co">#&gt;  7: TCGA-06-0644-01A-02D-0310-01       46       19        5       14         8      0.165</span></span>
<span id="cb22-30"><a href="cnsig.html#cb22-30"></a><span class="co">#&gt;  8: TCGA-A5-A0G2-01A-11D-A042-01       39       21        5       16        10      0.393</span></span>
<span id="cb22-31"><a href="cnsig.html#cb22-31"></a><span class="co">#&gt;  9: TCGA-99-7458-01A-11D-2035-01       48       26       10       16        13      0.318</span></span>
<span id="cb22-32"><a href="cnsig.html#cb22-32"></a><span class="co">#&gt; 10: TCGA-05-4417-01A-22D-1854-01       52       37       33        4        17      0.654</span></span></code></pre></div>
</div>
<div id="tally-components-1" class="section level2">
<h2>Tally Components</h2>
<p>Currently, there are two methods for generating sample-by-component matrix, i.e. “W” or “M”.</p>
<p>Options <code>sigminer.sex</code> and <code>sigminer.copynumber.max</code> are used to control the processing of sex and max values. Run <code>?sig_tally</code> to see more.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="cnsig.html#cb23-1"></a><span class="kw">options</span>(<span class="dt">sigminer.sex =</span> <span class="st">&quot;male&quot;</span>, <span class="dt">sigminer.copynumber.max =</span> <span class="dv">10</span>)</span>
<span id="cb23-2"><a href="cnsig.html#cb23-2"></a></span>
<span id="cb23-3"><a href="cnsig.html#cb23-3"></a><span class="co"># Load copy number object</span></span>
<span id="cb23-4"><a href="cnsig.html#cb23-4"></a><span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;toy_copynumber.RData&quot;</span>,</span>
<span id="cb23-5"><a href="cnsig.html#cb23-5"></a>  <span class="dt">package =</span> <span class="st">&quot;sigminer&quot;</span>, <span class="dt">mustWork =</span> <span class="ot">TRUE</span></span>
<span id="cb23-6"><a href="cnsig.html#cb23-6"></a>))</span>
<span id="cb23-7"><a href="cnsig.html#cb23-7"></a></span>
<span id="cb23-8"><a href="cnsig.html#cb23-8"></a><span class="co"># Use method designed by Wang, Shixiang et al.</span></span>
<span id="cb23-9"><a href="cnsig.html#cb23-9"></a>cn_tally_W &lt;-<span class="st"> </span><span class="kw">sig_tally</span>(cn, <span class="dt">method =</span> <span class="st">&quot;W&quot;</span>)</span>
<span id="cb23-10"><a href="cnsig.html#cb23-10"></a><span class="co"># Use method designed by Macintyre et al.</span></span>
<span id="cb23-11"><a href="cnsig.html#cb23-11"></a>cn_tally_M &lt;-<span class="st"> </span><span class="kw">sig_tally</span>(cn, <span class="dt">method =</span> <span class="st">&quot;M&quot;</span>)</span></code></pre></div>
<p>This step return a <code>list</code> containing information about copy number features, components and matrix for NMF etc.</p>
</div>
<div id="extract-signatures-1" class="section level2">
<h2>Extract Signatures</h2>
<p>When you get the matrix, you can just do the signature extractin as SBS signatures. So here we won’t talk much.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="cnsig.html#cb24-1"></a>cn_tally_W<span class="op">$</span>nmf_matrix[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</span>
<span id="cb24-2"><a href="cnsig.html#cb24-2"></a><span class="co">#&gt;                              BP10MB[0] BP10MB[1] BP10MB[2] BP10MB[3] BP10MB[4]</span></span>
<span id="cb24-3"><a href="cnsig.html#cb24-3"></a><span class="co">#&gt; TCGA-05-4417-01A-22D-1854-01       275        20         5         0         0</span></span>
<span id="cb24-4"><a href="cnsig.html#cb24-4"></a><span class="co">#&gt; TCGA-06-0644-01A-02D-0310-01       289         5         4         0         1</span></span>
<span id="cb24-5"><a href="cnsig.html#cb24-5"></a><span class="co">#&gt; TCGA-19-2621-01B-01D-0911-01       294         2         3         1         0</span></span>
<span id="cb24-6"><a href="cnsig.html#cb24-6"></a><span class="co">#&gt; TCGA-26-6174-01A-21D-1842-01       288         4         7         1         0</span></span>
<span id="cb24-7"><a href="cnsig.html#cb24-7"></a><span class="co">#&gt; TCGA-99-7458-01A-11D-2035-01       284         9         5         1         1</span></span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="cnsig.html#cb25-1"></a><span class="co"># library(NMF)</span></span>
<span id="cb25-2"><a href="cnsig.html#cb25-2"></a>sig_w &lt;-<span class="st"> </span><span class="kw">sig_extract</span>(cn_tally_W<span class="op">$</span>nmf_matrix, <span class="dt">n_sig =</span> <span class="dv">2</span>, <span class="dt">pConstant =</span> <span class="fl">1e-13</span>)</span></code></pre></div>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-macintyre2018copy">
<p>Macintyre, Geoff, Teodora E Goranova, Dilrini De Silva, Darren Ennis, Anna M Piskorz, Matthew Eldridge, Daoud Sie, et al. 2018. “Copy Number Signatures and Mutational Processes in Ovarian Carcinoma.” <em>Nature Genetics</em> 50 (9): 1262–70.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="sbssig.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="part-object-visualization.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/ShixiangWang/sigminer-doc/edit/master/02-copy-signature-identification.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["sigminer-doc.pdf", "sigminer-doc.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
