<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Copy Number Signature Identification | Sigminer: A Scalable Toolkit to Extract, Analyze and Visualize Mutational Signatures</title>
  <meta name="description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Copy Number Signature Identification | Sigminer: A Scalable Toolkit to Extract, Analyze and Visualize Mutational Signatures" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  <meta name="github-repo" content="ShixiangWang/sigminer-doc" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Copy Number Signature Identification | Sigminer: A Scalable Toolkit to Extract, Analyze and Visualize Mutational Signatures" />
  
  <meta name="twitter:description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  

<meta name="author" content="Shixiang Wang, ShanghaiTech University" />
<meta name="author" content="Xue-Song Liu, ShanghaiTech University" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="sbssig.html"/>
<link rel="next" href="sigfit.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.15/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">sigminer: Signature Miner</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#biological-significance-of-mutational-signature"><i class="fa fa-check"></i><b>1.1</b> Biological Significance of Mutational Signature</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#sigminer"><i class="fa fa-check"></i><b>1.2</b> Sigminer</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.3</b> Installation</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#issues-or-suggestions"><i class="fa fa-check"></i><b>1.4</b> Issues or Suggestions</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#preparation"><i class="fa fa-check"></i><b>1.5</b> Preparation</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#overview-of-contents"><i class="fa fa-check"></i><b>1.6</b> Overview of Contents</a></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#citation-and-license"><i class="fa fa-check"></i><b>1.7</b> Citation and LICENSE</a></li>
</ul></li>
<li class="part"><span><b>I Common Workflow</b></span></li>
<li class="chapter" data-level="2" data-path="sbssig.html"><a href="sbssig.html"><i class="fa fa-check"></i><b>2</b> COSMIC Signature Identification</a><ul>
<li class="chapter" data-level="2.1" data-path="sbssig.html"><a href="sbssig.html#read-data"><i class="fa fa-check"></i><b>2.1</b> Read Data</a><ul>
<li class="chapter" data-level="2.1.1" data-path="sbssig.html"><a href="sbssig.html#vcf-as-input"><i class="fa fa-check"></i><b>2.1.1</b> VCF as input</a></li>
<li class="chapter" data-level="2.1.2" data-path="sbssig.html"><a href="sbssig.html#maf-as-input"><i class="fa fa-check"></i><b>2.1.2</b> MAF as input</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="sbssig.html"><a href="sbssig.html#tally-components"><i class="fa fa-check"></i><b>2.2</b> Tally Components</a><ul>
<li class="chapter" data-level="2.2.1" data-path="sbssig.html"><a href="sbssig.html#the-most-common-96-components"><i class="fa fa-check"></i><b>2.2.1</b> The most common 96 components</a></li>
<li class="chapter" data-level="2.2.2" data-path="sbssig.html"><a href="sbssig.html#other-situations"><i class="fa fa-check"></i><b>2.2.2</b> Other Situations</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="sbssig.html"><a href="sbssig.html#extract-signatures"><i class="fa fa-check"></i><b>2.3</b> Extract Signatures</a><ul>
<li class="chapter" data-level="2.3.1" data-path="sbssig.html"><a href="sbssig.html#method-1-estimate-signature-number"><i class="fa fa-check"></i><b>2.3.1</b> Method 1: Estimate Signature Number</a></li>
<li class="chapter" data-level="2.3.2" data-path="sbssig.html"><a href="sbssig.html#method-1-extract-signatures"><i class="fa fa-check"></i><b>2.3.2</b> Method 1: Extract Signatures</a></li>
<li class="chapter" data-level="2.3.3" data-path="sbssig.html"><a href="sbssig.html#method-2-auto-extraction"><i class="fa fa-check"></i><b>2.3.3</b> Method 2: Auto-extraction</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="sbssig.html"><a href="sbssig.html#match-signatures"><i class="fa fa-check"></i><b>2.4</b> Match Signatures</a></li>
<li class="chapter" data-level="2.5" data-path="sbssig.html"><a href="sbssig.html#operate-signature"><i class="fa fa-check"></i><b>2.5</b> Operate <code>Signature</code></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cnsig.html"><a href="cnsig.html"><i class="fa fa-check"></i><b>3</b> Copy Number Signature Identification</a><ul>
<li class="chapter" data-level="3.1" data-path="cnsig.html"><a href="cnsig.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="sbssig.html"><a href="sbssig.html#read-data"><i class="fa fa-check"></i><b>3.2</b> Read Data</a></li>
<li class="chapter" data-level="3.3" data-path="sbssig.html"><a href="sbssig.html#tally-components"><i class="fa fa-check"></i><b>3.3</b> Tally Components</a></li>
<li class="chapter" data-level="3.4" data-path="sbssig.html"><a href="sbssig.html#extract-signatures"><i class="fa fa-check"></i><b>3.4</b> Extract Signatures</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sigfit.html"><a href="sigfit.html"><i class="fa fa-check"></i><b>4</b> Signature Fit: Sample Signature Exposure Quantification and Analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="sigfit.html"><a href="sigfit.html#fit-signatures-from-reference-databases"><i class="fa fa-check"></i><b>4.1</b> Fit Signatures from reference databases</a></li>
<li class="chapter" data-level="4.2" data-path="sigfit.html"><a href="sigfit.html#fit-custom-signatures"><i class="fa fa-check"></i><b>4.2</b> Fit Custom Signatures</a></li>
<li class="chapter" data-level="4.3" data-path="sigfit.html"><a href="sigfit.html#performance-comparison"><i class="fa fa-check"></i><b>4.3</b> Performance Comparison</a></li>
<li class="chapter" data-level="4.4" data-path="sigfit.html"><a href="sigfit.html#estimate-exposure-stability-by-bootstrap"><i class="fa fa-check"></i><b>4.4</b> Estimate Exposure Stability by Bootstrap</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="subtyping.html"><a href="subtyping.html"><i class="fa fa-check"></i><b>5</b> Subtype Prediction</a></li>
<li class="chapter" data-level="6" data-path="sigflow.html"><a href="sigflow.html"><i class="fa fa-check"></i><b>6</b> Sigflow Pipeline</a><ul>
<li class="chapter" data-level="6.1" data-path="sigflow.html"><a href="sigflow.html#cancer-type-specific-signature-index-database"><i class="fa fa-check"></i><b>6.1</b> Cancer type specific signature index database</a></li>
</ul></li>
<li class="part"><span><b>II Target Visualization</b></span></li>
<li class="chapter" data-level="7" data-path="cnobject.html"><a href="cnobject.html"><i class="fa fa-check"></i><b>7</b> CopyNumber Object</a><ul>
<li class="chapter" data-level="7.1" data-path="cnobject.html"><a href="cnobject.html#profile"><i class="fa fa-check"></i><b>7.1</b> Profile</a></li>
<li class="chapter" data-level="7.2" data-path="cnobject.html"><a href="cnobject.html#distribution"><i class="fa fa-check"></i><b>7.2</b> Distribution</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="sigobject.html"><a href="sigobject.html"><i class="fa fa-check"></i><b>8</b> Signature Object</a><ul>
<li class="chapter" data-level="8.1" data-path="sigobject.html"><a href="sigobject.html#signature-profile"><i class="fa fa-check"></i><b>8.1</b> Signature Profile</a><ul>
<li class="chapter" data-level="8.1.1" data-path="sigobject.html"><a href="sigobject.html#sbs-signature-profile"><i class="fa fa-check"></i><b>8.1.1</b> SBS Signature profile</a></li>
<li class="chapter" data-level="8.1.2" data-path="sigobject.html"><a href="sigobject.html#copy-number-signature-porfile"><i class="fa fa-check"></i><b>8.1.2</b> Copy Number Signature Porfile</a></li>
<li class="chapter" data-level="8.1.3" data-path="sigobject.html"><a href="sigobject.html#cosmic-signature-profile"><i class="fa fa-check"></i><b>8.1.3</b> COSMIC Signature Profile</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="sigobject.html"><a href="sigobject.html#exposure-profile"><i class="fa fa-check"></i><b>8.2</b> Exposure Profile</a></li>
<li class="chapter" data-level="8.3" data-path="sigobject.html"><a href="sigobject.html#consensus-map"><i class="fa fa-check"></i><b>8.3</b> Consensus Map</a></li>
<li class="chapter" data-level="8.4" data-path="sigobject.html"><a href="sigobject.html#catalogue-profile"><i class="fa fa-check"></i><b>8.4</b> Catalogue Profile</a></li>
</ul></li>
<li class="part"><span><b>III Universal Analysis</b></span></li>
<li class="chapter" data-level="9" data-path="association-analysis.html"><a href="association-analysis.html"><i class="fa fa-check"></i><b>9</b> Association Analysis and Visualization</a></li>
<li class="chapter" data-level="10" data-path="group-analysis.html"><a href="group-analysis.html"><i class="fa fa-check"></i><b>10</b> Group Analysis and Visualization</a><ul>
<li class="chapter" data-level="10.1" data-path="group-analysis.html"><a href="group-analysis.html#group-generation"><i class="fa fa-check"></i><b>10.1</b> Group Generation</a></li>
<li class="chapter" data-level="10.2" data-path="group-analysis.html"><a href="group-analysis.html#group-comparison-analysis"><i class="fa fa-check"></i><b>10.2</b> Group Comparison Analysis</a></li>
<li class="chapter" data-level="10.3" data-path="group-analysis.html"><a href="group-analysis.html#group-visualization"><i class="fa fa-check"></i><b>10.3</b> Group Visualization</a></li>
</ul></li>
<li class="part"><span><b>IV Useful Utilities</b></span></li>
<li class="chapter" data-level="11" data-path="datasets.html"><a href="datasets.html"><i class="fa fa-check"></i><b>11</b> Datasets</a><ul>
<li class="chapter" data-level="11.1" data-path="datasets.html"><a href="datasets.html#reference-annotation"><i class="fa fa-check"></i><b>11.1</b> Reference Annotation</a></li>
<li class="chapter" data-level="11.2" data-path="datasets.html"><a href="datasets.html#copy-number-components-setting"><i class="fa fa-check"></i><b>11.2</b> Copy Number components setting</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="convert.html"><a href="convert.html"><i class="fa fa-check"></i><b>12</b> SBS Signature Conversion</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Sigminer: A Scalable Toolkit to Extract, Analyze and Visualize Mutational Signatures</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="cnsig" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Copy Number Signature Identification</h1>
<div id="introduction" class="section level2">
<h2><span class="header-section-number">3.1</span> Introduction</h2>
<p>Unlike several mutation types presented in current COSMIC database for generating mutational signatures, it is hard to represent copy number features and generate the matrix for NMF decomposition.</p>
<p><span class="citation">Macintyre et al. (<a href="#ref-macintyre2018copy" role="doc-biblioref">2018</a>)</span> created a new method to generate the matrix for extracting signature by NMF algorithm. The steps are:</p>
<ul>
<li>derive 6 copy number features from absolute copy number profile</li>
<li>apply mixture modeling to breakdown each feature distribution into mixtures of Gaussian or mixtures of Poisson distributions</li>
<li>generate a sample-by-component matrix representing the sum of posterior probabilities of each copy-number event being assigned to each component.</li>
</ul>
<p>Based on previous work, our group devised a new method which discards the statistical modeling and create a fixed number of predefined components from 8 copy number features to generate the matrix as the input of NMF <span class="citation">(Wang et al. <a href="#ref-WangSX2020" role="doc-biblioref">2020</a>)</span>,
it is easier to reproduce the result, apply to different cancer types and compare results.
To test if the method would works, we applied it to prostate cancer and successfully identified 5 copy number signatures.</p>
<p>Currently, there are few studies focus on copy number signatures and no reference signature database for matching and explaining the etiologies (The signatures presented in two papers above can be used as references). If you study them, you should do extra work to explore and validate them. Furthermore, the input absolute copy number data may be generated by different methods and platforms, it is normal that the contribution of some copy number feature components varies a little and result in relatively lower signature similarity when comparing different cohorts or different copy number profile generation methods.</p>
</div>
<div id="read-data" class="section level2">
<h2><span class="header-section-number">3.2</span> Read Data</h2>
<p>The input requires absolute copy number profile with following information:</p>
<ul>
<li>Segment chromosome.</li>
<li>Segment start.</li>
<li>Segment end.</li>
<li><strong>Absolute copy number value for this segment</strong>: must be integer.</li>
<li>Sample ID.</li>
</ul>
<p>The input data can be result from any software which provides information above.</p>
<p>Useful softwares are listed below:</p>
<ul>
<li><a href="https://software.broadinstitute.org/cancer/cga/absolute">ABSOLUTE</a>.</li>
<li><a href="https://cran.r-project.org/web/packages/sequenza/index.html">Sequenza</a></li>
<li><a href="https://github.com/mskcc/facets">FACETS</a></li>
<li><a href="http://penncnv.openbioinformatics.org/en/latest/">PennCNV</a> &amp; <a href="https://www.crick.ac.uk/research/labs/peter-van-loo/software">ASCAT</a></li>
<li><a href="https://github.com/etal/cnvkit">CNVkit</a></li>
</ul>
<p>The import work is done by <code>read_copynumber()</code>, which supports <code>data.frame</code> or file, and even result directory from <strong>ABSOLUTE</strong>.</p>
<p>Option <code>sigminer.sex</code> is used to control the processing of sex. If you don’t care the sex chromosomes (i.e. X and Y),
you can ignore this setting after removing the X/Y segments, otherwise the <code>summary</code> in the result <code>cn</code> and tally process may be biased.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="cnsig.html#cb34-1"></a><span class="co">## Default is &quot;female&quot;</span></span>
<span id="cb34-2"><a href="cnsig.html#cb34-2"></a><span class="co">## You can ignore the setting if all samples are females</span></span>
<span id="cb34-3"><a href="cnsig.html#cb34-3"></a><span class="co">## But we recommend you set it</span></span>
<span id="cb34-4"><a href="cnsig.html#cb34-4"></a><span class="kw">options</span>(<span class="dt">sigminer.sex =</span> <span class="st">&quot;male&quot;</span>)</span>
<span id="cb34-5"><a href="cnsig.html#cb34-5"></a><span class="co">## For cohort contains both males and females,</span></span>
<span id="cb34-6"><a href="cnsig.html#cb34-6"></a><span class="co">## set a data.frame with two columns, i.e.</span></span>
<span id="cb34-7"><a href="cnsig.html#cb34-7"></a><span class="co">## options(sigminer.sex = sex_df),</span></span>
<span id="cb34-8"><a href="cnsig.html#cb34-8"></a><span class="co">## which</span></span>
<span id="cb34-9"><a href="cnsig.html#cb34-9"></a><span class="co">## sex_df = data.frame(sample = c(&quot;sample1&quot;, &quot;sample2&quot;,</span></span>
<span id="cb34-10"><a href="cnsig.html#cb34-10"></a><span class="co">##                     sex = &quot;female&quot;, &quot;male&quot;))</span></span>
<span id="cb34-11"><a href="cnsig.html#cb34-11"></a></span>
<span id="cb34-12"><a href="cnsig.html#cb34-12"></a><span class="co"># Load toy dataset of absolute copynumber profile</span></span>
<span id="cb34-13"><a href="cnsig.html#cb34-13"></a><span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;toy_segTab.RData&quot;</span>,</span>
<span id="cb34-14"><a href="cnsig.html#cb34-14"></a>  <span class="dt">package =</span> <span class="st">&quot;sigminer&quot;</span>, <span class="dt">mustWork =</span> <span class="ot">TRUE</span></span>
<span id="cb34-15"><a href="cnsig.html#cb34-15"></a>))</span>
<span id="cb34-16"><a href="cnsig.html#cb34-16"></a>cn &lt;-<span class="st"> </span><span class="kw">read_copynumber</span>(segTabs,</span>
<span id="cb34-17"><a href="cnsig.html#cb34-17"></a>  <span class="dt">seg_cols =</span> <span class="kw">c</span>(<span class="st">&quot;chromosome&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>, <span class="st">&quot;segVal&quot;</span>),</span>
<span id="cb34-18"><a href="cnsig.html#cb34-18"></a>  <span class="dt">genome_build =</span> <span class="st">&quot;hg19&quot;</span>, <span class="dt">complement =</span> <span class="ot">FALSE</span>, <span class="dt">verbose =</span> <span class="ot">TRUE</span></span>
<span id="cb34-19"><a href="cnsig.html#cb34-19"></a>)</span>
<span id="cb34-20"><a href="cnsig.html#cb34-20"></a><span class="co">#&gt; [36mℹ[39m [2020-10-08 18:02:08]: Started.</span></span>
<span id="cb34-21"><a href="cnsig.html#cb34-21"></a><span class="co">#&gt; [36mℹ[39m [2020-10-08 18:02:08]: Genome build  : hg19.</span></span>
<span id="cb34-22"><a href="cnsig.html#cb34-22"></a><span class="co">#&gt; [36mℹ[39m [2020-10-08 18:02:08]: Genome measure: called.</span></span>
<span id="cb34-23"><a href="cnsig.html#cb34-23"></a><span class="co">#&gt; [32m✓[39m [2020-10-08 18:02:08]: Chromosome size database for build obtained.</span></span>
<span id="cb34-24"><a href="cnsig.html#cb34-24"></a><span class="co">#&gt; [36mℹ[39m [2020-10-08 18:02:08]: Reading input.</span></span>
<span id="cb34-25"><a href="cnsig.html#cb34-25"></a><span class="co">#&gt; [32m✓[39m [2020-10-08 18:02:08]: A data frame as input detected.</span></span>
<span id="cb34-26"><a href="cnsig.html#cb34-26"></a><span class="co">#&gt; [32m✓[39m [2020-10-08 18:02:08]: Column names checked.</span></span>
<span id="cb34-27"><a href="cnsig.html#cb34-27"></a><span class="co">#&gt; [32m✓[39m [2020-10-08 18:02:08]: Column order set.</span></span>
<span id="cb34-28"><a href="cnsig.html#cb34-28"></a><span class="co">#&gt; [32m✓[39m [2020-10-08 18:02:08]: Chromosomes unified.</span></span>
<span id="cb34-29"><a href="cnsig.html#cb34-29"></a><span class="co">#&gt; [32m✓[39m [2020-10-08 18:02:08]: Data imported.</span></span>
<span id="cb34-30"><a href="cnsig.html#cb34-30"></a><span class="co">#&gt; [36mℹ[39m [2020-10-08 18:02:08]: Segments info:</span></span>
<span id="cb34-31"><a href="cnsig.html#cb34-31"></a><span class="co">#&gt; [36mℹ[39m [2020-10-08 18:02:08]:     Keep - 467</span></span>
<span id="cb34-32"><a href="cnsig.html#cb34-32"></a><span class="co">#&gt; [36mℹ[39m [2020-10-08 18:02:08]:     Drop - 0</span></span>
<span id="cb34-33"><a href="cnsig.html#cb34-33"></a><span class="co">#&gt; [32m✓[39m [2020-10-08 18:02:08]: Segments sorted.</span></span>
<span id="cb34-34"><a href="cnsig.html#cb34-34"></a><span class="co">#&gt; [36mℹ[39m [2020-10-08 18:02:08]: Joining adjacent segments with same copy number value. Be patient...</span></span>
<span id="cb34-35"><a href="cnsig.html#cb34-35"></a><span class="co">#&gt; [32m✓[39m [2020-10-08 18:02:09]: 400 segments left after joining.</span></span>
<span id="cb34-36"><a href="cnsig.html#cb34-36"></a><span class="co">#&gt; [32m✓[39m [2020-10-08 18:02:09]: Segmental table cleaned.</span></span>
<span id="cb34-37"><a href="cnsig.html#cb34-37"></a><span class="co">#&gt; [36mℹ[39m [2020-10-08 18:02:09]: Annotating.</span></span>
<span id="cb34-38"><a href="cnsig.html#cb34-38"></a><span class="co">#&gt; [32m✓[39m [2020-10-08 18:02:09]: Annotation done.</span></span>
<span id="cb34-39"><a href="cnsig.html#cb34-39"></a><span class="co">#&gt; [36mℹ[39m [2020-10-08 18:02:09]: Summarizing per sample.</span></span>
<span id="cb34-40"><a href="cnsig.html#cb34-40"></a><span class="co">#&gt; [32m✓[39m [2020-10-08 18:02:09]: Summarized.</span></span>
<span id="cb34-41"><a href="cnsig.html#cb34-41"></a><span class="co">#&gt; [36mℹ[39m [2020-10-08 18:02:09]: Generating CopyNumber object.</span></span>
<span id="cb34-42"><a href="cnsig.html#cb34-42"></a><span class="co">#&gt; [32m✓[39m [2020-10-08 18:02:09]: Generated.</span></span>
<span id="cb34-43"><a href="cnsig.html#cb34-43"></a><span class="co">#&gt; [36mℹ[39m [2020-10-08 18:02:09]: Validating object.</span></span>
<span id="cb34-44"><a href="cnsig.html#cb34-44"></a><span class="co">#&gt; [32m✓[39m [2020-10-08 18:02:09]: Done.</span></span>
<span id="cb34-45"><a href="cnsig.html#cb34-45"></a><span class="co">#&gt; [36mℹ[39m [2020-10-08 18:02:09]: 0.233 secs elapsed.</span></span>
<span id="cb34-46"><a href="cnsig.html#cb34-46"></a>cn</span>
<span id="cb34-47"><a href="cnsig.html#cb34-47"></a><span class="co">#&gt; An object of class CopyNumber </span></span>
<span id="cb34-48"><a href="cnsig.html#cb34-48"></a><span class="co">#&gt; =============================</span></span>
<span id="cb34-49"><a href="cnsig.html#cb34-49"></a><span class="co">#&gt;                           sample n_of_seg n_of_cnv n_of_amp n_of_del n_of_vchr cna_burden</span></span>
<span id="cb34-50"><a href="cnsig.html#cb34-50"></a><span class="co">#&gt;  1: TCGA-DF-A2KN-01A-11D-A17U-01       33        6        5        1         4      0.000</span></span>
<span id="cb34-51"><a href="cnsig.html#cb34-51"></a><span class="co">#&gt;  2: TCGA-19-2621-01B-01D-0911-01       33        8        5        3         5      0.099</span></span>
<span id="cb34-52"><a href="cnsig.html#cb34-52"></a><span class="co">#&gt;  3: TCGA-B6-A0X5-01A-21D-A107-01       28        8        4        4         2      0.087</span></span>
<span id="cb34-53"><a href="cnsig.html#cb34-53"></a><span class="co">#&gt;  4: TCGA-A8-A07S-01A-11D-A036-01       38       11        2        9         4      0.112</span></span>
<span id="cb34-54"><a href="cnsig.html#cb34-54"></a><span class="co">#&gt;  5: TCGA-26-6174-01A-21D-1842-01       43       13        8        5         8      0.119</span></span>
<span id="cb34-55"><a href="cnsig.html#cb34-55"></a><span class="co">#&gt;  6: TCGA-CV-7432-01A-11D-2128-01       40       16        7        9         9      0.198</span></span>
<span id="cb34-56"><a href="cnsig.html#cb34-56"></a><span class="co">#&gt;  7: TCGA-06-0644-01A-02D-0310-01       46       19        5       14         8      0.165</span></span>
<span id="cb34-57"><a href="cnsig.html#cb34-57"></a><span class="co">#&gt;  8: TCGA-A5-A0G2-01A-11D-A042-01       39       21        5       16        10      0.393</span></span>
<span id="cb34-58"><a href="cnsig.html#cb34-58"></a><span class="co">#&gt;  9: TCGA-99-7458-01A-11D-2035-01       48       26       10       16        13      0.318</span></span>
<span id="cb34-59"><a href="cnsig.html#cb34-59"></a><span class="co">#&gt; 10: TCGA-05-4417-01A-22D-1854-01       52       37       33        4        17      0.654</span></span></code></pre></div>
<blockquote>
<p>Currently, you can refer to <code>extract_facets_cnv()</code> and <code>extract_seqz_cnv()</code> in <a href="https://github.com/ShixiangWang/prad_signature/blob/master/analysis/src/99-functions.R" class="uri">https://github.com/ShixiangWang/prad_signature/blob/master/analysis/src/99-functions.R</a> to
see how to get tidy data from a result directory of FACETS or Sequenza.</p>
</blockquote>
</div>
<div id="tally-components" class="section level2">
<h2><span class="header-section-number">3.3</span> Tally Components</h2>
<p>Currently, there are two methods for generating sample-by-component matrix, i.e. “W” or “M”.</p>
<p>Option <code>sigminer.copynumber.max</code> is used to control the processing of max copy number values. Run <code>?sig_tally</code> to see more.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="cnsig.html#cb35-1"></a><span class="co">## Even you set max_copynumber = 20 in read_copynumber(),</span></span>
<span id="cb35-2"><a href="cnsig.html#cb35-2"></a><span class="co">## the segmental copy number may be greater than 20</span></span>
<span id="cb35-3"><a href="cnsig.html#cb35-3"></a><span class="co">## because for male samples, the X/Y segmental copy number</span></span>
<span id="cb35-4"><a href="cnsig.html#cb35-4"></a><span class="co">## values will be doubled in tally process.</span></span>
<span id="cb35-5"><a href="cnsig.html#cb35-5"></a><span class="co">## This setting will make copy number values of all segments</span></span>
<span id="cb35-6"><a href="cnsig.html#cb35-6"></a><span class="co">## not greater than 20.</span></span>
<span id="cb35-7"><a href="cnsig.html#cb35-7"></a><span class="kw">options</span>(<span class="dt">sigminer.copynumber.max =</span> <span class="dv">20</span>)</span>
<span id="cb35-8"><a href="cnsig.html#cb35-8"></a></span>
<span id="cb35-9"><a href="cnsig.html#cb35-9"></a><span class="co"># Load copy number object</span></span>
<span id="cb35-10"><a href="cnsig.html#cb35-10"></a><span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;toy_copynumber.RData&quot;</span>,</span>
<span id="cb35-11"><a href="cnsig.html#cb35-11"></a>  <span class="dt">package =</span> <span class="st">&quot;sigminer&quot;</span>, <span class="dt">mustWork =</span> <span class="ot">TRUE</span></span>
<span id="cb35-12"><a href="cnsig.html#cb35-12"></a>))</span>
<span id="cb35-13"><a href="cnsig.html#cb35-13"></a></span>
<span id="cb35-14"><a href="cnsig.html#cb35-14"></a><span class="co"># Use method designed by Wang, Shixiang et al.</span></span>
<span id="cb35-15"><a href="cnsig.html#cb35-15"></a>cn_tally_W &lt;-<span class="st"> </span><span class="kw">sig_tally</span>(cn, <span class="dt">method =</span> <span class="st">&quot;W&quot;</span>)</span>
<span id="cb35-16"><a href="cnsig.html#cb35-16"></a><span class="co"># Use method designed by Macintyre et al.</span></span>
<span id="cb35-17"><a href="cnsig.html#cb35-17"></a>cn_tally_M &lt;-<span class="st"> </span><span class="kw">sig_tally</span>(cn, <span class="dt">method =</span> <span class="st">&quot;M&quot;</span>)</span></code></pre></div>
<blockquote>
<p>You can set <code>options(sigminer.sex = "male", sigminer.copynumber.max = 20)</code> at the top of your code
to avoid setting them in two places.</p>
<p><strong>Of note</strong>, the <code>sigminer.copynumber.max</code> option only has effect on <code>sig_tally()</code> with method “W” or “M”,
the <code>sigminer.sex</code> option has effects on <code>read_copynumber()</code> and <code>sig_tally()</code> with method “W” or “M”.</p>
</blockquote>
<p>This step return a <code>list</code> containing information about copy number features, components and matrix for NMF etc.</p>
</div>
<div id="extract-signatures" class="section level2">
<h2><span class="header-section-number">3.4</span> Extract Signatures</h2>
<p>When you get the matrix, you can just do the signature extraction as SBS signatures (see chapter Chapter <a href="sbssig.html#sbssig">2</a>). So here we won’t talk much.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="cnsig.html#cb36-1"></a>cn_tally_W<span class="op">$</span>nmf_matrix[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</span>
<span id="cb36-2"><a href="cnsig.html#cb36-2"></a><span class="co">#&gt;                              BP10MB[0] BP10MB[1] BP10MB[2] BP10MB[3] BP10MB[4]</span></span>
<span id="cb36-3"><a href="cnsig.html#cb36-3"></a><span class="co">#&gt; TCGA-05-4417-01A-22D-1854-01       275        20         5         0         0</span></span>
<span id="cb36-4"><a href="cnsig.html#cb36-4"></a><span class="co">#&gt; TCGA-06-0644-01A-02D-0310-01       289         5         4         0         1</span></span>
<span id="cb36-5"><a href="cnsig.html#cb36-5"></a><span class="co">#&gt; TCGA-19-2621-01B-01D-0911-01       294         2         3         1         0</span></span>
<span id="cb36-6"><a href="cnsig.html#cb36-6"></a><span class="co">#&gt; TCGA-26-6174-01A-21D-1842-01       288         4         7         1         0</span></span>
<span id="cb36-7"><a href="cnsig.html#cb36-7"></a><span class="co">#&gt; TCGA-99-7458-01A-11D-2035-01       284         9         5         1         1</span></span></code></pre></div>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="cnsig.html#cb37-1"></a><span class="co"># library(NMF)</span></span>
<span id="cb37-2"><a href="cnsig.html#cb37-2"></a>sig_w &lt;-<span class="st"> </span><span class="kw">sig_extract</span>(cn_tally_W<span class="op">$</span>nmf_matrix, <span class="dt">n_sig =</span> <span class="dv">2</span>, <span class="dt">pConstant =</span> <span class="fl">1e-13</span>)</span></code></pre></div>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-macintyre2018copy">
<p>Macintyre, Geoff, Teodora E Goranova, Dilrini De Silva, Darren Ennis, Anna M Piskorz, Matthew Eldridge, Daoud Sie, et al. 2018. “Copy Number Signatures and Mutational Processes in Ovarian Carcinoma.” <em>Nature Genetics</em> 50 (9): 1262–70.</p>
</div>
<div id="ref-WangSX2020">
<p>Wang, Shixiang, Huimin Li, Minfang Song, Zaoke He, Tao Wu, Xuan Wang, Ziyu Tao, Kai Wu, and Xue-Song Liu. 2020. “Copy Number Signature Analyses in Prostate Cancer Reveal Distinct Etiologies and Clinical Outcomes.” <em>medRxiv</em>. <a href="https://doi.org/10.1101/2020.04.27.20082404">https://doi.org/10.1101/2020.04.27.20082404</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="sbssig.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="sigfit.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/ShixiangWang/sigminer-doc/edit/master/02-copy-signature-identification.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["sigminer-doc.pdf", "sigminer-doc.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
