<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Group Analysis and Visualization | Sigminer: A Scalable Toolkit to Extract, Analyze and Visualize Mutational Signatures</title>
  <meta name="description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Group Analysis and Visualization | Sigminer: A Scalable Toolkit to Extract, Analyze and Visualize Mutational Signatures" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  <meta name="github-repo" content="ShixiangWang/sigminer-doc" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Group Analysis and Visualization | Sigminer: A Scalable Toolkit to Extract, Analyze and Visualize Mutational Signatures" />
  
  <meta name="twitter:description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  

<meta name="author" content="Shixiang Wang, ShanghaiTech University" />
<meta name="author" content="Xue-Song Liu, ShanghaiTech University" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="association-analysis.html"/>
<link rel="next" href="datasets.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.15/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">sigminer: Signature Miner</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#biological-significance-of-mutational-signature"><i class="fa fa-check"></i><b>1.1</b> Biological Significance of Mutational Signature</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#sigminer"><i class="fa fa-check"></i><b>1.2</b> Sigminer</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.3</b> Installation</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#issues-or-suggestions"><i class="fa fa-check"></i><b>1.4</b> Issues or Suggestions</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#preparation"><i class="fa fa-check"></i><b>1.5</b> Preparation</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#overview-of-contents"><i class="fa fa-check"></i><b>1.6</b> Overview of Contents</a></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#citation-and-license"><i class="fa fa-check"></i><b>1.7</b> Citation and LICENSE</a></li>
</ul></li>
<li class="part"><span><b>I Common Workflow</b></span></li>
<li class="chapter" data-level="2" data-path="sbssig.html"><a href="sbssig.html"><i class="fa fa-check"></i><b>2</b> COSMIC Signature Identification</a><ul>
<li class="chapter" data-level="2.1" data-path="sbssig.html"><a href="sbssig.html#read-data"><i class="fa fa-check"></i><b>2.1</b> Read Data</a><ul>
<li class="chapter" data-level="2.1.1" data-path="sbssig.html"><a href="sbssig.html#vcf-as-input"><i class="fa fa-check"></i><b>2.1.1</b> VCF as input</a></li>
<li class="chapter" data-level="2.1.2" data-path="sbssig.html"><a href="sbssig.html#maf-as-input"><i class="fa fa-check"></i><b>2.1.2</b> MAF as input</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="sbssig.html"><a href="sbssig.html#tally-components"><i class="fa fa-check"></i><b>2.2</b> Tally Components</a><ul>
<li class="chapter" data-level="2.2.1" data-path="sbssig.html"><a href="sbssig.html#the-most-common-96-components"><i class="fa fa-check"></i><b>2.2.1</b> The most common 96 components</a></li>
<li class="chapter" data-level="2.2.2" data-path="sbssig.html"><a href="sbssig.html#other-situations"><i class="fa fa-check"></i><b>2.2.2</b> Other Situations</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="sbssig.html"><a href="sbssig.html#extract-signatures"><i class="fa fa-check"></i><b>2.3</b> Extract Signatures</a><ul>
<li class="chapter" data-level="2.3.1" data-path="sbssig.html"><a href="sbssig.html#method-1-estimate-signature-number"><i class="fa fa-check"></i><b>2.3.1</b> Method 1: Estimate Signature Number</a></li>
<li class="chapter" data-level="2.3.2" data-path="sbssig.html"><a href="sbssig.html#method-1-extract-signatures"><i class="fa fa-check"></i><b>2.3.2</b> Method 1: Extract Signatures</a></li>
<li class="chapter" data-level="2.3.3" data-path="sbssig.html"><a href="sbssig.html#method-2-auto-extraction"><i class="fa fa-check"></i><b>2.3.3</b> Method 2: Auto-extraction</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="sbssig.html"><a href="sbssig.html#match-signatures"><i class="fa fa-check"></i><b>2.4</b> Match Signatures</a></li>
<li class="chapter" data-level="2.5" data-path="sbssig.html"><a href="sbssig.html#operate-signature"><i class="fa fa-check"></i><b>2.5</b> Operate <code>Signature</code></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cnsig.html"><a href="cnsig.html"><i class="fa fa-check"></i><b>3</b> Copy Number Signature Identification</a><ul>
<li class="chapter" data-level="3.1" data-path="cnsig.html"><a href="cnsig.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="sbssig.html"><a href="sbssig.html#read-data"><i class="fa fa-check"></i><b>3.2</b> Read Data</a></li>
<li class="chapter" data-level="3.3" data-path="sbssig.html"><a href="sbssig.html#tally-components"><i class="fa fa-check"></i><b>3.3</b> Tally Components</a></li>
<li class="chapter" data-level="3.4" data-path="sbssig.html"><a href="sbssig.html#extract-signatures"><i class="fa fa-check"></i><b>3.4</b> Extract Signatures</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sigfit.html"><a href="sigfit.html"><i class="fa fa-check"></i><b>4</b> Signature Fit: Sample Signature Exposure Quantification and Analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="sigfit.html"><a href="sigfit.html#fit-signatures-from-reference-databases"><i class="fa fa-check"></i><b>4.1</b> Fit Signatures from reference databases</a></li>
<li class="chapter" data-level="4.2" data-path="sigfit.html"><a href="sigfit.html#fit-custom-signatures"><i class="fa fa-check"></i><b>4.2</b> Fit Custom Signatures</a></li>
<li class="chapter" data-level="4.3" data-path="sigfit.html"><a href="sigfit.html#performance-comparison"><i class="fa fa-check"></i><b>4.3</b> Performance Comparison</a></li>
<li class="chapter" data-level="4.4" data-path="sigfit.html"><a href="sigfit.html#estimate-exposure-stability-by-bootstrap"><i class="fa fa-check"></i><b>4.4</b> Estimate Exposure Stability by Bootstrap</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="subtyping.html"><a href="subtyping.html"><i class="fa fa-check"></i><b>5</b> Subtype Prediction</a></li>
<li class="chapter" data-level="6" data-path="sigflow.html"><a href="sigflow.html"><i class="fa fa-check"></i><b>6</b> Sigflow Pipeline</a><ul>
<li class="chapter" data-level="6.1" data-path="sigflow.html"><a href="sigflow.html#cancer-type-specific-signature-index-database"><i class="fa fa-check"></i><b>6.1</b> Cancer type specific signature index database</a></li>
</ul></li>
<li class="part"><span><b>II Target Visualization</b></span></li>
<li class="chapter" data-level="7" data-path="cnobject.html"><a href="cnobject.html"><i class="fa fa-check"></i><b>7</b> CopyNumber Object</a><ul>
<li class="chapter" data-level="7.1" data-path="cnobject.html"><a href="cnobject.html#profile"><i class="fa fa-check"></i><b>7.1</b> Profile</a></li>
<li class="chapter" data-level="7.2" data-path="cnobject.html"><a href="cnobject.html#distribution"><i class="fa fa-check"></i><b>7.2</b> Distribution</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="sigobject.html"><a href="sigobject.html"><i class="fa fa-check"></i><b>8</b> Signature Object</a><ul>
<li class="chapter" data-level="8.1" data-path="sigobject.html"><a href="sigobject.html#signature-profile"><i class="fa fa-check"></i><b>8.1</b> Signature Profile</a><ul>
<li class="chapter" data-level="8.1.1" data-path="sigobject.html"><a href="sigobject.html#sbs-signature-profile"><i class="fa fa-check"></i><b>8.1.1</b> SBS Signature profile</a></li>
<li class="chapter" data-level="8.1.2" data-path="sigobject.html"><a href="sigobject.html#copy-number-signature-porfile"><i class="fa fa-check"></i><b>8.1.2</b> Copy Number Signature Porfile</a></li>
<li class="chapter" data-level="8.1.3" data-path="sigobject.html"><a href="sigobject.html#cosmic-signature-profile"><i class="fa fa-check"></i><b>8.1.3</b> COSMIC Signature Profile</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="sigobject.html"><a href="sigobject.html#exposure-profile"><i class="fa fa-check"></i><b>8.2</b> Exposure Profile</a></li>
<li class="chapter" data-level="8.3" data-path="sigobject.html"><a href="sigobject.html#consensus-map"><i class="fa fa-check"></i><b>8.3</b> Consensus Map</a></li>
<li class="chapter" data-level="8.4" data-path="sigobject.html"><a href="sigobject.html#catalogue-profile"><i class="fa fa-check"></i><b>8.4</b> Catalogue Profile</a></li>
</ul></li>
<li class="part"><span><b>III Universal Analysis</b></span></li>
<li class="chapter" data-level="9" data-path="association-analysis.html"><a href="association-analysis.html"><i class="fa fa-check"></i><b>9</b> Association Analysis and Visualization</a><ul>
<li class="chapter" data-level="9.1" data-path="association-analysis.html"><a href="association-analysis.html#general-numeric-association"><i class="fa fa-check"></i><b>9.1</b> General numeric association</a></li>
<li class="chapter" data-level="9.2" data-path="association-analysis.html"><a href="association-analysis.html#comprehensive-association"><i class="fa fa-check"></i><b>9.2</b> Comprehensive association</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="group-analysis.html"><a href="group-analysis.html"><i class="fa fa-check"></i><b>10</b> Group Analysis and Visualization</a><ul>
<li class="chapter" data-level="10.1" data-path="group-analysis.html"><a href="group-analysis.html#group-generation"><i class="fa fa-check"></i><b>10.1</b> Group Generation</a></li>
<li class="chapter" data-level="10.2" data-path="group-analysis.html"><a href="group-analysis.html#group-comparison-analysis"><i class="fa fa-check"></i><b>10.2</b> Group Comparison Analysis</a></li>
<li class="chapter" data-level="10.3" data-path="group-analysis.html"><a href="group-analysis.html#group-visualization"><i class="fa fa-check"></i><b>10.3</b> Group Visualization</a></li>
</ul></li>
<li class="part"><span><b>IV Useful Utilities</b></span></li>
<li class="chapter" data-level="11" data-path="datasets.html"><a href="datasets.html"><i class="fa fa-check"></i><b>11</b> Datasets</a><ul>
<li class="chapter" data-level="11.1" data-path="datasets.html"><a href="datasets.html#reference-annotation"><i class="fa fa-check"></i><b>11.1</b> Reference Annotation</a></li>
<li class="chapter" data-level="11.2" data-path="datasets.html"><a href="datasets.html#copy-number-components-setting"><i class="fa fa-check"></i><b>11.2</b> Copy Number components setting</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="convert.html"><a href="convert.html"><i class="fa fa-check"></i><b>12</b> SBS Signature Conversion</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Sigminer: A Scalable Toolkit to Extract, Analyze and Visualize Mutational Signatures</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="group-analysis" class="section level1">
<h1><span class="header-section-number">Chapter 10</span> Group Analysis and Visualization</h1>
<p>Group analysis is a common task in cancer study. <strong>Sigminer</strong> supports dividing samples into multiple groups and comparing genotype/phenotype feature measures.</p>
<div id="group-generation" class="section level2">
<h2><span class="header-section-number">10.1</span> Group Generation</h2>
<p>There are multiple methods to generate groups, including ‘consensus’ (default, can be only used by result from <code>sig_extract()</code>), ‘k-means’ etc. After determining groups, <strong>sigminer</strong> will assign each group to a signature with maximum fraction. We may say a group is <code>Sig_x</code> enriched.</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="group-analysis.html#cb80-1"></a>mt_grps &lt;-<span class="st"> </span><span class="kw">get_groups</span>(mt_sig, <span class="dt">method =</span> <span class="st">&quot;consensus&quot;</span>, <span class="dt">match_consensus =</span> <span class="ot">TRUE</span>)</span>
<span id="cb80-2"><a href="group-analysis.html#cb80-2"></a><span class="co">#&gt; [36mℹ[39m [2020-10-09 00:03:46]: Started.</span></span>
<span id="cb80-3"><a href="group-analysis.html#cb80-3"></a><span class="co">#&gt; [32m✓[39m [2020-10-09 00:03:46]: &#39;Signature&#39; object detected.</span></span>
<span id="cb80-4"><a href="group-analysis.html#cb80-4"></a><span class="co">#&gt; [36mℹ[39m [2020-10-09 00:03:46]: Obtaining clusters from the hierarchical clustering of the consensus matrix...</span></span>
<span id="cb80-5"><a href="group-analysis.html#cb80-5"></a><span class="co">#&gt; [36mℹ[39m [2020-10-09 00:03:46]: Finding the dominant signature of each group...</span></span>
<span id="cb80-6"><a href="group-analysis.html#cb80-6"></a><span class="co">#&gt; =&gt; Generating a table of group and dominant signature:</span></span>
<span id="cb80-7"><a href="group-analysis.html#cb80-7"></a><span class="co">#&gt;    </span></span>
<span id="cb80-8"><a href="group-analysis.html#cb80-8"></a><span class="co">#&gt;     Sig1 Sig2 Sig3 Sig4 Sig5</span></span>
<span id="cb80-9"><a href="group-analysis.html#cb80-9"></a><span class="co">#&gt;   1   14    0    0    0    0</span></span>
<span id="cb80-10"><a href="group-analysis.html#cb80-10"></a><span class="co">#&gt;   2    0    2    4    0    0</span></span>
<span id="cb80-11"><a href="group-analysis.html#cb80-11"></a><span class="co">#&gt;   3    1    0    8   15    0</span></span>
<span id="cb80-12"><a href="group-analysis.html#cb80-12"></a><span class="co">#&gt;   4    0    2    0    0    8</span></span>
<span id="cb80-13"><a href="group-analysis.html#cb80-13"></a><span class="co">#&gt;   5    0   42    3    1    0</span></span>
<span id="cb80-14"><a href="group-analysis.html#cb80-14"></a><span class="co">#&gt; =&gt; Assigning a group to a signature with the maxium fraction (stored in &#39;map_table&#39; attr)...</span></span>
<span id="cb80-15"><a href="group-analysis.html#cb80-15"></a><span class="co">#&gt; [36mℹ[39m [2020-10-09 00:03:46]: Summarizing...</span></span>
<span id="cb80-16"><a href="group-analysis.html#cb80-16"></a><span class="co">#&gt; 	group #1: 14 samples with Sig1 enriched.</span></span>
<span id="cb80-17"><a href="group-analysis.html#cb80-17"></a><span class="co">#&gt; 	group #2: 6 samples with Sig3 enriched.</span></span>
<span id="cb80-18"><a href="group-analysis.html#cb80-18"></a><span class="co">#&gt; 	group #3: 24 samples with Sig4 enriched.</span></span>
<span id="cb80-19"><a href="group-analysis.html#cb80-19"></a><span class="co">#&gt; 	group #4: 10 samples with Sig5 enriched.</span></span>
<span id="cb80-20"><a href="group-analysis.html#cb80-20"></a><span class="co">#&gt; 	group #5: 46 samples with Sig2 enriched.</span></span>
<span id="cb80-21"><a href="group-analysis.html#cb80-21"></a><span class="co">#&gt; [33m![39m [2020-10-09 00:03:46]: The &#39;enrich_sig&#39; column is set to dominant signature in one group, please check and make it consistent with biological meaning (correct it by hand if necessary).</span></span>
<span id="cb80-22"><a href="group-analysis.html#cb80-22"></a><span class="co">#&gt; [36mℹ[39m [2020-10-09 00:03:46]: 0.088 secs elapsed.</span></span>
<span id="cb80-23"><a href="group-analysis.html#cb80-23"></a><span class="kw">head</span>(mt_grps)</span>
<span id="cb80-24"><a href="group-analysis.html#cb80-24"></a><span class="co">#&gt;                          sample group silhouette_width enrich_sig</span></span>
<span id="cb80-25"><a href="group-analysis.html#cb80-25"></a><span class="co">#&gt; 1: TCGA-PE-A5DD-01A-12D-A27P-09     1            0.572       Sig1</span></span>
<span id="cb80-26"><a href="group-analysis.html#cb80-26"></a><span class="co">#&gt; 2: TCGA-D8-A1JJ-01A-31D-A14K-09     1            0.279       Sig1</span></span>
<span id="cb80-27"><a href="group-analysis.html#cb80-27"></a><span class="co">#&gt; 3: TCGA-BH-A18K-01A-11D-A12B-09     1            0.125       Sig1</span></span>
<span id="cb80-28"><a href="group-analysis.html#cb80-28"></a><span class="co">#&gt; 4: TCGA-AC-A2FO-01A-11D-A17W-09     1            0.259       Sig1</span></span>
<span id="cb80-29"><a href="group-analysis.html#cb80-29"></a><span class="co">#&gt; 5: TCGA-E2-A1IH-01A-11D-A188-09     1            0.535       Sig1</span></span>
<span id="cb80-30"><a href="group-analysis.html#cb80-30"></a><span class="co">#&gt; 6: TCGA-E2-A152-01A-11D-A12B-09     1            0.482       Sig1</span></span></code></pre></div>
<p>The returned sample orders match sample orders in clustered consensus matrix.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="group-analysis.html#cb81-1"></a><span class="kw">show_sig_consensusmap</span>(mt_sig)</span></code></pre></div>
<p><img src="sigminer-doc_files/figure-html/unnamed-chunk-93-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Sometimes, the mapping between groups and enriched signatures may not right. Users should check it and even correct it manually.</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="group-analysis.html#cb82-1"></a><span class="kw">attr</span>(mt_grps, <span class="st">&quot;map_table&quot;</span>)</span>
<span id="cb82-2"><a href="group-analysis.html#cb82-2"></a><span class="co">#&gt;    </span></span>
<span id="cb82-3"><a href="group-analysis.html#cb82-3"></a><span class="co">#&gt;     Sig1 Sig2 Sig3 Sig4 Sig5</span></span>
<span id="cb82-4"><a href="group-analysis.html#cb82-4"></a><span class="co">#&gt;   1   14    0    0    0    0</span></span>
<span id="cb82-5"><a href="group-analysis.html#cb82-5"></a><span class="co">#&gt;   2    0    2    4    0    0</span></span>
<span id="cb82-6"><a href="group-analysis.html#cb82-6"></a><span class="co">#&gt;   3    1    0    8   15    0</span></span>
<span id="cb82-7"><a href="group-analysis.html#cb82-7"></a><span class="co">#&gt;   4    0    2    0    0    8</span></span>
<span id="cb82-8"><a href="group-analysis.html#cb82-8"></a><span class="co">#&gt;   5    0   42    3    1    0</span></span></code></pre></div>
</div>
<div id="group-comparison-analysis" class="section level2">
<h2><span class="header-section-number">10.2</span> Group Comparison Analysis</h2>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="group-analysis.html#cb83-1"></a><span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;toy_copynumber_signature_by_M.RData&quot;</span>,</span>
<span id="cb83-2"><a href="group-analysis.html#cb83-2"></a>  <span class="dt">package =</span> <span class="st">&quot;sigminer&quot;</span>, <span class="dt">mustWork =</span> <span class="ot">TRUE</span></span>
<span id="cb83-3"><a href="group-analysis.html#cb83-3"></a>))</span>
<span id="cb83-4"><a href="group-analysis.html#cb83-4"></a></span>
<span id="cb83-5"><a href="group-analysis.html#cb83-5"></a><span class="co"># Assign samples to clusters</span></span>
<span id="cb83-6"><a href="group-analysis.html#cb83-6"></a>groups &lt;-<span class="st"> </span><span class="kw">get_groups</span>(sig, <span class="dt">method =</span> <span class="st">&quot;k-means&quot;</span>)</span>
<span id="cb83-7"><a href="group-analysis.html#cb83-7"></a><span class="co">#&gt; [36mℹ[39m [2020-10-09 00:03:46]: Started.</span></span>
<span id="cb83-8"><a href="group-analysis.html#cb83-8"></a><span class="co">#&gt; [32m✓[39m [2020-10-09 00:03:46]: &#39;Signature&#39; object detected.</span></span>
<span id="cb83-9"><a href="group-analysis.html#cb83-9"></a><span class="co">#&gt; [36mℹ[39m [2020-10-09 00:03:46]: Running k-means with 2 clusters...</span></span>
<span id="cb83-10"><a href="group-analysis.html#cb83-10"></a><span class="co">#&gt; [36mℹ[39m [2020-10-09 00:03:46]: Generating a table of group and signature contribution (stored in &#39;map_table&#39; attr):</span></span>
<span id="cb83-11"><a href="group-analysis.html#cb83-11"></a><span class="co">#&gt;          Sig1      Sig2</span></span>
<span id="cb83-12"><a href="group-analysis.html#cb83-12"></a><span class="co">#&gt; 1 0.003428449 0.9965716</span></span>
<span id="cb83-13"><a href="group-analysis.html#cb83-13"></a><span class="co">#&gt; 2 0.031799383 0.9682006</span></span>
<span id="cb83-14"><a href="group-analysis.html#cb83-14"></a><span class="co">#&gt; [36mℹ[39m [2020-10-09 00:03:46]: Assigning a group to a signature with the maximum fraction...</span></span>
<span id="cb83-15"><a href="group-analysis.html#cb83-15"></a><span class="co">#&gt; [36mℹ[39m [2020-10-09 00:03:46]: Summarizing...</span></span>
<span id="cb83-16"><a href="group-analysis.html#cb83-16"></a><span class="co">#&gt; 	group #1: 6 samples with Sig2 enriched.</span></span>
<span id="cb83-17"><a href="group-analysis.html#cb83-17"></a><span class="co">#&gt; 	group #2: 4 samples with Sig2 enriched.</span></span>
<span id="cb83-18"><a href="group-analysis.html#cb83-18"></a><span class="co">#&gt; [33m![39m [2020-10-09 00:03:46]: The &#39;enrich_sig&#39; column is set to dominant signature in one group, please check and make it consistent with biological meaning (correct it by hand if necessary).</span></span>
<span id="cb83-19"><a href="group-analysis.html#cb83-19"></a><span class="co">#&gt; [36mℹ[39m [2020-10-09 00:03:46]: 0.042 secs elapsed.</span></span>
<span id="cb83-20"><a href="group-analysis.html#cb83-20"></a></span>
<span id="cb83-21"><a href="group-analysis.html#cb83-21"></a><span class="kw">set.seed</span>(<span class="dv">1234</span>)</span>
<span id="cb83-22"><a href="group-analysis.html#cb83-22"></a></span>
<span id="cb83-23"><a href="group-analysis.html#cb83-23"></a>groups<span class="op">$</span>prob &lt;-<span class="st"> </span><span class="kw">rnorm</span>(<span class="dv">10</span>)</span>
<span id="cb83-24"><a href="group-analysis.html#cb83-24"></a>groups<span class="op">$</span>new_group &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="kw">c</span>(<span class="st">&quot;1&quot;</span>, <span class="st">&quot;2&quot;</span>, <span class="st">&quot;3&quot;</span>, <span class="st">&quot;4&quot;</span>, <span class="ot">NA</span>), <span class="dt">size =</span> <span class="kw">nrow</span>(groups), <span class="dt">replace =</span> <span class="ot">TRUE</span>)</span>
<span id="cb83-25"><a href="group-analysis.html#cb83-25"></a></span>
<span id="cb83-26"><a href="group-analysis.html#cb83-26"></a><span class="co"># Compare groups (filter NAs for categorical coloumns)</span></span>
<span id="cb83-27"><a href="group-analysis.html#cb83-27"></a>groups.cmp &lt;-<span class="st"> </span><span class="kw">get_group_comparison</span>(groups[, <span class="dv">-1</span>],</span>
<span id="cb83-28"><a href="group-analysis.html#cb83-28"></a>  <span class="dt">col_group =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb83-29"><a href="group-analysis.html#cb83-29"></a>  <span class="dt">cols_to_compare =</span> <span class="kw">c</span>(<span class="st">&quot;prob&quot;</span>, <span class="st">&quot;new_group&quot;</span>),</span>
<span id="cb83-30"><a href="group-analysis.html#cb83-30"></a>  <span class="dt">type =</span> <span class="kw">c</span>(<span class="st">&quot;co&quot;</span>, <span class="st">&quot;ca&quot;</span>), <span class="dt">verbose =</span> <span class="ot">TRUE</span></span>
<span id="cb83-31"><a href="group-analysis.html#cb83-31"></a>)</span>
<span id="cb83-32"><a href="group-analysis.html#cb83-32"></a><span class="co">#&gt; Treat prob as continuous variable.</span></span>
<span id="cb83-33"><a href="group-analysis.html#cb83-33"></a><span class="co">#&gt; Treat new_group as categorical variable.</span></span>
<span id="cb83-34"><a href="group-analysis.html#cb83-34"></a></span>
<span id="cb83-35"><a href="group-analysis.html#cb83-35"></a><span class="co"># Compare groups (Set NAs of categorical columns to &#39;Rest&#39;)</span></span>
<span id="cb83-36"><a href="group-analysis.html#cb83-36"></a>groups.cmp2 &lt;-<span class="st"> </span><span class="kw">get_group_comparison</span>(groups[, <span class="dv">-1</span>],</span>
<span id="cb83-37"><a href="group-analysis.html#cb83-37"></a>  <span class="dt">col_group =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb83-38"><a href="group-analysis.html#cb83-38"></a>  <span class="dt">cols_to_compare =</span> <span class="kw">c</span>(<span class="st">&quot;prob&quot;</span>, <span class="st">&quot;new_group&quot;</span>),</span>
<span id="cb83-39"><a href="group-analysis.html#cb83-39"></a>  <span class="dt">type =</span> <span class="kw">c</span>(<span class="st">&quot;co&quot;</span>, <span class="st">&quot;ca&quot;</span>), <span class="dt">NAs =</span> <span class="st">&quot;Rest&quot;</span>, <span class="dt">verbose =</span> <span class="ot">TRUE</span></span>
<span id="cb83-40"><a href="group-analysis.html#cb83-40"></a>)</span>
<span id="cb83-41"><a href="group-analysis.html#cb83-41"></a><span class="co">#&gt; Treat prob as continuous variable.</span></span>
<span id="cb83-42"><a href="group-analysis.html#cb83-42"></a><span class="co">#&gt; Treat new_group as categorical variable.</span></span></code></pre></div>
</div>
<div id="group-visualization" class="section level2">
<h2><span class="header-section-number">10.3</span> Group Visualization</h2>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="group-analysis.html#cb84-1"></a>ggcomp &lt;-<span class="st"> </span><span class="kw">show_group_comparison</span>(groups.cmp2)</span>
<span id="cb84-2"><a href="group-analysis.html#cb84-2"></a>ggcomp<span class="op">$</span>co_comb</span></code></pre></div>
<p><img src="sigminer-doc_files/figure-html/unnamed-chunk-96-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="group-analysis.html#cb85-1"></a>ggcomp<span class="op">$</span>ca_comb</span></code></pre></div>
<p><img src="sigminer-doc_files/figure-html/unnamed-chunk-96-2.png" width="672" style="display: block; margin: auto;" /></p>

</div>
</div>



            </section>

          </div>
        </div>
      </div>
<a href="association-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="datasets.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/ShixiangWang/sigminer-doc/edit/master/05-analysis-and-visualization.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["sigminer-doc.pdf", "sigminer-doc.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
