<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>sigminer: Genomic Alteration Signature Analysis in R</title>
  <meta name="description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="sigminer: Genomic Alteration Signature Analysis in R" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  <meta name="github-repo" content="ShixiangWang/sigminer-doc" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="sigminer: Genomic Alteration Signature Analysis in R" />
  
  <meta name="twitter:description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  

<meta name="author" content="Shixiang Wang, ShanghaiTech University" />
<meta name="author" content="Xue-Song Liu, ShanghaiTech University" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="part-signature-identification.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">sigminer: Signature Miner</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i>Installation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#issues-or-suggestions"><i class="fa fa-check"></i>Issues or Suggestions</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#preparation"><i class="fa fa-check"></i>Preparation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#citation-and-license"><i class="fa fa-check"></i>Citation and LICENSE</a></li>
</ul></li>
<li class="part"><span><b>I Signature Identification</b></span></li>
<li class="chapter" data-level="1" data-path="sbssig.html"><a href="sbssig.html"><i class="fa fa-check"></i><b>1</b> SBS Signature Identification</a>
<ul>
<li class="chapter" data-level="" data-path="sbssig.html"><a href="sbssig.html#read-data"><i class="fa fa-check"></i>Read Data</a></li>
<li class="chapter" data-level="" data-path="sbssig.html"><a href="sbssig.html#tally-components"><i class="fa fa-check"></i>Tally Components</a></li>
<li class="chapter" data-level="" data-path="sbssig.html"><a href="sbssig.html#extract-signatures"><i class="fa fa-check"></i>Extract Signatures</a>
<ul>
<li class="chapter" data-level="" data-path="sbssig.html"><a href="sbssig.html#method-1-estimate-signature-number"><i class="fa fa-check"></i>Method 1: Estimate Signature Number</a></li>
<li class="chapter" data-level="" data-path="sbssig.html"><a href="sbssig.html#method-1-extract-signatures"><i class="fa fa-check"></i>Method 1: Extract Signatures</a></li>
<li class="chapter" data-level="" data-path="sbssig.html"><a href="sbssig.html#method-2-auto-extract"><i class="fa fa-check"></i>Method 2: Auto-extract</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="sbssig.html"><a href="sbssig.html#match-signatures"><i class="fa fa-check"></i>Match Signatures</a></li>
<li><a href="sbssig.html#operate-signature">Operate <code>Signature</code></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cnsig.html"><a href="cnsig.html"><i class="fa fa-check"></i><b>2</b> Copy Number Signature Identification</a>
<ul>
<li class="chapter" data-level="" data-path="cnsig.html"><a href="cnsig.html#introduction-1"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="cnsig.html"><a href="cnsig.html#read-data-1"><i class="fa fa-check"></i>Read Data</a></li>
<li class="chapter" data-level="" data-path="cnsig.html"><a href="cnsig.html#tally-components-1"><i class="fa fa-check"></i>Tally Components</a></li>
<li class="chapter" data-level="" data-path="cnsig.html"><a href="cnsig.html#extract-signatures-1"><i class="fa fa-check"></i>Extract Signatures</a></li>
</ul></li>
<li class="part"><span><b>II Object Visualization</b></span></li>
<li class="chapter" data-level="3" data-path="cnobject.html"><a href="cnobject.html"><i class="fa fa-check"></i><b>3</b> <code>CopyNumber</code> Object</a>
<ul>
<li class="chapter" data-level="" data-path="cnobject.html"><a href="cnobject.html#profile"><i class="fa fa-check"></i>Profile</a></li>
<li class="chapter" data-level="" data-path="cnobject.html"><a href="cnobject.html#distribution"><i class="fa fa-check"></i>Distribution</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sigobject.html"><a href="sigobject.html"><i class="fa fa-check"></i><b>4</b> <code>Signature</code> Object</a>
<ul>
<li class="chapter" data-level="" data-path="sigobject.html"><a href="sigobject.html#signature-profile"><i class="fa fa-check"></i>Signature Profile</a>
<ul>
<li class="chapter" data-level="" data-path="sigobject.html"><a href="sigobject.html#sbs-signature-profile"><i class="fa fa-check"></i>SBS Signature profile</a></li>
<li class="chapter" data-level="" data-path="sigobject.html"><a href="sigobject.html#copy-number-signature-porfile"><i class="fa fa-check"></i>Copy Number Signature Porfile</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="sigobject.html"><a href="sigobject.html#exposure-profile"><i class="fa fa-check"></i>Exposure Profile</a></li>
<li class="chapter" data-level="" data-path="sigobject.html"><a href="sigobject.html#consensus-map"><i class="fa fa-check"></i>Consensus Map</a></li>
</ul></li>
<li class="part"><span><b>III Signature Analysis and Visualization</b></span></li>
<li class="chapter" data-level="5" data-path="association-analysis.html"><a href="association-analysis.html"><i class="fa fa-check"></i><b>5</b> Association Analysis and Visualization</a></li>
<li class="chapter" data-level="6" data-path="group-analysis.html"><a href="group-analysis.html"><i class="fa fa-check"></i><b>6</b> Group Analysis and Visualization</a>
<ul>
<li class="chapter" data-level="" data-path="group-analysis.html"><a href="group-analysis.html#group-generation"><i class="fa fa-check"></i>Group Generation</a></li>
<li class="chapter" data-level="" data-path="group-analysis.html"><a href="group-analysis.html#group-comparison-analysis"><i class="fa fa-check"></i>Group Comparison Analysis</a></li>
<li class="chapter" data-level="" data-path="group-analysis.html"><a href="group-analysis.html#group-visualization"><i class="fa fa-check"></i>Group Visualization</a></li>
</ul></li>
<li class="part"><span><b>IV Others</b></span></li>
<li class="chapter" data-level="7" data-path="datasets.html"><a href="datasets.html"><i class="fa fa-check"></i><b>7</b> Datasets</a>
<ul>
<li class="chapter" data-level="" data-path="datasets.html"><a href="datasets.html#reference-annotation"><i class="fa fa-check"></i>Reference Annotation</a></li>
<li class="chapter" data-level="" data-path="datasets.html"><a href="datasets.html#copy-number-components-setting"><i class="fa fa-check"></i>Copy Number components setting</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">sigminer: Genomic Alteration Signature Analysis in R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">sigminer: Genomic Alteration Signature Analysis in R</h1>
<p class="author"><em>Shixiang Wang, ShanghaiTech University</em></p>
<p class="author"><em>Xue-Song Liu, ShanghaiTech University</em></p>
<p class="date"><em>last revised on 2020-03-19</em></p>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Genomic alterations including single nucleotide substitution (SBS),
copy number alteration (CNA), etc. are the major force for cancer
initialization and development. Due to the specificity of molecular
lesions caused by genomic alterations, we can generate characteristic
alteration spectra, called ‘signature’.</p>
<p>To illustrate the biological significance of genome alteration signatures, we show some
well organized figures here.</p>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-2"></span>
<img src="fig/sbs_signature_overview_nat_review.png" alt="The illustration of SBS signature, fig source: https://www.nature.com/articles/nrg3729"  />
<p class="caption">
Figure 0.1: The illustration of SBS signature, fig source: <a href="https://www.nature.com/articles/nrg3729" class="uri">https://www.nature.com/articles/nrg3729</a>
</p>
</div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-3"></span>
<img src="fig/sbs_signature_overview.png" alt="The illustration of SBS signature (2), fig source: https://www.nature.com/articles/s41467-018-05228-y"  />
<p class="caption">
Figure 0.2: The illustration of SBS signature (2), fig source: <a href="https://www.nature.com/articles/s41467-018-05228-y" class="uri">https://www.nature.com/articles/s41467-018-05228-y</a>
</p>
</div>
<p>SBS signature is previously known as mutational signature. SBS signatures are well studied and related to single-strand changes, typically caused by defective DNA repair.
Common etiologies contain aging, defective DNA mismatch repair, smoking, ultraviolet light exposure and APOBOE.</p>
<p>Currently, all SBS signatures are summarized in COSMIC database, including two versions: <a href="https://cancer.sanger.ac.uk/cosmic/signatures_v2">v2</a> and <a href="https://cancer.sanger.ac.uk/cosmic/signatures/SBS/">v3</a>.</p>
<blockquote>
<p>Recently, <span class="citation">Alexandrov et al. (<a href="#ref-alexandrov2020repertoire" role="doc-biblioref">2020</a>)</span> extends the concept of mutational signature to three types of mutation: SBS, DBS and INDEL. Here, we use genomic alteration signature to refer any kind of mutation type can be used to generate signature.</p>
</blockquote>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-4"></span>
<img src="fig/cn_signature_overview.png" alt="The illustration of copy number signatures, fig source: https://www.nature.com/articles/s41588-018-0212-y"  />
<p class="caption">
Figure 0.3: The illustration of copy number signatures, fig source: <a href="https://www.nature.com/articles/s41588-018-0212-y" class="uri">https://www.nature.com/articles/s41588-018-0212-y</a>
</p>
</div>
<p>Copy number signatures are less studied and many works are still to be done.</p>
<p>To study signatures, we created an easy-to-use and scalable toolkit for genomic alteration signature analysis and visualization in R. We named it <strong>sigminer</strong> (<code>signature</code> + <code>miner</code>).
This tool can help users to extract, analyze and visualize signatures from genomic alteration records,
thus providing new insight into cancer study.</p>
<p>Currently, <strong>sigminer</strong> supports two types of signature:</p>
<ul>
<li>SBS signature in the form of 96 components</li>
<li>Copy number signature by the method either from <span class="citation">Macintyre et al. (<a href="#ref-macintyre2018copy" role="doc-biblioref">2018</a>)</span> or from our group work.</li>
</ul>
<div id="installation" class="section level2">
<h2>Installation</h2>
<p>The <strong>sigminer</strong> package can be installed from <del>CRAN or</del> Github:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="index.html#cb1-1"></a>remotes<span class="op">::</span><span class="kw">install_github</span>(<span class="st">&quot;ShixiangWang/sigminer&quot;</span>)</span></code></pre></div>
</div>
<div id="issues-or-suggestions" class="section level2">
<h2>Issues or Suggestions</h2>
<p>Any issue can be posted on <a href="https://github.com/ShixiangWang/sigminer/issues">GitHub issue</a>, we will reply ASAP.</p>
<p>We have plans to extend other mutational signatures (like INDEL and DBS) to <strong>sigminer</strong> or
extend reading input from other softwares, contribution is welcome.</p>
</div>
<div id="preparation" class="section level2">
<h2>Preparation</h2>
<p>To reproduce the examples shown in this manual, users should load the following packages firstly.
<strong>sigminer</strong> is requred to have version &gt;= 1.0.0.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="index.html#cb2-1"></a><span class="kw">library</span>(sigminer)</span>
<span id="cb2-2"><a href="index.html#cb2-2"></a><span class="kw">library</span>(NMF)</span></code></pre></div>
<p>Current manual uses <strong>sigminer</strong> 1.0.0. More info about <strong>sigminer</strong> can be given as:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="index.html#cb3-1"></a><span class="kw">hello</span>()</span>
<span id="cb3-2"><a href="index.html#cb3-2"></a><span class="co">#&gt; Thanks for using &#39;sigminer&#39; package!</span></span>
<span id="cb3-3"><a href="index.html#cb3-3"></a><span class="co">#&gt; =========================================================================</span></span>
<span id="cb3-4"><a href="index.html#cb3-4"></a><span class="co">#&gt; Version: 1.0.0</span></span>
<span id="cb3-5"><a href="index.html#cb3-5"></a><span class="co">#&gt; </span></span>
<span id="cb3-6"><a href="index.html#cb3-6"></a><span class="co">#&gt; Project home : https://github.com/ShixiangWang/sigminer</span></span>
<span id="cb3-7"><a href="index.html#cb3-7"></a><span class="co">#&gt; Bug report   : https://github.com/ShixiangWang/sigminer/issues</span></span>
<span id="cb3-8"><a href="index.html#cb3-8"></a><span class="co">#&gt; Documentation: https://shixiangwang.github.io/sigminer-doc/</span></span>
<span id="cb3-9"><a href="index.html#cb3-9"></a><span class="co">#&gt; =========================================================================</span></span>
<span id="cb3-10"><a href="index.html#cb3-10"></a><span class="co">#&gt; </span></span></code></pre></div>
</div>
<div id="citation-and-license" class="section level2">
<h2>Citation and LICENSE</h2>
<p>If you use <strong>sigminer</strong> in academic field, please cite our work:</p>
<p><strong><em>Copy number signature analyses in prostate cancer reveal distinct etiologies and clinical outcomes, under submission</em></strong></p>
<hr />
<p>The software is made available for non commercial research purposes only
under the <a href="https://github.com/ShixiangWang/sigminer/blob/master/LICENSE.md">MIT</a>. However, notwithstanding any provision of
the MIT License, the software currently may not be used for commercial
purposes without explicit written permission after contacting Shixiang
Wang <a href="mailto:wangshx@shanghaitech.edu.cn" class="email">wangshx@shanghaitech.edu.cn</a> or Xue-Song Liu
<a href="mailto:liuxs@shanghaitech.edu.cn" class="email">liuxs@shanghaitech.edu.cn</a>.</p>
<p>MIT © 2019-2020 Shixiang Wang, Xue-Song Liu</p>
<p>MIT © 2018 Geoffrey Macintyre</p>
<p>MIT © 2018 Anand Mayakonda</p>
<hr />
<p><strong><a href="https://github.com/XSLiuLab">Cancer Biology Group</a> @ShanghaiTech</strong></p>
<p><strong>Research group led by Xue-Song Liu in ShanghaiTech University</strong></p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-alexandrov2020repertoire">
<p>Alexandrov, Ludmil B, Jaegil Kim, Nicholas J Haradhvala, Mi Ni Huang, Alvin Wei Tian Ng, Yang Wu, Arnoud Boot, et al. 2020. “The Repertoire of Mutational Signatures in Human Cancer.” <em>Nature</em> 578 (7793): 94–101.</p>
</div>
<div id="ref-macintyre2018copy">
<p>Macintyre, Geoff, Teodora E Goranova, Dilrini De Silva, Darren Ennis, Anna M Piskorz, Matthew Eldridge, Daoud Sie, et al. 2018. “Copy Number Signatures and Mutational Processes in Ovarian Carcinoma.” <em>Nature Genetics</em> 50 (9): 1262–70.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="part-signature-identification.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/ShixiangWang/sigminer-doc/edit/master/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["sigminer-doc.pdf", "sigminer-doc.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
