<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 COSMIC Signature Identification | Sigminer: A Scalable Toolkit to Extract, Analyze and Visualize Mutational Signatures</title>
  <meta name="description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 COSMIC Signature Identification | Sigminer: A Scalable Toolkit to Extract, Analyze and Visualize Mutational Signatures" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  <meta name="github-repo" content="ShixiangWang/sigminer-doc" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 COSMIC Signature Identification | Sigminer: A Scalable Toolkit to Extract, Analyze and Visualize Mutational Signatures" />
  
  <meta name="twitter:description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  

<meta name="author" content="Shixiang Wang, ShanghaiTech University" />
<meta name="author" content="Xue-Song Liu, ShanghaiTech University" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="cnsig.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">sigminer: Signature Miner</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#biological-significance-of-mutational-signature"><i class="fa fa-check"></i><b>1.1</b> Biological Significance of Mutational Signature</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#sigminer"><i class="fa fa-check"></i><b>1.2</b> Sigminer</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.3</b> Installation</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#issues-or-suggestions"><i class="fa fa-check"></i><b>1.4</b> Issues or Suggestions</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#preparation"><i class="fa fa-check"></i><b>1.5</b> Preparation</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#overview-of-contents"><i class="fa fa-check"></i><b>1.6</b> Overview of Contents</a></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#citation-and-license"><i class="fa fa-check"></i><b>1.7</b> Citation and LICENSE</a></li>
</ul></li>
<li class="part"><span><b>I Common Workflow</b></span></li>
<li class="chapter" data-level="2" data-path="sbssig.html"><a href="sbssig.html"><i class="fa fa-check"></i><b>2</b> COSMIC Signature Identification</a><ul>
<li class="chapter" data-level="2.1" data-path="sbssig.html"><a href="sbssig.html#read-data"><i class="fa fa-check"></i><b>2.1</b> Read Data</a></li>
<li class="chapter" data-level="2.2" data-path="sbssig.html"><a href="sbssig.html#tally-components"><i class="fa fa-check"></i><b>2.2</b> Tally Components</a><ul>
<li class="chapter" data-level="2.2.1" data-path="sbssig.html"><a href="sbssig.html#the-most-common-96-components"><i class="fa fa-check"></i><b>2.2.1</b> The most common 96 components</a></li>
<li class="chapter" data-level="2.2.2" data-path="sbssig.html"><a href="sbssig.html#other-situations"><i class="fa fa-check"></i><b>2.2.2</b> Other Situations</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="sbssig.html"><a href="sbssig.html#extract-signatures"><i class="fa fa-check"></i><b>2.3</b> Extract Signatures</a><ul>
<li class="chapter" data-level="2.3.1" data-path="sbssig.html"><a href="sbssig.html#method-1-estimate-signature-number"><i class="fa fa-check"></i><b>2.3.1</b> Method 1: Estimate Signature Number</a></li>
<li class="chapter" data-level="2.3.2" data-path="sbssig.html"><a href="sbssig.html#method-1-extract-signatures"><i class="fa fa-check"></i><b>2.3.2</b> Method 1: Extract Signatures</a></li>
<li class="chapter" data-level="2.3.3" data-path="sbssig.html"><a href="sbssig.html#method-2-auto-extract"><i class="fa fa-check"></i><b>2.3.3</b> Method 2: Auto-extract</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="sbssig.html"><a href="sbssig.html#match-signatures"><i class="fa fa-check"></i><b>2.4</b> Match Signatures</a></li>
<li class="chapter" data-level="2.5" data-path="sbssig.html"><a href="sbssig.html#operate-signature"><i class="fa fa-check"></i><b>2.5</b> Operate <code>Signature</code></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cnsig.html"><a href="cnsig.html"><i class="fa fa-check"></i><b>3</b> Copy Number Signature Identification</a><ul>
<li class="chapter" data-level="3.1" data-path="cnsig.html"><a href="cnsig.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="cnsig.html"><a href="cnsig.html#read-data-1"><i class="fa fa-check"></i><b>3.2</b> Read Data</a></li>
<li class="chapter" data-level="3.3" data-path="cnsig.html"><a href="cnsig.html#tally-components-1"><i class="fa fa-check"></i><b>3.3</b> Tally Components</a></li>
<li class="chapter" data-level="3.4" data-path="cnsig.html"><a href="cnsig.html#extract-signatures-1"><i class="fa fa-check"></i><b>3.4</b> Extract Signatures</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sigfit.html"><a href="sigfit.html"><i class="fa fa-check"></i><b>4</b> Signature Fit: Sample Signature Exposure Quantification and Analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="sigfit.html"><a href="sigfit.html#fit-signatures-from-reference-databases"><i class="fa fa-check"></i><b>4.1</b> Fit Signatures from reference databases</a></li>
<li class="chapter" data-level="4.2" data-path="sigfit.html"><a href="sigfit.html#fit-custom-signatures"><i class="fa fa-check"></i><b>4.2</b> Fit Custom Signatures</a></li>
<li class="chapter" data-level="4.3" data-path="sigfit.html"><a href="sigfit.html#performance-comparison"><i class="fa fa-check"></i><b>4.3</b> Performance Comparison</a></li>
<li class="chapter" data-level="4.4" data-path="sigfit.html"><a href="sigfit.html#estimate-exposure-stability-by-bootstrap"><i class="fa fa-check"></i><b>4.4</b> Estimate Exposure Stability by Bootstrap</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="subtyping.html"><a href="subtyping.html"><i class="fa fa-check"></i><b>5</b> Subtype Prediction</a></li>
<li class="chapter" data-level="6" data-path="sigflow.html"><a href="sigflow.html"><i class="fa fa-check"></i><b>6</b> Sigflow Pipeline</a><ul>
<li class="chapter" data-level="6.1" data-path="sigflow.html"><a href="sigflow.html#installation-1"><i class="fa fa-check"></i><b>6.1</b> Installation</a><ul>
<li class="chapter" data-level="6.1.1" data-path="sigflow.html"><a href="sigflow.html#step-by-step-installation"><i class="fa fa-check"></i><b>6.1.1</b> Step-by-step installation</a></li>
<li class="chapter" data-level="6.1.2" data-path="sigflow.html"><a href="sigflow.html#install-by-docker"><i class="fa fa-check"></i><b>6.1.2</b> Install by Docker</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="sigflow.html"><a href="sigflow.html#usage"><i class="fa fa-check"></i><b>6.2</b> Usage</a></li>
<li class="chapter" data-level="6.3" data-path="sigflow.html"><a href="sigflow.html#suggestion-and-bug-report"><i class="fa fa-check"></i><b>6.3</b> Suggestion and bug report</a></li>
<li class="chapter" data-level="6.4" data-path="sigflow.html"><a href="sigflow.html#examples"><i class="fa fa-check"></i><b>6.4</b> Examples</a><ul>
<li class="chapter" data-level="6.4.1" data-path="sigflow.html"><a href="sigflow.html#extract-command"><i class="fa fa-check"></i><b>6.4.1</b> <code>extract</code> command</a></li>
<li class="chapter" data-level="6.4.2" data-path="sigflow.html"><a href="sigflow.html#fit-command"><i class="fa fa-check"></i><b>6.4.2</b> <code>fit</code> command</a></li>
<li class="chapter" data-level="6.4.3" data-path="sigflow.html"><a href="sigflow.html#bt-command"><i class="fa fa-check"></i><b>6.4.3</b> <code>bt</code> command</a></li>
<li class="chapter" data-level="6.4.4" data-path="sigflow.html"><a href="sigflow.html#how-to-use-docker-to-run-sigflow"><i class="fa fa-check"></i><b>6.4.4</b> How to use Docker to run Sigflow</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Target Visualization</b></span></li>
<li class="chapter" data-level="7" data-path="cnobject.html"><a href="cnobject.html"><i class="fa fa-check"></i><b>7</b> CopyNumber Object</a><ul>
<li class="chapter" data-level="7.1" data-path="cnobject.html"><a href="cnobject.html#profile"><i class="fa fa-check"></i><b>7.1</b> Profile</a></li>
<li class="chapter" data-level="7.2" data-path="cnobject.html"><a href="cnobject.html#distribution"><i class="fa fa-check"></i><b>7.2</b> Distribution</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="sigobject.html"><a href="sigobject.html"><i class="fa fa-check"></i><b>8</b> Signature Object</a><ul>
<li class="chapter" data-level="8.1" data-path="sigobject.html"><a href="sigobject.html#signature-profile"><i class="fa fa-check"></i><b>8.1</b> Signature Profile</a><ul>
<li class="chapter" data-level="8.1.1" data-path="sigobject.html"><a href="sigobject.html#sbs-signature-profile"><i class="fa fa-check"></i><b>8.1.1</b> SBS Signature profile</a></li>
<li class="chapter" data-level="8.1.2" data-path="sigobject.html"><a href="sigobject.html#copy-number-signature-porfile"><i class="fa fa-check"></i><b>8.1.2</b> Copy Number Signature Porfile</a></li>
<li class="chapter" data-level="8.1.3" data-path="sigobject.html"><a href="sigobject.html#cosmic-signature-profile"><i class="fa fa-check"></i><b>8.1.3</b> COSMIC Signature Profile</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="sigobject.html"><a href="sigobject.html#exposure-profile"><i class="fa fa-check"></i><b>8.2</b> Exposure Profile</a></li>
<li class="chapter" data-level="8.3" data-path="sigobject.html"><a href="sigobject.html#consensus-map"><i class="fa fa-check"></i><b>8.3</b> Consensus Map</a></li>
<li class="chapter" data-level="8.4" data-path="sigobject.html"><a href="sigobject.html#catalogue-profile"><i class="fa fa-check"></i><b>8.4</b> Catalogue Profile</a></li>
</ul></li>
<li class="part"><span><b>III Universal Analysis</b></span></li>
<li class="chapter" data-level="9" data-path="association-analysis.html"><a href="association-analysis.html"><i class="fa fa-check"></i><b>9</b> Association Analysis and Visualization</a></li>
<li class="chapter" data-level="10" data-path="group-analysis.html"><a href="group-analysis.html"><i class="fa fa-check"></i><b>10</b> Group Analysis and Visualization</a><ul>
<li class="chapter" data-level="10.1" data-path="group-analysis.html"><a href="group-analysis.html#group-generation"><i class="fa fa-check"></i><b>10.1</b> Group Generation</a></li>
<li class="chapter" data-level="10.2" data-path="group-analysis.html"><a href="group-analysis.html#group-comparison-analysis"><i class="fa fa-check"></i><b>10.2</b> Group Comparison Analysis</a></li>
<li class="chapter" data-level="10.3" data-path="group-analysis.html"><a href="group-analysis.html#group-visualization"><i class="fa fa-check"></i><b>10.3</b> Group Visualization</a></li>
</ul></li>
<li class="part"><span><b>IV Useful Utilities</b></span></li>
<li class="chapter" data-level="11" data-path="datasets.html"><a href="datasets.html"><i class="fa fa-check"></i><b>11</b> Datasets</a><ul>
<li class="chapter" data-level="11.1" data-path="datasets.html"><a href="datasets.html#reference-annotation"><i class="fa fa-check"></i><b>11.1</b> Reference Annotation</a></li>
<li class="chapter" data-level="11.2" data-path="datasets.html"><a href="datasets.html#copy-number-components-setting"><i class="fa fa-check"></i><b>11.2</b> Copy Number components setting</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="convert.html"><a href="convert.html"><i class="fa fa-check"></i><b>12</b> SBS Signature Conversion</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Sigminer: A Scalable Toolkit to Extract, Analyze and Visualize Mutational Signatures</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="sbssig" class="section level1">
<h1><span class="header-section-number">Chapter 2</span> COSMIC Signature Identification</h1>
<p>In this chapter, we will introduce how to identify COSMIC signatures from records of variant calling data.
The COSMIC signatures include three type of signatures: SBS, DBS and ID (short for INDEL).</p>
<p>The signature identification procedure has been divided into 3 steps:</p>
<ol style="list-style-type: decimal">
<li>Read mutation data.</li>
<li>Tally components: for SBS, it means classifying SBS records into 96 components (the most common case) and generate sample matrix.</li>
<li>Extract signatures: estimate signature number and identify signatures.</li>
</ol>
<div id="read-data" class="section level2">
<h2><span class="header-section-number">2.1</span> Read Data</h2>
<p>The input data should be in <a href="https://docs.gdc.cancer.gov/Data/File_Formats/MAF_Format/">MAF</a> format, it can be either a MAF file or a <code>data.frame</code>.
There is a famous R/Bioconductor package <a href="https://github.com/PoisonAlien/maftools"><strong>Maftools</strong></a> <span class="citation">(Mayakonda et al. <a href="#ref-mayakonda2018maftools">2018</a>)</span> for analyzing MAF data.
Here, we use the example MAF file stored in <strong>Maftools</strong> as input and extract SBS signatures from the data.</p>
<blockquote>
<p>Make sure <code>library(sigminer)</code> before running the following code.</p>
</blockquote>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">laml.maf &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;tcga_laml.maf.gz&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;maftools&quot;</span>, <span class="dt">mustWork =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">laml &lt;-<span class="st"> </span><span class="kw">read_maf</span>(<span class="dt">maf =</span> laml.maf)</a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co">#&gt; -Reading</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#&gt; -Validating</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co">#&gt; -Silent variants: 475 </span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt; -Summarizing</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt; -Processing clinical data</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="co">#&gt; --Missing clinical data</span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="co">#&gt; -Finished in 0.559s elapsed (0.512s cpu)</span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10">laml</a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt; An object of class  MAF </span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt;                    ID          summary  Mean Median</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="co">#&gt;  1:        NCBI_Build               37    NA     NA</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co">#&gt;  2:            Center genome.wustl.edu    NA     NA</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co">#&gt;  3:           Samples              193    NA     NA</span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="co">#&gt;  4:            nGenes             1241    NA     NA</span></a>
<a class="sourceLine" id="cb6-17" data-line-number="17"><span class="co">#&gt;  5:   Frame_Shift_Del               52 0.269      0</span></a>
<a class="sourceLine" id="cb6-18" data-line-number="18"><span class="co">#&gt;  6:   Frame_Shift_Ins               91 0.472      0</span></a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="co">#&gt;  7:      In_Frame_Del               10 0.052      0</span></a>
<a class="sourceLine" id="cb6-20" data-line-number="20"><span class="co">#&gt;  8:      In_Frame_Ins               42 0.218      0</span></a>
<a class="sourceLine" id="cb6-21" data-line-number="21"><span class="co">#&gt;  9: Missense_Mutation             1342 6.953      7</span></a>
<a class="sourceLine" id="cb6-22" data-line-number="22"><span class="co">#&gt; 10: Nonsense_Mutation              103 0.534      0</span></a>
<a class="sourceLine" id="cb6-23" data-line-number="23"><span class="co">#&gt; 11:       Splice_Site               92 0.477      0</span></a>
<a class="sourceLine" id="cb6-24" data-line-number="24"><span class="co">#&gt; 12:             total             1732 8.974      9</span></a></code></pre></div>
<p>The <code>laml</code> is a <code>MAF</code> object. The <code>MAF</code> class is exported from <strong>Maftools</strong> to <strong>sigminer</strong>. So <code>laml</code> can be directly use functions provided by <strong>Maftools</strong>.</p>
<p>As a <code>MAF</code> object, the mutation records are stored in slot <code>data</code> and <code>maf.silent</code>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">head</span>(laml<span class="op">@</span>data)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="co">#&gt;    Hugo_Symbol Entrez_Gene_Id           Center NCBI_Build Chromosome Start_Position End_Position Strand Variant_Classification</span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co">#&gt; 1:      ABCA10          10349 genome.wustl.edu         37         17       67170917     67170917      +            Splice_Site</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt; 2:       ABCA4             24 genome.wustl.edu         37          1       94490594     94490594      +      Missense_Mutation</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="co">#&gt; 3:      ABCB11           8647 genome.wustl.edu         37          2      169780250    169780250      +      Missense_Mutation</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#&gt; 4:       ABCC3           8714 genome.wustl.edu         37         17       48760974     48760974      +      Missense_Mutation</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co">#&gt; 5:       ABCF1             23 genome.wustl.edu         37          6       30554429     30554429      +      Missense_Mutation</span></a>
<a class="sourceLine" id="cb7-8" data-line-number="8"><span class="co">#&gt; 6:       ABCG4          64137 genome.wustl.edu         37         11      119031351    119031351      +      Missense_Mutation</span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9"><span class="co">#&gt;    Variant_Type Reference_Allele Tumor_Seq_Allele1 Tumor_Seq_Allele2 Tumor_Sample_Barcode Protein_Change i_TumorVAF_WU</span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"><span class="co">#&gt; 1:          SNP                T                 T                 C         TCGA-AB-2988        p.K960R      45.66000</span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="co">#&gt; 2:          SNP                C                 C                 T         TCGA-AB-2869       p.R1517H      38.12000</span></a>
<a class="sourceLine" id="cb7-12" data-line-number="12"><span class="co">#&gt; 3:          SNP                G                 G                 A         TCGA-AB-3009       p.A1283V      46.97218</span></a>
<a class="sourceLine" id="cb7-13" data-line-number="13"><span class="co">#&gt; 4:          SNP                C                 C                 T         TCGA-AB-2887       p.P1271S      56.41000</span></a>
<a class="sourceLine" id="cb7-14" data-line-number="14"><span class="co">#&gt; 5:          SNP                G                 G                 A         TCGA-AB-2920        p.G658S      40.95000</span></a>
<a class="sourceLine" id="cb7-15" data-line-number="15"><span class="co">#&gt; 6:          SNP                A                 A                 G         TCGA-AB-2934        p.Y567C      32.84000</span></a>
<a class="sourceLine" id="cb7-16" data-line-number="16"><span class="co">#&gt;    i_transcript_name</span></a>
<a class="sourceLine" id="cb7-17" data-line-number="17"><span class="co">#&gt; 1:       NM_080282.3</span></a>
<a class="sourceLine" id="cb7-18" data-line-number="18"><span class="co">#&gt; 2:       NM_000350.2</span></a>
<a class="sourceLine" id="cb7-19" data-line-number="19"><span class="co">#&gt; 3:       NM_003742.2</span></a>
<a class="sourceLine" id="cb7-20" data-line-number="20"><span class="co">#&gt; 4:       NM_003786.1</span></a>
<a class="sourceLine" id="cb7-21" data-line-number="21"><span class="co">#&gt; 5:    NM_001025091.1</span></a>
<a class="sourceLine" id="cb7-22" data-line-number="22"><span class="co">#&gt; 6:       NM_022169.1</span></a>
<a class="sourceLine" id="cb7-23" data-line-number="23"><span class="kw">head</span>(laml<span class="op">@</span>maf.silent)</a>
<a class="sourceLine" id="cb7-24" data-line-number="24"><span class="co">#&gt;    Hugo_Symbol Entrez_Gene_Id           Center NCBI_Build Chromosome Start_Position End_Position Strand Variant_Classification</span></a>
<a class="sourceLine" id="cb7-25" data-line-number="25"><span class="co">#&gt; 1:      ABCC11          85320 genome.wustl.edu         37         16       48244997     48244997      +                 Silent</span></a>
<a class="sourceLine" id="cb7-26" data-line-number="26"><span class="co">#&gt; 2:        ACAN            176 genome.wustl.edu         37         15       89401084     89401084      +                 Silent</span></a>
<a class="sourceLine" id="cb7-27" data-line-number="27"><span class="co">#&gt; 3:       ACAT1             38 genome.wustl.edu         37         11      108009744    108009744      +                 Silent</span></a>
<a class="sourceLine" id="cb7-28" data-line-number="28"><span class="co">#&gt; 4:       ACCN2             41 genome.wustl.edu         37         12       50452780     50452780      +                 Silent</span></a>
<a class="sourceLine" id="cb7-29" data-line-number="29"><span class="co">#&gt; 5:       ACTA2             59 genome.wustl.edu         37         10       90695109     90695109      +                 Silent</span></a>
<a class="sourceLine" id="cb7-30" data-line-number="30"><span class="co">#&gt; 6:       ACTL9         284382 genome.wustl.edu         37         19        8808551      8808551      +                 Silent</span></a>
<a class="sourceLine" id="cb7-31" data-line-number="31"><span class="co">#&gt;    Variant_Type Reference_Allele Tumor_Seq_Allele1 Tumor_Seq_Allele2 Tumor_Sample_Barcode Protein_Change i_TumorVAF_WU</span></a>
<a class="sourceLine" id="cb7-32" data-line-number="32"><span class="co">#&gt; 1:          SNP                G                 G                 A         TCGA-AB-2830        p.I490I    34.2700000</span></a>
<a class="sourceLine" id="cb7-33" data-line-number="33"><span class="co">#&gt; 2:          SNP                C                 C                 T         TCGA-AB-2898       p.S1756S    38.3000000</span></a>
<a class="sourceLine" id="cb7-34" data-line-number="34"><span class="co">#&gt; 3:          SNP                T                 T                 G         TCGA-AB-2887        p.T185T    49.0400000</span></a>
<a class="sourceLine" id="cb7-35" data-line-number="35"><span class="co">#&gt; 4:          SNP                C                 C                 G         TCGA-AB-3009         p.L77L    48.1000000</span></a>
<a class="sourceLine" id="cb7-36" data-line-number="36"><span class="co">#&gt; 5:          SNP                C                 C                 T         TCGA-AB-2973        p.P335P     0.2012072</span></a>
<a class="sourceLine" id="cb7-37" data-line-number="37"><span class="co">#&gt; 6:          SNP                G                 G                 A         TCGA-AB-2936        p.F167F    46.1500000</span></a>
<a class="sourceLine" id="cb7-38" data-line-number="38"><span class="co">#&gt;    i_transcript_name</span></a>
<a class="sourceLine" id="cb7-39" data-line-number="39"><span class="co">#&gt; 1:       NM_032583.3</span></a>
<a class="sourceLine" id="cb7-40" data-line-number="40"><span class="co">#&gt; 2:       NM_013227.2</span></a>
<a class="sourceLine" id="cb7-41" data-line-number="41"><span class="co">#&gt; 3:       NM_000019.3</span></a>
<a class="sourceLine" id="cb7-42" data-line-number="42"><span class="co">#&gt; 4:       NM_020039.2</span></a>
<a class="sourceLine" id="cb7-43" data-line-number="43"><span class="co">#&gt; 5:       NM_001613.1</span></a>
<a class="sourceLine" id="cb7-44" data-line-number="44"><span class="co">#&gt; 6:       NM_178525.3</span></a></code></pre></div>
<p>The <code>data</code> slot contains non-silent variants, and the <code>maf.silent</code> slot contains silent variants.
Default uses Variant Classifications with High/Moderate variant consequences as non-silent variants. <a href="http://asia.ensembl.org/Help/Glossary?id=535" class="uri">http://asia.ensembl.org/Help/Glossary?id=535</a>: “Frame_Shift_Del”, “Frame_Shift_Ins”, “Splice_Site”, “Translation_Start_Site”,“Nonsense_Mutation”, “Nonstop_Mutation”, “In_Frame_Del”,“In_Frame_Ins”, “Missense_Mutation” (see <code>?read_maf</code>). If you want to change, please set <code>vc_nonSyn</code> option.</p>
<p>Other slots in <code>MAF</code> object are summary data either by sample or gene/variant type etc.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">slotNames</span>(laml)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="co">#&gt; [1] &quot;data&quot;                           &quot;variants.per.sample&quot;            &quot;variant.type.summary&quot;          </span></a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="co">#&gt; [4] &quot;variant.classification.summary&quot; &quot;gene.summary&quot;                   &quot;summary&quot;                       </span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="co">#&gt; [7] &quot;maf.silent&quot;                     &quot;clinical.data&quot;</span></a></code></pre></div>
</div>
<div id="tally-components" class="section level2">
<h2><span class="header-section-number">2.2</span> Tally Components</h2>
<div id="the-most-common-96-components" class="section level3">
<h3><span class="header-section-number">2.2.1</span> The most common 96 components</h3>
<p>According to 3-nucleotide context (mutated base, 5’ and 3’ adjacent bases) and base complementary pairing principle, we can divide all SBS mutations into 96 mutation types. We call each mutation type as a <em>component</em> here.</p>
<blockquote>
<p>This classification is based the six substitution subtypes: C&gt;A, C&gt;G, C&gt;T, T&gt;A, T&gt;C, and T&gt;G (all substitutions are referred to by the pyrimidine of the mutated Watson—Crick base pair). Further, each of the substitutions is examined by incorporating information on the bases immediately 5’ and 3’ to each mutated base generating 96 possible mutation types (6 types of substitution x 4 types of 5’ base x 4 types of 3’ base).</p>
</blockquote>
<p>We tally components in each sample, and generate a sample-by-component matrix.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">mt_tally &lt;-<span class="st"> </span><span class="kw">sig_tally</span>(</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">  laml,</a>
<a class="sourceLine" id="cb9-3" data-line-number="3">  <span class="dt">ref_genome =</span> <span class="st">&quot;BSgenome.Hsapiens.UCSC.hg19&quot;</span>,</a>
<a class="sourceLine" id="cb9-4" data-line-number="4">  <span class="dt">useSyn =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5">)</a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="co">#&gt; ℹ [2020-08-10 10:31:58]: Started.</span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="co">#&gt; Attaching package: &#39;S4Vectors&#39;</span></a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="co">#&gt; The following object is masked from &#39;package:NMF&#39;:</span></a>
<a class="sourceLine" id="cb9-10" data-line-number="10"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="co">#&gt;     nrun</span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="co">#&gt; The following object is masked from &#39;package:pkgmaker&#39;:</span></a>
<a class="sourceLine" id="cb9-13" data-line-number="13"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="co">#&gt;     new2</span></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="co">#&gt; The following object is masked from &#39;package:base&#39;:</span></a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-17" data-line-number="17"><span class="co">#&gt;     expand.grid</span></a>
<a class="sourceLine" id="cb9-18" data-line-number="18"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-19" data-line-number="19"><span class="co">#&gt; Attaching package: &#39;Biostrings&#39;</span></a>
<a class="sourceLine" id="cb9-20" data-line-number="20"><span class="co">#&gt; The following object is masked from &#39;package:base&#39;:</span></a>
<a class="sourceLine" id="cb9-21" data-line-number="21"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb9-22" data-line-number="22"><span class="co">#&gt;     strsplit</span></a>
<a class="sourceLine" id="cb9-23" data-line-number="23"><span class="co">#&gt; ✓ [2020-08-10 10:32:04]: Reference genome loaded.</span></a>
<a class="sourceLine" id="cb9-24" data-line-number="24"><span class="co">#&gt; ✓ [2020-08-10 10:32:04]: Variants from MAF object queried.</span></a>
<a class="sourceLine" id="cb9-25" data-line-number="25"><span class="co">#&gt; ✓ [2020-08-10 10:32:04]: Chromosome names checked.</span></a>
<a class="sourceLine" id="cb9-26" data-line-number="26"><span class="co">#&gt; ✓ [2020-08-10 10:32:04]: Sex chromosomes properly handled.</span></a>
<a class="sourceLine" id="cb9-27" data-line-number="27"><span class="co">#&gt; ✓ [2020-08-10 10:32:04]: Variant start and end position checked.</span></a>
<a class="sourceLine" id="cb9-28" data-line-number="28"><span class="co">#&gt; ✓ [2020-08-10 10:32:04]: Variant data for matrix generation preprocessed.</span></a>
<a class="sourceLine" id="cb9-29" data-line-number="29"><span class="co">#&gt; ℹ [2020-08-10 10:32:04]: SBS matrix generation - start.</span></a>
<a class="sourceLine" id="cb9-30" data-line-number="30"><span class="co">#&gt; ℹ [2020-08-10 10:32:04]: Extracting 5&#39; and 3&#39; adjacent bases.</span></a>
<a class="sourceLine" id="cb9-31" data-line-number="31"><span class="co">#&gt; ℹ [2020-08-10 10:32:06]: Extracting +/- 20bp around mutated bases for background C&gt;T estimation.</span></a>
<a class="sourceLine" id="cb9-32" data-line-number="32"><span class="co">#&gt; ℹ [2020-08-10 10:32:07]: Estimating APOBEC enrichment scores.</span></a>
<a class="sourceLine" id="cb9-33" data-line-number="33"><span class="co">#&gt; ℹ [2020-08-10 10:32:07]: Performing one-way Fisher&#39;s test for APOBEC enrichment.</span></a>
<a class="sourceLine" id="cb9-34" data-line-number="34"><span class="co">#&gt; ✓ [2020-08-10 10:32:07]: APOBEC related mutations are enriched in 3.297% of samples (APOBEC enrichment score &gt; 2; 6 of 182 samples)</span></a>
<a class="sourceLine" id="cb9-35" data-line-number="35"><span class="co">#&gt; ℹ [2020-08-10 10:32:07]: Creating SBS sample-by-component matrices.</span></a>
<a class="sourceLine" id="cb9-36" data-line-number="36"><span class="co">#&gt; ✓ [2020-08-10 10:32:07]: SBS-6 matrix created.</span></a>
<a class="sourceLine" id="cb9-37" data-line-number="37"><span class="co">#&gt; ✓ [2020-08-10 10:32:07]: SBS-96 matrix created.</span></a>
<a class="sourceLine" id="cb9-38" data-line-number="38"><span class="co">#&gt; ✓ [2020-08-10 10:32:07]: SBS-1536 matrix created.</span></a>
<a class="sourceLine" id="cb9-39" data-line-number="39"><span class="co">#&gt; ℹ [2020-08-10 10:32:07]: Return SBS-96 as major matrix.</span></a>
<a class="sourceLine" id="cb9-40" data-line-number="40"><span class="co">#&gt; ✓ [2020-08-10 10:32:07]: Done.</span></a>
<a class="sourceLine" id="cb9-41" data-line-number="41"><span class="co">#&gt; ℹ [2020-08-10 10:32:07]: 8.778 secs elapsed.</span></a></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">mt_tally<span class="op">$</span>nmf_matrix[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="co">#&gt;              A[T&gt;C]A C[T&gt;C]A G[T&gt;C]A T[T&gt;C]A A[C&gt;T]A</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="co">#&gt; TCGA-AB-2802       0       0       0       0       0</span></a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="co">#&gt; TCGA-AB-2803       0       0       0       0       2</span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="co">#&gt; TCGA-AB-2804       0       0       0       0       2</span></a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="co">#&gt; TCGA-AB-2805       0       0       0       0       0</span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="co">#&gt; TCGA-AB-2806       0       0       0       0       0</span></a></code></pre></div>
<p>We use notion <code>left[ref&gt;mut]right</code> to mark each component, e.g. <code>C[T&gt;G]A</code> means a base T with 5’ adjacent base C and 3’ adjacent base A is mutated to base G.</p>
</div>
<div id="other-situations" class="section level3">
<h3><span class="header-section-number">2.2.2</span> Other Situations</h3>
<p>Above we show the most common SBS classifications, there are other situations supported by <strong>sigminer</strong>, including other classifications for SBS records and other mutation types (DBS and ID). All situations about SBS, DBS and ID signatures are well documented in <a href="https://osf.io/s93d5/wiki/home/">wiki of SigProfilerMatrixGenerator package</a>.</p>
<div id="other-sbs-classifications" class="section level4">
<h4><span class="header-section-number">2.2.2.1</span> Other SBS classifications</h4>
<p>After <code>sig_tally()</code> called, the most used matrix is store in <code>nmf_matrix</code>, and all matrices generated by <strong>sigminer</strong> are stored in <code>all_matrices</code>.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">str</span>(mt_tally<span class="op">$</span>all_matrices, <span class="dt">max.level =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="co">#&gt; List of 3</span></a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="co">#&gt;  $ SBS_6   : int [1:193, 1:6] 1 0 0 2 1 0 1 1 2 2 ...</span></a>
<a class="sourceLine" id="cb11-4" data-line-number="4"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb11-5" data-line-number="5"><span class="co">#&gt;  $ SBS_96  : int [1:193, 1:96] 0 0 0 0 0 0 1 0 1 0 ...</span></a>
<a class="sourceLine" id="cb11-6" data-line-number="6"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb11-7" data-line-number="7"><span class="co">#&gt;  $ SBS_1536: int [1:193, 1:1536] 0 0 0 0 0 0 0 0 0 0 ...</span></a>
<a class="sourceLine" id="cb11-8" data-line-number="8"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a></code></pre></div>
<p>If you add the strand classification, all matrices can be generated by <strong>sigminer</strong> will return.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">mt_tally2 &lt;-<span class="st"> </span><span class="kw">sig_tally</span>(</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">  laml,</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">  <span class="dt">ref_genome =</span> <span class="st">&quot;BSgenome.Hsapiens.UCSC.hg19&quot;</span>,</a>
<a class="sourceLine" id="cb12-4" data-line-number="4">  <span class="dt">useSyn =</span> <span class="ot">TRUE</span>, <span class="dt">add_trans_bias =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb12-5" data-line-number="5">)</a>
<a class="sourceLine" id="cb12-6" data-line-number="6"><span class="co">#&gt; ℹ [2020-08-10 10:32:08]: Started.</span></a>
<a class="sourceLine" id="cb12-7" data-line-number="7"><span class="co">#&gt; ✓ [2020-08-10 10:32:08]: Reference genome loaded.</span></a>
<a class="sourceLine" id="cb12-8" data-line-number="8"><span class="co">#&gt; ✓ [2020-08-10 10:32:08]: Variants from MAF object queried.</span></a>
<a class="sourceLine" id="cb12-9" data-line-number="9"><span class="co">#&gt; ✓ [2020-08-10 10:32:08]: Chromosome names checked.</span></a>
<a class="sourceLine" id="cb12-10" data-line-number="10"><span class="co">#&gt; ✓ [2020-08-10 10:32:08]: Sex chromosomes properly handled.</span></a>
<a class="sourceLine" id="cb12-11" data-line-number="11"><span class="co">#&gt; ✓ [2020-08-10 10:32:08]: Variant start and end position checked.</span></a>
<a class="sourceLine" id="cb12-12" data-line-number="12"><span class="co">#&gt; ✓ [2020-08-10 10:32:08]: Variant data for matrix generation preprocessed.</span></a>
<a class="sourceLine" id="cb12-13" data-line-number="13"><span class="co">#&gt; ℹ [2020-08-10 10:32:08]: SBS matrix generation - start.</span></a>
<a class="sourceLine" id="cb12-14" data-line-number="14"><span class="co">#&gt; ℹ [2020-08-10 10:32:08]: Extracting 5&#39; and 3&#39; adjacent bases.</span></a>
<a class="sourceLine" id="cb12-15" data-line-number="15"><span class="co">#&gt; ℹ [2020-08-10 10:32:08]: Extracting +/- 20bp around mutated bases for background C&gt;T estimation.</span></a>
<a class="sourceLine" id="cb12-16" data-line-number="16"><span class="co">#&gt; ℹ [2020-08-10 10:32:09]: Estimating APOBEC enrichment scores.</span></a>
<a class="sourceLine" id="cb12-17" data-line-number="17"><span class="co">#&gt; ℹ [2020-08-10 10:32:09]: Performing one-way Fisher&#39;s test for APOBEC enrichment.</span></a>
<a class="sourceLine" id="cb12-18" data-line-number="18"><span class="co">#&gt; ✓ [2020-08-10 10:32:09]: APOBEC related mutations are enriched in 3.297% of samples (APOBEC enrichment score &gt; 2; 6 of 182 samples)</span></a>
<a class="sourceLine" id="cb12-19" data-line-number="19"><span class="co">#&gt; ℹ [2020-08-10 10:32:09]: Creating SBS sample-by-component matrices.</span></a>
<a class="sourceLine" id="cb12-20" data-line-number="20"><span class="co">#&gt; ✓ [2020-08-10 10:32:09]: SBS-6 matrix created.</span></a>
<a class="sourceLine" id="cb12-21" data-line-number="21"><span class="co">#&gt; ✓ [2020-08-10 10:32:09]: SBS-96 matrix created.</span></a>
<a class="sourceLine" id="cb12-22" data-line-number="22"><span class="co">#&gt; ✓ [2020-08-10 10:32:09]: SBS-1536 matrix created.</span></a>
<a class="sourceLine" id="cb12-23" data-line-number="23"><span class="co">#&gt; ✓ [2020-08-10 10:32:09]: SBS-24 (6x4) matrix created.</span></a>
<a class="sourceLine" id="cb12-24" data-line-number="24"><span class="co">#&gt; ✓ [2020-08-10 10:32:09]: SBS-384 (96x4) matrix created.</span></a>
<a class="sourceLine" id="cb12-25" data-line-number="25"><span class="co">#&gt; ✓ [2020-08-10 10:32:09]: SBS-6144 (1536x4) matrix created.</span></a>
<a class="sourceLine" id="cb12-26" data-line-number="26"><span class="co">#&gt; ℹ [2020-08-10 10:32:09]: Return SBS-384 as major matrix.</span></a>
<a class="sourceLine" id="cb12-27" data-line-number="27"><span class="co">#&gt; ✓ [2020-08-10 10:32:09]: Done.</span></a>
<a class="sourceLine" id="cb12-28" data-line-number="28"><span class="co">#&gt; ℹ [2020-08-10 10:32:09]: 1.447 secs elapsed.</span></a>
<a class="sourceLine" id="cb12-29" data-line-number="29"></a>
<a class="sourceLine" id="cb12-30" data-line-number="30"><span class="kw">str</span>(mt_tally2<span class="op">$</span>all_matrices, <span class="dt">max.level =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb12-31" data-line-number="31"><span class="co">#&gt; List of 6</span></a>
<a class="sourceLine" id="cb12-32" data-line-number="32"><span class="co">#&gt;  $ SBS_6   : int [1:193, 1:6] 1 0 0 2 1 0 1 1 2 2 ...</span></a>
<a class="sourceLine" id="cb12-33" data-line-number="33"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb12-34" data-line-number="34"><span class="co">#&gt;  $ SBS_24  : int [1:193, 1:24] 1 0 0 1 1 0 0 1 1 1 ...</span></a>
<a class="sourceLine" id="cb12-35" data-line-number="35"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb12-36" data-line-number="36"><span class="co">#&gt;  $ SBS_96  : int [1:193, 1:96] 0 0 0 0 0 0 1 0 1 0 ...</span></a>
<a class="sourceLine" id="cb12-37" data-line-number="37"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb12-38" data-line-number="38"><span class="co">#&gt;  $ SBS_384 : int [1:193, 1:384] 0 0 0 0 0 0 0 0 0 0 ...</span></a>
<a class="sourceLine" id="cb12-39" data-line-number="39"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb12-40" data-line-number="40"><span class="co">#&gt;  $ SBS_1536: int [1:193, 1:1536] 0 0 0 0 0 0 0 0 0 0 ...</span></a>
<a class="sourceLine" id="cb12-41" data-line-number="41"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb12-42" data-line-number="42"><span class="co">#&gt;  $ SBS_6144: int [1:193, 1:6144] 0 0 0 0 0 0 0 0 0 0 ...</span></a>
<a class="sourceLine" id="cb12-43" data-line-number="43"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a></code></pre></div>
</div>
<div id="dbs-and-id-components" class="section level4">
<h4><span class="header-section-number">2.2.2.2</span> DBS and ID components</h4>
<p>If you want to generate DBS or ID matrices, just modify the <code>mode</code> option.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">mt_tally_DBS &lt;-<span class="st"> </span><span class="kw">sig_tally</span>(</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">  laml,</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">  <span class="dt">ref_genome =</span> <span class="st">&quot;BSgenome.Hsapiens.UCSC.hg19&quot;</span>,</a>
<a class="sourceLine" id="cb13-4" data-line-number="4">  <span class="dt">useSyn =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb13-5" data-line-number="5">  <span class="dt">mode =</span> <span class="st">&quot;DBS&quot;</span>,</a>
<a class="sourceLine" id="cb13-6" data-line-number="6">  <span class="dt">add_trans_bias =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb13-7" data-line-number="7">)</a>
<a class="sourceLine" id="cb13-8" data-line-number="8"><span class="co">#&gt; ℹ [2020-08-10 10:32:09]: Started.</span></a>
<a class="sourceLine" id="cb13-9" data-line-number="9"><span class="co">#&gt; ✓ [2020-08-10 10:32:09]: Reference genome loaded.</span></a>
<a class="sourceLine" id="cb13-10" data-line-number="10"><span class="co">#&gt; ✓ [2020-08-10 10:32:09]: Variants from MAF object queried.</span></a>
<a class="sourceLine" id="cb13-11" data-line-number="11"><span class="co">#&gt; ✓ [2020-08-10 10:32:09]: Chromosome names checked.</span></a>
<a class="sourceLine" id="cb13-12" data-line-number="12"><span class="co">#&gt; ✓ [2020-08-10 10:32:09]: Sex chromosomes properly handled.</span></a>
<a class="sourceLine" id="cb13-13" data-line-number="13"><span class="co">#&gt; ✓ [2020-08-10 10:32:09]: Variant start and end position checked.</span></a>
<a class="sourceLine" id="cb13-14" data-line-number="14"><span class="co">#&gt; ✓ [2020-08-10 10:32:09]: Variant data for matrix generation preprocessed.</span></a>
<a class="sourceLine" id="cb13-15" data-line-number="15"><span class="co">#&gt; ℹ [2020-08-10 10:32:09]: DBS matrix generation - start.</span></a>
<a class="sourceLine" id="cb13-16" data-line-number="16"><span class="co">#&gt; ℹ [2020-08-10 10:32:09]: Searching DBS records...</span></a>
<a class="sourceLine" id="cb13-17" data-line-number="17"><span class="co">#&gt; ✓ [2020-08-10 10:32:11]: Done.</span></a>
<a class="sourceLine" id="cb13-18" data-line-number="18"><span class="co">#&gt; x [2020-08-10 10:32:11]: Zero DBSs to analyze!</span></a>
<a class="sourceLine" id="cb13-19" data-line-number="19"><span class="co">#&gt; Error in send_stop(&quot;Zero DBSs to analyze!&quot;):</span></a>
<a class="sourceLine" id="cb13-20" data-line-number="20"><span class="co">#&gt; ℹ [2020-08-10 10:32:11]: 1.516 secs elapsed.</span></a>
<a class="sourceLine" id="cb13-21" data-line-number="21"></a>
<a class="sourceLine" id="cb13-22" data-line-number="22"><span class="kw">str</span>(mt_tally_DBS<span class="op">$</span>all_matrices, <span class="dt">max.level =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb13-23" data-line-number="23"><span class="co">#&gt; Error in str(mt_tally_DBS$all_matrices, max.level = 1): object &#39;mt_tally_DBS&#39; not found</span></a></code></pre></div>
<p>Program will stop if no records to analyze.</p>
<p>Let’s see ID records.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">mt_tally_ID &lt;-<span class="st"> </span><span class="kw">sig_tally</span>(</a>
<a class="sourceLine" id="cb14-2" data-line-number="2">  laml,</a>
<a class="sourceLine" id="cb14-3" data-line-number="3">  <span class="dt">ref_genome =</span> <span class="st">&quot;BSgenome.Hsapiens.UCSC.hg19&quot;</span>,</a>
<a class="sourceLine" id="cb14-4" data-line-number="4">  <span class="dt">useSyn =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb14-5" data-line-number="5">  <span class="dt">mode =</span> <span class="st">&quot;ID&quot;</span>,</a>
<a class="sourceLine" id="cb14-6" data-line-number="6">  <span class="dt">add_trans_bias =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb14-7" data-line-number="7">)</a>
<a class="sourceLine" id="cb14-8" data-line-number="8"><span class="co">#&gt; ℹ [2020-08-10 10:32:11]: Started.</span></a>
<a class="sourceLine" id="cb14-9" data-line-number="9"><span class="co">#&gt; ✓ [2020-08-10 10:32:11]: Reference genome loaded.</span></a>
<a class="sourceLine" id="cb14-10" data-line-number="10"><span class="co">#&gt; ✓ [2020-08-10 10:32:11]: Variants from MAF object queried.</span></a>
<a class="sourceLine" id="cb14-11" data-line-number="11"><span class="co">#&gt; ✓ [2020-08-10 10:32:11]: Chromosome names checked.</span></a>
<a class="sourceLine" id="cb14-12" data-line-number="12"><span class="co">#&gt; ✓ [2020-08-10 10:32:11]: Sex chromosomes properly handled.</span></a>
<a class="sourceLine" id="cb14-13" data-line-number="13"><span class="co">#&gt; ✓ [2020-08-10 10:32:11]: Variant start and end position checked.</span></a>
<a class="sourceLine" id="cb14-14" data-line-number="14"><span class="co">#&gt; ✓ [2020-08-10 10:32:11]: Variant data for matrix generation preprocessed.</span></a>
<a class="sourceLine" id="cb14-15" data-line-number="15"><span class="co">#&gt; ℹ [2020-08-10 10:32:11]: INDEL matrix generation - start.</span></a>
<a class="sourceLine" id="cb14-16" data-line-number="16"><span class="co">#&gt; ✓ [2020-08-10 10:32:12]: Reference sequences queried from genome.</span></a>
<a class="sourceLine" id="cb14-17" data-line-number="17"><span class="co">#&gt; ✓ [2020-08-10 10:32:12]: INDEL length extracted.</span></a>
<a class="sourceLine" id="cb14-18" data-line-number="18"><span class="co">#&gt; ✓ [2020-08-10 10:32:12]: Adjacent copies counted.</span></a>
<a class="sourceLine" id="cb14-19" data-line-number="19"><span class="co">#&gt; ✓ [2020-08-10 10:32:12]: Microhomology size calculated.</span></a>
<a class="sourceLine" id="cb14-20" data-line-number="20"><span class="co">#&gt; ✓ [2020-08-10 10:32:12]: INDEL records classified into different components (types).</span></a>
<a class="sourceLine" id="cb14-21" data-line-number="21"><span class="co">#&gt; ✓ [2020-08-10 10:32:12]: ID-28 matrix created.</span></a>
<a class="sourceLine" id="cb14-22" data-line-number="22"><span class="co">#&gt; ✓ [2020-08-10 10:32:12]: ID-83 matrix created.</span></a>
<a class="sourceLine" id="cb14-23" data-line-number="23"><span class="co">#&gt; ✓ [2020-08-10 10:32:12]: ID-415 matrix created.</span></a>
<a class="sourceLine" id="cb14-24" data-line-number="24"><span class="co">#&gt; ℹ [2020-08-10 10:32:12]: Return ID-415 as major matrix.</span></a>
<a class="sourceLine" id="cb14-25" data-line-number="25"><span class="co">#&gt; ✓ [2020-08-10 10:32:12]: Done.</span></a>
<a class="sourceLine" id="cb14-26" data-line-number="26"><span class="co">#&gt; ℹ [2020-08-10 10:32:12]: 1.053 secs elapsed.</span></a>
<a class="sourceLine" id="cb14-27" data-line-number="27"></a>
<a class="sourceLine" id="cb14-28" data-line-number="28"><span class="kw">str</span>(mt_tally_ID<span class="op">$</span>all_matrices, <span class="dt">max.level =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb14-29" data-line-number="29"><span class="co">#&gt; List of 3</span></a>
<a class="sourceLine" id="cb14-30" data-line-number="30"><span class="co">#&gt;  $ ID_28 : int [1:193, 1:28] 0 0 0 0 0 0 0 0 0 0 ...</span></a>
<a class="sourceLine" id="cb14-31" data-line-number="31"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb14-32" data-line-number="32"><span class="co">#&gt;  $ ID_83 : int [1:193, 1:83] 0 0 0 0 0 0 0 0 0 0 ...</span></a>
<a class="sourceLine" id="cb14-33" data-line-number="33"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb14-34" data-line-number="34"><span class="co">#&gt;  $ ID_415:&#39;data.frame&#39;:  193 obs. of  415 variables:</span></a></code></pre></div>
</div>
<div id="take-togother" class="section level4">
<h4><span class="header-section-number">2.2.2.3</span> Take togother</h4>
<p>If you want to get all matrices for SBS, DBS and ID at the same time, you don’t need to write a for loop or type three times to do this.
Just set <code>mode='ALL'</code>, <strong>sigminer</strong> will do it for you.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">mt_tally_all &lt;-<span class="st"> </span><span class="kw">sig_tally</span>(</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">  laml,</a>
<a class="sourceLine" id="cb15-3" data-line-number="3">  <span class="dt">ref_genome =</span> <span class="st">&quot;BSgenome.Hsapiens.UCSC.hg19&quot;</span>,</a>
<a class="sourceLine" id="cb15-4" data-line-number="4">  <span class="dt">useSyn =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb15-5" data-line-number="5">  <span class="dt">mode =</span> <span class="st">&quot;ALL&quot;</span>,</a>
<a class="sourceLine" id="cb15-6" data-line-number="6">  <span class="dt">add_trans_bias =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb15-7" data-line-number="7">)</a>
<a class="sourceLine" id="cb15-8" data-line-number="8"><span class="co">#&gt; ℹ [2020-08-10 10:32:13]: Started.</span></a>
<a class="sourceLine" id="cb15-9" data-line-number="9"><span class="co">#&gt; ✓ [2020-08-10 10:32:13]: Reference genome loaded.</span></a>
<a class="sourceLine" id="cb15-10" data-line-number="10"><span class="co">#&gt; ✓ [2020-08-10 10:32:13]: Variants from MAF object queried.</span></a>
<a class="sourceLine" id="cb15-11" data-line-number="11"><span class="co">#&gt; ✓ [2020-08-10 10:32:13]: Chromosome names checked.</span></a>
<a class="sourceLine" id="cb15-12" data-line-number="12"><span class="co">#&gt; ✓ [2020-08-10 10:32:13]: Sex chromosomes properly handled.</span></a>
<a class="sourceLine" id="cb15-13" data-line-number="13"><span class="co">#&gt; ✓ [2020-08-10 10:32:13]: Variant start and end position checked.</span></a>
<a class="sourceLine" id="cb15-14" data-line-number="14"><span class="co">#&gt; ✓ [2020-08-10 10:32:13]: Variant data for matrix generation preprocessed.</span></a>
<a class="sourceLine" id="cb15-15" data-line-number="15"><span class="co">#&gt; ℹ [2020-08-10 10:32:13]: All types of matrices generation - start.</span></a>
<a class="sourceLine" id="cb15-16" data-line-number="16"><span class="co">#&gt; ℹ [2020-08-10 10:32:13]: SBS matrix generation - start.</span></a>
<a class="sourceLine" id="cb15-17" data-line-number="17"><span class="co">#&gt; ℹ [2020-08-10 10:32:13]: Extracting 5&#39; and 3&#39; adjacent bases.</span></a>
<a class="sourceLine" id="cb15-18" data-line-number="18"><span class="co">#&gt; ℹ [2020-08-10 10:32:13]: Extracting +/- 20bp around mutated bases for background C&gt;T estimation.</span></a>
<a class="sourceLine" id="cb15-19" data-line-number="19"><span class="co">#&gt; ℹ [2020-08-10 10:32:14]: Estimating APOBEC enrichment scores.</span></a>
<a class="sourceLine" id="cb15-20" data-line-number="20"><span class="co">#&gt; ℹ [2020-08-10 10:32:14]: Performing one-way Fisher&#39;s test for APOBEC enrichment.</span></a>
<a class="sourceLine" id="cb15-21" data-line-number="21"><span class="co">#&gt; ✓ [2020-08-10 10:32:14]: APOBEC related mutations are enriched in 3.297% of samples (APOBEC enrichment score &gt; 2; 6 of 182 samples)</span></a>
<a class="sourceLine" id="cb15-22" data-line-number="22"><span class="co">#&gt; ℹ [2020-08-10 10:32:14]: Creating SBS sample-by-component matrices.</span></a>
<a class="sourceLine" id="cb15-23" data-line-number="23"><span class="co">#&gt; ✓ [2020-08-10 10:32:14]: SBS-6 matrix created.</span></a>
<a class="sourceLine" id="cb15-24" data-line-number="24"><span class="co">#&gt; ✓ [2020-08-10 10:32:14]: SBS-96 matrix created.</span></a>
<a class="sourceLine" id="cb15-25" data-line-number="25"><span class="co">#&gt; ✓ [2020-08-10 10:32:14]: SBS-1536 matrix created.</span></a>
<a class="sourceLine" id="cb15-26" data-line-number="26"><span class="co">#&gt; ✓ [2020-08-10 10:32:14]: SBS-24 (6x4) matrix created.</span></a>
<a class="sourceLine" id="cb15-27" data-line-number="27"><span class="co">#&gt; ✓ [2020-08-10 10:32:14]: SBS-384 (96x4) matrix created.</span></a>
<a class="sourceLine" id="cb15-28" data-line-number="28"><span class="co">#&gt; ✓ [2020-08-10 10:32:14]: SBS-6144 (1536x4) matrix created.</span></a>
<a class="sourceLine" id="cb15-29" data-line-number="29"><span class="co">#&gt; ℹ [2020-08-10 10:32:14]: Return SBS-384 as major matrix.</span></a>
<a class="sourceLine" id="cb15-30" data-line-number="30"><span class="co">#&gt; ℹ [2020-08-10 10:32:14]: DBS matrix generation - start.</span></a>
<a class="sourceLine" id="cb15-31" data-line-number="31"><span class="co">#&gt; ℹ [2020-08-10 10:32:14]: Searching DBS records...</span></a>
<a class="sourceLine" id="cb15-32" data-line-number="32"><span class="co">#&gt; ✓ [2020-08-10 10:32:15]: Done.</span></a>
<a class="sourceLine" id="cb15-33" data-line-number="33"><span class="co">#&gt; x [2020-08-10 10:32:15]: Zero DBSs to analyze!</span></a>
<a class="sourceLine" id="cb15-34" data-line-number="34"><span class="co">#&gt; ℹ [2020-08-10 10:32:15]: INDEL matrix generation - start.</span></a>
<a class="sourceLine" id="cb15-35" data-line-number="35"><span class="co">#&gt; ✓ [2020-08-10 10:32:16]: Reference sequences queried from genome.</span></a>
<a class="sourceLine" id="cb15-36" data-line-number="36"><span class="co">#&gt; ✓ [2020-08-10 10:32:16]: INDEL length extracted.</span></a>
<a class="sourceLine" id="cb15-37" data-line-number="37"><span class="co">#&gt; ✓ [2020-08-10 10:32:16]: Adjacent copies counted.</span></a>
<a class="sourceLine" id="cb15-38" data-line-number="38"><span class="co">#&gt; ✓ [2020-08-10 10:32:16]: Microhomology size calculated.</span></a>
<a class="sourceLine" id="cb15-39" data-line-number="39"><span class="co">#&gt; ✓ [2020-08-10 10:32:16]: INDEL records classified into different components (types).</span></a>
<a class="sourceLine" id="cb15-40" data-line-number="40"><span class="co">#&gt; ✓ [2020-08-10 10:32:16]: ID-28 matrix created.</span></a>
<a class="sourceLine" id="cb15-41" data-line-number="41"><span class="co">#&gt; ✓ [2020-08-10 10:32:16]: ID-83 matrix created.</span></a>
<a class="sourceLine" id="cb15-42" data-line-number="42"><span class="co">#&gt; ✓ [2020-08-10 10:32:16]: ID-415 matrix created.</span></a>
<a class="sourceLine" id="cb15-43" data-line-number="43"><span class="co">#&gt; ℹ [2020-08-10 10:32:16]: Return ID-415 as major matrix.</span></a>
<a class="sourceLine" id="cb15-44" data-line-number="44"><span class="co">#&gt; ℹ [2020-08-10 10:32:16]: All types of matrices generation (APOBEC scores included) - end.</span></a>
<a class="sourceLine" id="cb15-45" data-line-number="45"><span class="co">#&gt; ✓ [2020-08-10 10:32:16]: Done.</span></a>
<a class="sourceLine" id="cb15-46" data-line-number="46"><span class="co">#&gt; ℹ [2020-08-10 10:32:16]: 3.239 secs elapsed.</span></a>
<a class="sourceLine" id="cb15-47" data-line-number="47"></a>
<a class="sourceLine" id="cb15-48" data-line-number="48"><span class="kw">str</span>(mt_tally_all, <span class="dt">max.level =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb15-49" data-line-number="49"><span class="co">#&gt; List of 10</span></a>
<a class="sourceLine" id="cb15-50" data-line-number="50"><span class="co">#&gt;  $ SBS_6        : int [1:193, 1:6] 1 0 0 2 1 0 1 1 2 2 ...</span></a>
<a class="sourceLine" id="cb15-51" data-line-number="51"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb15-52" data-line-number="52"><span class="co">#&gt;  $ SBS_24       : int [1:193, 1:24] 1 0 0 1 1 0 0 1 1 1 ...</span></a>
<a class="sourceLine" id="cb15-53" data-line-number="53"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb15-54" data-line-number="54"><span class="co">#&gt;  $ SBS_96       : int [1:193, 1:96] 0 0 0 0 0 0 1 0 1 0 ...</span></a>
<a class="sourceLine" id="cb15-55" data-line-number="55"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb15-56" data-line-number="56"><span class="co">#&gt;  $ SBS_384      : int [1:193, 1:384] 0 0 0 0 0 0 0 0 0 0 ...</span></a>
<a class="sourceLine" id="cb15-57" data-line-number="57"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb15-58" data-line-number="58"><span class="co">#&gt;  $ SBS_1536     : int [1:193, 1:1536] 0 0 0 0 0 0 0 0 0 0 ...</span></a>
<a class="sourceLine" id="cb15-59" data-line-number="59"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb15-60" data-line-number="60"><span class="co">#&gt;  $ SBS_6144     : int [1:193, 1:6144] 0 0 0 0 0 0 0 0 0 0 ...</span></a>
<a class="sourceLine" id="cb15-61" data-line-number="61"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb15-62" data-line-number="62"><span class="co">#&gt;  $ ID_28        : int [1:193, 1:28] 0 0 0 0 0 0 0 0 0 0 ...</span></a>
<a class="sourceLine" id="cb15-63" data-line-number="63"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb15-64" data-line-number="64"><span class="co">#&gt;  $ ID_83        : int [1:193, 1:83] 0 0 0 0 0 0 0 0 0 0 ...</span></a>
<a class="sourceLine" id="cb15-65" data-line-number="65"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb15-66" data-line-number="66"><span class="co">#&gt;  $ ID_415       :&#39;data.frame&#39;:   193 obs. of  415 variables:</span></a>
<a class="sourceLine" id="cb15-67" data-line-number="67"><span class="co">#&gt;  $ APOBEC_scores:Classes &#39;data.table&#39; and &#39;data.frame&#39;:  182 obs. of  44 variables:</span></a>
<a class="sourceLine" id="cb15-68" data-line-number="68"><span class="co">#&gt;   ..- attr(*, &quot;.internal.selfref&quot;)=&lt;externalptr&gt; </span></a>
<a class="sourceLine" id="cb15-69" data-line-number="69"><span class="co">#&gt;   ..- attr(*, &quot;index&quot;)= int(0) </span></a>
<a class="sourceLine" id="cb15-70" data-line-number="70"><span class="co">#&gt;   .. ..- attr(*, &quot;__APOBEC_Enriched&quot;)= int [1:182] 106 147 5 6 8 9 10 11 12 13 ...</span></a></code></pre></div>
<p>Please note, in this case, just a list containing matrices will return.</p>
</div>
</div>
</div>
<div id="extract-signatures" class="section level2">
<h2><span class="header-section-number">2.3</span> Extract Signatures</h2>
<p>There are two methods can be used to implement signature extraction.</p>
<ol style="list-style-type: decimal">
<li>Firstly estimate signature number and then extract signatures. This method is powered by <a href="https://github.com/renozao/NMF"><strong>NMF</strong></a> package <span class="citation">(Gaujoux and Seoighe <a href="#ref-gaujoux2010flexible">2010</a>)</span>.</li>
<li>Auto-extract signatures by automatic relevance determination technique in nonnegative matrix factorization <span class="citation">(Tan and Févotte <a href="#ref-tan2012automatic">2012</a>)</span>, the code is implemented by <strong>SignatureAnalyzer</strong> <span class="citation">(Kim et al. <a href="#ref-kim2016somatic">2016</a>)</span> and exported to <strong>sigminer</strong>.</li>
</ol>
<div id="method-1-estimate-signature-number" class="section level3">
<h3><span class="header-section-number">2.3.1</span> Method 1: Estimate Signature Number</h3>
<p>To estimate signature number, we firstly run NMF for multiple times and calculate some measures from the results, then select the best signature number based on the measure vs. signature number plot.</p>
<p>This is done by <code>sig_estimate()</code> function. Let’s try signature number 2-5. For simplicity, we just run NMF twice for each signature number. We use 4 cores to speed up the computation.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">mt_est &lt;-<span class="st"> </span><span class="kw">sig_estimate</span>(mt_tally<span class="op">$</span>nmf_matrix,</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">  <span class="dt">range =</span> <span class="dv">2</span><span class="op">:</span><span class="dv">5</span>,</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">  <span class="dt">nrun =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb16-4" data-line-number="4">  <span class="dt">use_random =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb16-5" data-line-number="5">  <span class="dt">cores =</span> <span class="dv">4</span>,</a>
<a class="sourceLine" id="cb16-6" data-line-number="6">  <span class="dt">pConstant =</span> <span class="fl">1e-13</span>,</a>
<a class="sourceLine" id="cb16-7" data-line-number="7">  <span class="dt">verbose =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb16-8" data-line-number="8">)</a></code></pre></div>
<blockquote>
<p><code>pConstant</code> option is set to avoid errors raised by <strong>NMF</strong> package.</p>
</blockquote>
<p>We can show signature number survey for different measures by <code>show_sig_number_survey2()</code>.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="co">## You can also select the measures to show</span></a>
<a class="sourceLine" id="cb17-2" data-line-number="2"><span class="co">## by &#39;what&#39; option</span></a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="kw">show_sig_number_survey2</span>(mt_est<span class="op">$</span>survey, mt_est<span class="op">$</span>survey.random)</a></code></pre></div>
<p><img src="sigminer-doc_files/figure-html/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>For the details of all the measures above, please read <span class="citation">Gaujoux and Seoighe (<a href="#ref-gaujoux2010flexible">2010</a>)</span> and <a href="https://cran.r-project.org/web/packages/NMF/vignettes/">vignette</a> of R package <strong>NMF</strong>.</p>
<p>Typically, measure <strong>cophenetic</strong> is used for determing the signature number. We can easily generate an elbow plot with <code>function</code>show_sig_number_survey()`.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw">show_sig_number_survey</span>(mt_est<span class="op">$</span>survey, <span class="dt">right_y =</span> <span class="ot">NULL</span>)</a></code></pre></div>
<p><img src="sigminer-doc_files/figure-html/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Cophenetic indicates the robustness of consensus matrix clustering. In this situation, 2 or 3 is good.</p>
<blockquote>
<p>The most common approach is to use the cophenetic correlation coefficient. Brunet et al. suggested choosing the smallest value of r for which this coefficient starts decreasing. <span class="citation">(Gaujoux and Seoighe <a href="#ref-gaujoux2010flexible">2010</a>)</span></p>
</blockquote>
<p>However, there are no gold standard to determine the signature number. Sometimes, you should consider multiple measures. Remember, the most important thing is that you should have a good biological explanation for each signature.
The best solution in study may not be the best solution in math.</p>
</div>
<div id="method-1-extract-signatures" class="section level3">
<h3><span class="header-section-number">2.3.2</span> Method 1: Extract Signatures</h3>
<p>After selecting a proper signature, you can now extract signatures. In general, use 30~50 NMF runs will get a robust result.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">mt_sig &lt;-<span class="st"> </span><span class="kw">sig_extract</span>(mt_tally<span class="op">$</span>nmf_matrix,</a>
<a class="sourceLine" id="cb19-2" data-line-number="2">  <span class="dt">n_sig =</span> <span class="dv">3</span>,</a>
<a class="sourceLine" id="cb19-3" data-line-number="3">  <span class="dt">nrun =</span> <span class="dv">10</span>,</a>
<a class="sourceLine" id="cb19-4" data-line-number="4">  <span class="dt">cores =</span> <span class="dv">4</span>,</a>
<a class="sourceLine" id="cb19-5" data-line-number="5">  <span class="dt">pConstant =</span> <span class="fl">1e-13</span></a>
<a class="sourceLine" id="cb19-6" data-line-number="6">)</a></code></pre></div>
</div>
<div id="method-2-auto-extract" class="section level3">
<h3><span class="header-section-number">2.3.3</span> Method 2: Auto-extract</h3>
<p>If you have no idea to select a optimal signature number from technologies above, you can try auto-extract feature here. It uses a bayesian variant of NMF algorithm. You need to set a maximum signature number (default is <code>25</code>) and run times to get the result.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">mt_sig2 &lt;-<span class="st"> </span><span class="kw">sig_auto_extract</span>(mt_tally<span class="op">$</span>nmf_matrix, <span class="dt">K0 =</span> <span class="dv">10</span>, <span class="dt">nrun =</span> <span class="dv">10</span>)</a></code></pre></div>
<p>At default, the code will use ‘robust’ strategy to return the result (see <code>strategy</code> option). It means that if you run 10 times and 6 of them return <code>4</code> signatures, then the optimal result with <code>4</code> signatures will be returned.</p>
<p>The info of each run can be given as:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">mt_sig2<span class="op">$</span>Raw<span class="op">$</span>summary_run</a>
<a class="sourceLine" id="cb21-2" data-line-number="2"><span class="co">#&gt; # A tibble: 10 x 4</span></a>
<a class="sourceLine" id="cb21-3" data-line-number="3"><span class="co">#&gt;      Run     K posterior file                                                                        </span></a>
<a class="sourceLine" id="cb21-4" data-line-number="4"><span class="co">#&gt;    &lt;int&gt; &lt;int&gt;     &lt;dbl&gt; &lt;chr&gt;                                                                       </span></a>
<a class="sourceLine" id="cb21-5" data-line-number="5"><span class="co">#&gt;  1     8     2     1822. /var/folders/mx/rfkl27z90c96wbmn3_kjk8c80000gn/T//RtmpTrCKrZ/BayesNMF.8.rds </span></a>
<a class="sourceLine" id="cb21-6" data-line-number="6"><span class="co">#&gt;  2     6     2     1815. /var/folders/mx/rfkl27z90c96wbmn3_kjk8c80000gn/T//RtmpTrCKrZ/BayesNMF.6.rds </span></a>
<a class="sourceLine" id="cb21-7" data-line-number="7"><span class="co">#&gt;  3     9     3     1793. /var/folders/mx/rfkl27z90c96wbmn3_kjk8c80000gn/T//RtmpTrCKrZ/BayesNMF.9.rds </span></a>
<a class="sourceLine" id="cb21-8" data-line-number="8"><span class="co">#&gt;  4     2     2     1789. /var/folders/mx/rfkl27z90c96wbmn3_kjk8c80000gn/T//RtmpTrCKrZ/BayesNMF.2.rds </span></a>
<a class="sourceLine" id="cb21-9" data-line-number="9"><span class="co">#&gt;  5     5     2     1785. /var/folders/mx/rfkl27z90c96wbmn3_kjk8c80000gn/T//RtmpTrCKrZ/BayesNMF.5.rds </span></a>
<a class="sourceLine" id="cb21-10" data-line-number="10"><span class="co">#&gt;  6     3     3     1766. /var/folders/mx/rfkl27z90c96wbmn3_kjk8c80000gn/T//RtmpTrCKrZ/BayesNMF.3.rds </span></a>
<a class="sourceLine" id="cb21-11" data-line-number="11"><span class="co">#&gt;  7    10     2     1764. /var/folders/mx/rfkl27z90c96wbmn3_kjk8c80000gn/T//RtmpTrCKrZ/BayesNMF.10.rds</span></a>
<a class="sourceLine" id="cb21-12" data-line-number="12"><span class="co">#&gt;  8     1     4     1740. /var/folders/mx/rfkl27z90c96wbmn3_kjk8c80000gn/T//RtmpTrCKrZ/BayesNMF.1.rds </span></a>
<a class="sourceLine" id="cb21-13" data-line-number="13"><span class="co">#&gt;  9     4     3     1718. /var/folders/mx/rfkl27z90c96wbmn3_kjk8c80000gn/T//RtmpTrCKrZ/BayesNMF.4.rds </span></a>
<a class="sourceLine" id="cb21-14" data-line-number="14"><span class="co">#&gt; 10     7     3     1706. /var/folders/mx/rfkl27z90c96wbmn3_kjk8c80000gn/T//RtmpTrCKrZ/BayesNMF.7.rds</span></a></code></pre></div>
</div>
</div>
<div id="match-signatures" class="section level2">
<h2><span class="header-section-number">2.4</span> Match Signatures</h2>
<p>After extracting signatures, we need to know their etiologies. This can be done by comparing the identified signatures and reference signatures from COSMIC database.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">sim &lt;-<span class="st"> </span><span class="kw">get_sig_similarity</span>(mt_sig)</a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="co">#&gt; -Comparing against COSMIC signatures</span></a>
<a class="sourceLine" id="cb22-3" data-line-number="3"><span class="co">#&gt; ------------------------------------</span></a>
<a class="sourceLine" id="cb22-4" data-line-number="4"><span class="co">#&gt; --Found Sig1 most similar to COSMIC_1</span></a>
<a class="sourceLine" id="cb22-5" data-line-number="5"><span class="co">#&gt;    Aetiology: spontaneous deamination of 5-methylcytosine [similarity: 0.871]</span></a>
<a class="sourceLine" id="cb22-6" data-line-number="6"><span class="co">#&gt; --Found Sig2 most similar to COSMIC_1</span></a>
<a class="sourceLine" id="cb22-7" data-line-number="7"><span class="co">#&gt;    Aetiology: spontaneous deamination of 5-methylcytosine [similarity: 0.841]</span></a>
<a class="sourceLine" id="cb22-8" data-line-number="8"><span class="co">#&gt; --Found Sig3 most similar to COSMIC_1</span></a>
<a class="sourceLine" id="cb22-9" data-line-number="9"><span class="co">#&gt;    Aetiology: spontaneous deamination of 5-methylcytosine [similarity: 0.776]</span></a>
<a class="sourceLine" id="cb22-10" data-line-number="10"><span class="co">#&gt; ------------------------------------</span></a>
<a class="sourceLine" id="cb22-11" data-line-number="11"><span class="co">#&gt; Return result invisiblely.</span></a></code></pre></div>
<p>The result object <code>sim</code> is a list.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1"><span class="kw">str</span>(sim)</a>
<a class="sourceLine" id="cb23-2" data-line-number="2"><span class="co">#&gt; List of 4</span></a>
<a class="sourceLine" id="cb23-3" data-line-number="3"><span class="co">#&gt;  $ similarity  : num [1:3, 1:30] 0.871 0.841 0.776 0.221 0.049 0.211 0.277 0.345 0.286 0.217 ...</span></a>
<a class="sourceLine" id="cb23-4" data-line-number="4"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb23-5" data-line-number="5"><span class="co">#&gt;   .. ..$ : chr [1:3] &quot;Sig1&quot; &quot;Sig2&quot; &quot;Sig3&quot;</span></a>
<a class="sourceLine" id="cb23-6" data-line-number="6"><span class="co">#&gt;   .. ..$ : chr [1:30] &quot;COSMIC_1&quot; &quot;COSMIC_2&quot; &quot;COSMIC_3&quot; &quot;COSMIC_4&quot; ...</span></a>
<a class="sourceLine" id="cb23-7" data-line-number="7"><span class="co">#&gt;  $ aetiology_db:List of 1</span></a>
<a class="sourceLine" id="cb23-8" data-line-number="8"><span class="co">#&gt;   ..$ : chr [1:30] &quot;spontaneous deamination of 5-methylcytosine&quot; &quot;APOBEC Cytidine Deaminase (C&gt;T)&quot; &quot;defects in DNA-DSB repair by HR&quot; &quot;exposure to tobacco (smoking) mutagens&quot; ...</span></a>
<a class="sourceLine" id="cb23-9" data-line-number="9"><span class="co">#&gt;  $ best_match  :List of 3</span></a>
<a class="sourceLine" id="cb23-10" data-line-number="10"><span class="co">#&gt;   ..$ Sig1:List of 2</span></a>
<a class="sourceLine" id="cb23-11" data-line-number="11"><span class="co">#&gt;   .. ..$ aetiology : chr &quot;spontaneous deamination of 5-methylcytosine&quot;</span></a>
<a class="sourceLine" id="cb23-12" data-line-number="12"><span class="co">#&gt;   .. ..$ best_match: chr &quot;Best match: COSMIC_1 [similarity: 0.871]&quot;</span></a>
<a class="sourceLine" id="cb23-13" data-line-number="13"><span class="co">#&gt;   ..$ Sig2:List of 2</span></a>
<a class="sourceLine" id="cb23-14" data-line-number="14"><span class="co">#&gt;   .. ..$ aetiology : chr &quot;spontaneous deamination of 5-methylcytosine&quot;</span></a>
<a class="sourceLine" id="cb23-15" data-line-number="15"><span class="co">#&gt;   .. ..$ best_match: chr &quot;Best match: COSMIC_1 [similarity: 0.841]&quot;</span></a>
<a class="sourceLine" id="cb23-16" data-line-number="16"><span class="co">#&gt;   ..$ Sig3:List of 2</span></a>
<a class="sourceLine" id="cb23-17" data-line-number="17"><span class="co">#&gt;   .. ..$ aetiology : chr &quot;spontaneous deamination of 5-methylcytosine&quot;</span></a>
<a class="sourceLine" id="cb23-18" data-line-number="18"><span class="co">#&gt;   .. ..$ best_match: chr &quot;Best match: COSMIC_1 [similarity: 0.776]&quot;</span></a>
<a class="sourceLine" id="cb23-19" data-line-number="19"><span class="co">#&gt;  $ rss         : num [1:3, 1:30] 0.016 0.0191 0.0277 0.2517 0.2901 ...</span></a>
<a class="sourceLine" id="cb23-20" data-line-number="20"><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></a>
<a class="sourceLine" id="cb23-21" data-line-number="21"><span class="co">#&gt;   .. ..$ : chr [1:3] &quot;Sig1&quot; &quot;Sig2&quot; &quot;Sig3&quot;</span></a>
<a class="sourceLine" id="cb23-22" data-line-number="22"><span class="co">#&gt;   .. ..$ : chr [1:30] &quot;COSMIC_1&quot; &quot;COSMIC_2&quot; &quot;COSMIC_3&quot; &quot;COSMIC_4&quot; ...</span></a>
<a class="sourceLine" id="cb23-23" data-line-number="23"><span class="co">#&gt;  - attr(*, &quot;class&quot;)= chr [1:2] &quot;similarity&quot; &quot;list&quot;</span></a></code></pre></div>
<p>From the result we can see that only one signature is clear. If you find unknow signatures in your study, you should explore the etiologies by other analyses and even experiments.</p>
<p>The similarity matrix can be plotted.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(sim<span class="op">$</span>similarity)</a></code></pre></div>
<p><img src="sigminer-doc_files/figure-html/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>You can also try the COSMIC signature database V3 with:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1">sim_v3 &lt;-<span class="st"> </span><span class="kw">get_sig_similarity</span>(mt_sig, <span class="dt">sig_db =</span> <span class="st">&quot;SBS&quot;</span>)</a>
<a class="sourceLine" id="cb25-2" data-line-number="2"><span class="co">#&gt; -Comparing against COSMIC signatures</span></a>
<a class="sourceLine" id="cb25-3" data-line-number="3"><span class="co">#&gt; ------------------------------------</span></a>
<a class="sourceLine" id="cb25-4" data-line-number="4"><span class="co">#&gt; --Found Sig1 most similar to SBS1</span></a>
<a class="sourceLine" id="cb25-5" data-line-number="5"><span class="co">#&gt;    Aetiology: spontaneous or enzymatic deamination of 5-methylcytosine [similarity: 0.844]</span></a>
<a class="sourceLine" id="cb25-6" data-line-number="6"><span class="co">#&gt; --Found Sig2 most similar to SBS1</span></a>
<a class="sourceLine" id="cb25-7" data-line-number="7"><span class="co">#&gt;    Aetiology: spontaneous or enzymatic deamination of 5-methylcytosine [similarity: 0.782]</span></a>
<a class="sourceLine" id="cb25-8" data-line-number="8"><span class="co">#&gt; --Found Sig3 most similar to SBS6</span></a>
<a class="sourceLine" id="cb25-9" data-line-number="9"><span class="co">#&gt;    Aetiology: defective DNA mismatch repair [similarity: 0.694]</span></a>
<a class="sourceLine" id="cb25-10" data-line-number="10"><span class="co">#&gt; ------------------------------------</span></a>
<a class="sourceLine" id="cb25-11" data-line-number="11"><span class="co">#&gt; Return result invisiblely.</span></a></code></pre></div>
</div>
<div id="operate-signature" class="section level2">
<h2><span class="header-section-number">2.5</span> Operate <code>Signature</code></h2>
<p>The result of <code>sig_extract()</code> or <code>sig_auto_extract()</code> is a <code>list</code> with <code>Signature</code> class. You can use <code>$</code> or use operation function to obtain the data stored in it.</p>
<p>To get the signature matrix:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="co"># Or mt_sig$Signature</span></a>
<a class="sourceLine" id="cb26-2" data-line-number="2"><span class="kw">sig_signature</span>(mt_sig)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, ]</a>
<a class="sourceLine" id="cb26-3" data-line-number="3"><span class="co">#&gt;                 Sig1         Sig2         Sig3</span></a>
<a class="sourceLine" id="cb26-4" data-line-number="4"><span class="co">#&gt; A[T&gt;C]A 4.347168e-02 2.419528e-14 5.443553e-19</span></a>
<a class="sourceLine" id="cb26-5" data-line-number="5"><span class="co">#&gt; C[T&gt;C]A 5.587475e-19 1.208329e-14 1.030792e-02</span></a>
<a class="sourceLine" id="cb26-6" data-line-number="6"><span class="co">#&gt; G[T&gt;C]A 4.699434e-14 3.233990e-03 5.602686e-03</span></a>
<a class="sourceLine" id="cb26-7" data-line-number="7"><span class="co">#&gt; T[T&gt;C]A 3.498551e-15 4.448091e-15 2.945119e-03</span></a>
<a class="sourceLine" id="cb26-8" data-line-number="8"><span class="co">#&gt; A[C&gt;T]A 2.003145e-14 7.513116e-02 1.388966e-14</span></a></code></pre></div>
<p>To get signature exposure matrix:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1"><span class="co"># Or mt_sig$Exposure</span></a>
<a class="sourceLine" id="cb27-2" data-line-number="2"><span class="kw">sig_exposure</span>(mt_sig)[, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</a>
<a class="sourceLine" id="cb27-3" data-line-number="3"><span class="co">#&gt;      TCGA-AB-2802 TCGA-AB-2803 TCGA-AB-2804 TCGA-AB-2805 TCGA-AB-2806</span></a>
<a class="sourceLine" id="cb27-4" data-line-number="4"><span class="co">#&gt; Sig1     5.967386     4.220468 6.470106e-12 7.828328e+00     4.979286</span></a>
<a class="sourceLine" id="cb27-5" data-line-number="5"><span class="co">#&gt; Sig2     1.490521     7.651933 7.000000e+00 6.171672e+00     5.190129</span></a>
<a class="sourceLine" id="cb27-6" data-line-number="6"><span class="co">#&gt; Sig3     2.542092     2.127600 2.134479e-12 4.303842e-13     4.830585</span></a></code></pre></div>
<p><code>get_sig_exposure()</code> may be more useful, it can be used to return a <code>data.frame</code> and set an exposure threshold.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="kw">get_sig_exposure</span>(mt_sig)</a>
<a class="sourceLine" id="cb28-2" data-line-number="2"><span class="co">#&gt;            sample         Sig1         Sig2         Sig3</span></a>
<a class="sourceLine" id="cb28-3" data-line-number="3"><span class="co">#&gt;   1: TCGA-AB-2802 5.967386e+00 1.490521e+00 2.542092e+00</span></a>
<a class="sourceLine" id="cb28-4" data-line-number="4"><span class="co">#&gt;   2: TCGA-AB-2803 4.220468e+00 7.651933e+00 2.127600e+00</span></a>
<a class="sourceLine" id="cb28-5" data-line-number="5"><span class="co">#&gt;   3: TCGA-AB-2804 6.470106e-12 7.000000e+00 2.134479e-12</span></a>
<a class="sourceLine" id="cb28-6" data-line-number="6"><span class="co">#&gt;   4: TCGA-AB-2805 7.828328e+00 6.171672e+00 4.303842e-13</span></a>
<a class="sourceLine" id="cb28-7" data-line-number="7"><span class="co">#&gt;   5: TCGA-AB-2806 4.979286e+00 5.190129e+00 4.830585e+00</span></a>
<a class="sourceLine" id="cb28-8" data-line-number="8"><span class="co">#&gt;  ---                                                    </span></a>
<a class="sourceLine" id="cb28-9" data-line-number="9"><span class="co">#&gt; 189: TCGA-AB-3007 7.000000e+00 2.234347e-12 6.506815e-12</span></a>
<a class="sourceLine" id="cb28-10" data-line-number="10"><span class="co">#&gt; 190: TCGA-AB-3008 4.000000e+00 1.393582e-11 1.371823e-11</span></a>
<a class="sourceLine" id="cb28-11" data-line-number="11"><span class="co">#&gt; 191: TCGA-AB-3009 9.664115e+00 9.658138e+00 1.667775e+01</span></a>
<a class="sourceLine" id="cb28-12" data-line-number="12"><span class="co">#&gt; 192: TCGA-AB-3011 6.000000e+00 4.721216e-12 2.653254e-12</span></a>
<a class="sourceLine" id="cb28-13" data-line-number="13"><span class="co">#&gt; 193: TCGA-AB-3012 4.657863e+00 1.242847e-11 2.342137e+00</span></a></code></pre></div>
<p>For plotting signature profile and exposure profile, please go to chapter <a href="sigobject.html#sigobject">8</a>.</p>
<p>For optimizing signature exposure, please use <code>sig_fit()</code>.</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-gaujoux2010flexible">
<p>Gaujoux, Renaud, and Cathal Seoighe. 2010. “A Flexible R Package for Nonnegative Matrix Factorization.” <em>BMC Bioinformatics</em> 11 (1). Springer: 367.</p>
</div>
<div id="ref-kim2016somatic">
<p>Kim, Jaegil, Kent W Mouw, Paz Polak, Lior Z Braunstein, Atanas Kamburov, Grace Tiao, David J Kwiatkowski, et al. 2016. “Somatic Ercc2 Mutations Are Associated with a Distinct Genomic Signature in Urothelial Tumors.” <em>Nature Genetics</em> 48 (6). Nature Publishing Group: 600.</p>
</div>
<div id="ref-mayakonda2018maftools">
<p>Mayakonda, Anand, De-Chen Lin, Yassen Assenov, Christoph Plass, and H Phillip Koeffler. 2018. “Maftools: Efficient and Comprehensive Analysis of Somatic Variants in Cancer.” <em>Genome Research</em> 28 (11). Cold Spring Harbor Lab: 1747–56.</p>
</div>
<div id="ref-tan2012automatic">
<p>Tan, Vincent YF, and Cédric Févotte. 2012. “Automatic Relevance Determination in Nonnegative Matrix Factorization with the/Spl Beta/-Divergence.” <em>IEEE Transactions on Pattern Analysis and Machine Intelligence</em> 35 (7). IEEE: 1592–1605.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="cnsig.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/ShixiangWang/sigminer-doc/edit/master/01-COSMIC-signature-identification.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["sigminer-doc.pdf", "sigminer-doc.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
