<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 SBS Signature Identification | sigminer: Genomic Alteration Signature Analysis in R</title>
  <meta name="description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 SBS Signature Identification | sigminer: Genomic Alteration Signature Analysis in R" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  <meta name="github-repo" content="ShixiangWang/sigminer-doc" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 SBS Signature Identification | sigminer: Genomic Alteration Signature Analysis in R" />
  
  <meta name="twitter:description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  

<meta name="author" content="Shixiang Wang, ShanghaiTech University" />
<meta name="author" content="Xue-Song Liu, ShanghaiTech University" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="part-common-workflow.html"/>
<link rel="next" href="cnsig.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">sigminer: Signature Miner</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#biological-significance-of-signature"><i class="fa fa-check"></i><b>1.1</b> Biological Significance of Signature</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#sigminer"><i class="fa fa-check"></i><b>1.2</b> Sigminer</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.3</b> Installation</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#issues-or-suggestions"><i class="fa fa-check"></i><b>1.4</b> Issues or Suggestions</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#preparation"><i class="fa fa-check"></i><b>1.5</b> Preparation</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#overview-of-contents"><i class="fa fa-check"></i><b>1.6</b> Overview of Contents</a></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#citation-and-license"><i class="fa fa-check"></i><b>1.7</b> Citation and LICENSE</a></li>
</ul></li>
<li class="part"><span><b>I Common Workflow</b></span></li>
<li class="chapter" data-level="2" data-path="sbssig.html"><a href="sbssig.html"><i class="fa fa-check"></i><b>2</b> SBS Signature Identification</a>
<ul>
<li class="chapter" data-level="2.1" data-path="sbssig.html"><a href="sbssig.html#read-data"><i class="fa fa-check"></i><b>2.1</b> Read Data</a></li>
<li class="chapter" data-level="2.2" data-path="sbssig.html"><a href="sbssig.html#tally-components"><i class="fa fa-check"></i><b>2.2</b> Tally Components</a></li>
<li class="chapter" data-level="2.3" data-path="sbssig.html"><a href="sbssig.html#extract-signatures"><i class="fa fa-check"></i><b>2.3</b> Extract Signatures</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="sbssig.html"><a href="sbssig.html#method-1-estimate-signature-number"><i class="fa fa-check"></i><b>2.3.1</b> Method 1: Estimate Signature Number</a></li>
<li class="chapter" data-level="2.3.2" data-path="sbssig.html"><a href="sbssig.html#method-1-extract-signatures"><i class="fa fa-check"></i><b>2.3.2</b> Method 1: Extract Signatures</a></li>
<li class="chapter" data-level="2.3.3" data-path="sbssig.html"><a href="sbssig.html#method-2-auto-extract"><i class="fa fa-check"></i><b>2.3.3</b> Method 2: Auto-extract</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="sbssig.html"><a href="sbssig.html#match-signatures"><i class="fa fa-check"></i><b>2.4</b> Match Signatures</a></li>
<li class="chapter" data-level="2.5" data-path="sbssig.html"><a href="sbssig.html#operate-signature"><i class="fa fa-check"></i><b>2.5</b> Operate <code>Signature</code></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cnsig.html"><a href="cnsig.html"><i class="fa fa-check"></i><b>3</b> Copy Number Signature Identification</a>
<ul>
<li class="chapter" data-level="3.1" data-path="cnsig.html"><a href="cnsig.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="cnsig.html"><a href="cnsig.html#read-data-1"><i class="fa fa-check"></i><b>3.2</b> Read Data</a></li>
<li class="chapter" data-level="3.3" data-path="cnsig.html"><a href="cnsig.html#tally-components-1"><i class="fa fa-check"></i><b>3.3</b> Tally Components</a></li>
<li class="chapter" data-level="3.4" data-path="cnsig.html"><a href="cnsig.html#extract-signatures-1"><i class="fa fa-check"></i><b>3.4</b> Extract Signatures</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sigfit.html"><a href="sigfit.html"><i class="fa fa-check"></i><b>4</b> Signature Fit: Sample Exposure Quantification</a>
<ul>
<li class="chapter" data-level="4.1" data-path="sigfit.html"><a href="sigfit.html#fit-signatures-from-reference-databases"><i class="fa fa-check"></i><b>4.1</b> Fit Signatures from reference databases</a></li>
<li class="chapter" data-level="4.2" data-path="sigfit.html"><a href="sigfit.html#fit-custom-signatures"><i class="fa fa-check"></i><b>4.2</b> Fit Custom Signatures</a></li>
<li class="chapter" data-level="4.3" data-path="sigfit.html"><a href="sigfit.html#performance-comparison"><i class="fa fa-check"></i><b>4.3</b> Performance Comparison</a></li>
</ul></li>
<li class="part"><span><b>II Target Visualization</b></span></li>
<li class="chapter" data-level="5" data-path="cnobject.html"><a href="cnobject.html"><i class="fa fa-check"></i><b>5</b> CopyNumber Object</a>
<ul>
<li class="chapter" data-level="5.1" data-path="cnobject.html"><a href="cnobject.html#profile"><i class="fa fa-check"></i><b>5.1</b> Profile</a></li>
<li class="chapter" data-level="5.2" data-path="cnobject.html"><a href="cnobject.html#distribution"><i class="fa fa-check"></i><b>5.2</b> Distribution</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="sigobject.html"><a href="sigobject.html"><i class="fa fa-check"></i><b>6</b> Signature Object</a>
<ul>
<li class="chapter" data-level="6.1" data-path="sigobject.html"><a href="sigobject.html#signature-profile"><i class="fa fa-check"></i><b>6.1</b> Signature Profile</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="sigobject.html"><a href="sigobject.html#sbs-signature-profile"><i class="fa fa-check"></i><b>6.1.1</b> SBS Signature profile</a></li>
<li class="chapter" data-level="6.1.2" data-path="sigobject.html"><a href="sigobject.html#copy-number-signature-porfile"><i class="fa fa-check"></i><b>6.1.2</b> Copy Number Signature Porfile</a></li>
<li class="chapter" data-level="6.1.3" data-path="sigobject.html"><a href="sigobject.html#cosmic-signature-profile"><i class="fa fa-check"></i><b>6.1.3</b> COSMIC Signature Profile</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="sigobject.html"><a href="sigobject.html#exposure-profile"><i class="fa fa-check"></i><b>6.2</b> Exposure Profile</a></li>
<li class="chapter" data-level="6.3" data-path="sigobject.html"><a href="sigobject.html#consensus-map"><i class="fa fa-check"></i><b>6.3</b> Consensus Map</a></li>
<li class="chapter" data-level="6.4" data-path="sigobject.html"><a href="sigobject.html#catalogue-profile"><i class="fa fa-check"></i><b>6.4</b> Catalogue Profile</a></li>
</ul></li>
<li class="part"><span><b>III Universal Analysis</b></span></li>
<li class="chapter" data-level="7" data-path="association-analysis.html"><a href="association-analysis.html"><i class="fa fa-check"></i><b>7</b> Association Analysis and Visualization</a></li>
<li class="chapter" data-level="8" data-path="group-analysis.html"><a href="group-analysis.html"><i class="fa fa-check"></i><b>8</b> Group Analysis and Visualization</a>
<ul>
<li class="chapter" data-level="8.1" data-path="group-analysis.html"><a href="group-analysis.html#group-generation"><i class="fa fa-check"></i><b>8.1</b> Group Generation</a></li>
<li class="chapter" data-level="8.2" data-path="group-analysis.html"><a href="group-analysis.html#group-comparison-analysis"><i class="fa fa-check"></i><b>8.2</b> Group Comparison Analysis</a></li>
<li class="chapter" data-level="8.3" data-path="group-analysis.html"><a href="group-analysis.html#group-visualization"><i class="fa fa-check"></i><b>8.3</b> Group Visualization</a></li>
</ul></li>
<li class="part"><span><b>IV Useful Utilities</b></span></li>
<li class="chapter" data-level="9" data-path="datasets.html"><a href="datasets.html"><i class="fa fa-check"></i><b>9</b> Datasets</a>
<ul>
<li class="chapter" data-level="9.1" data-path="datasets.html"><a href="datasets.html#reference-annotation"><i class="fa fa-check"></i><b>9.1</b> Reference Annotation</a></li>
<li class="chapter" data-level="9.2" data-path="datasets.html"><a href="datasets.html#copy-number-components-setting"><i class="fa fa-check"></i><b>9.2</b> Copy Number components setting</a></li>
<li class="chapter" data-level="9.3" data-path="datasets.html"><a href="datasets.html#sbs-signature-conversion"><i class="fa fa-check"></i><b>9.3</b> SBS Signature Conversion</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">sigminer: Genomic Alteration Signature Analysis in R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="sbssig" class="section level1">
<h1><span class="header-section-number">Chapter 2</span> SBS Signature Identification</h1>
<p>In this chapter, we will introduce how to identify SBS signatures from SBS records of variant calling data.</p>
<p>The signature identification procedure has been divided into 3 steps:</p>
<ol style="list-style-type: decimal">
<li>Read mutation data</li>
<li>Tally components: classify SBS into 96 components and generate sample matrix</li>
<li>Extract signatures: estimate signature number and identify signatures</li>
</ol>
<div id="read-data" class="section level2">
<h2><span class="header-section-number">2.1</span> Read Data</h2>
<p>The input data should be in <a href="https://docs.gdc.cancer.gov/Data/File_Formats/MAF_Format/">MAF</a> format, it can be either a MAF file or a <code>data.frame</code>.
There is a famous R/Bioconductor package <a href="https://github.com/PoisonAlien/maftools"><strong>Maftools</strong></a> <span class="citation">(Mayakonda et al. <a href="#ref-mayakonda2018maftools" role="doc-biblioref">2018</a>)</span> for analyzing MAF data.
Here, we use the example MAF file stored in <strong>Maftools</strong> as input and extract SBS signatures from the data.</p>
<blockquote>
<p>Make sure <code>library(sigminer)</code> before running the following code.</p>
</blockquote>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="sbssig.html#cb5-1"></a>laml.maf &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;tcga_laml.maf.gz&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;maftools&quot;</span>, <span class="dt">mustWork =</span> <span class="ot">TRUE</span>)</span>
<span id="cb5-2"><a href="sbssig.html#cb5-2"></a>laml &lt;-<span class="st"> </span><span class="kw">read_maf</span>(<span class="dt">maf =</span> laml.maf)</span>
<span id="cb5-3"><a href="sbssig.html#cb5-3"></a><span class="co">#&gt; -Reading</span></span>
<span id="cb5-4"><a href="sbssig.html#cb5-4"></a><span class="co">#&gt; -Validating</span></span>
<span id="cb5-5"><a href="sbssig.html#cb5-5"></a><span class="co">#&gt; -Silent variants: 475 </span></span>
<span id="cb5-6"><a href="sbssig.html#cb5-6"></a><span class="co">#&gt; -Summarizing</span></span>
<span id="cb5-7"><a href="sbssig.html#cb5-7"></a><span class="co">#&gt; -Processing clinical data</span></span>
<span id="cb5-8"><a href="sbssig.html#cb5-8"></a><span class="co">#&gt; --Missing clinical data</span></span>
<span id="cb5-9"><a href="sbssig.html#cb5-9"></a><span class="co">#&gt; -Finished in 0.470s elapsed (0.440s cpu)</span></span>
<span id="cb5-10"><a href="sbssig.html#cb5-10"></a>laml</span>
<span id="cb5-11"><a href="sbssig.html#cb5-11"></a><span class="co">#&gt; An object of class  MAF </span></span>
<span id="cb5-12"><a href="sbssig.html#cb5-12"></a><span class="co">#&gt;                    ID          summary  Mean Median</span></span>
<span id="cb5-13"><a href="sbssig.html#cb5-13"></a><span class="co">#&gt;  1:        NCBI_Build               37    NA     NA</span></span>
<span id="cb5-14"><a href="sbssig.html#cb5-14"></a><span class="co">#&gt;  2:            Center genome.wustl.edu    NA     NA</span></span>
<span id="cb5-15"><a href="sbssig.html#cb5-15"></a><span class="co">#&gt;  3:           Samples              193    NA     NA</span></span>
<span id="cb5-16"><a href="sbssig.html#cb5-16"></a><span class="co">#&gt;  4:            nGenes             1241    NA     NA</span></span>
<span id="cb5-17"><a href="sbssig.html#cb5-17"></a><span class="co">#&gt;  5:   Frame_Shift_Del               52 0.271      0</span></span>
<span id="cb5-18"><a href="sbssig.html#cb5-18"></a><span class="co">#&gt;  6:   Frame_Shift_Ins               91 0.474      0</span></span>
<span id="cb5-19"><a href="sbssig.html#cb5-19"></a><span class="co">#&gt;  7:      In_Frame_Del               10 0.052      0</span></span>
<span id="cb5-20"><a href="sbssig.html#cb5-20"></a><span class="co">#&gt;  8:      In_Frame_Ins               42 0.219      0</span></span>
<span id="cb5-21"><a href="sbssig.html#cb5-21"></a><span class="co">#&gt;  9: Missense_Mutation             1342 6.990      7</span></span>
<span id="cb5-22"><a href="sbssig.html#cb5-22"></a><span class="co">#&gt; 10: Nonsense_Mutation              103 0.536      0</span></span>
<span id="cb5-23"><a href="sbssig.html#cb5-23"></a><span class="co">#&gt; 11:       Splice_Site               92 0.479      0</span></span>
<span id="cb5-24"><a href="sbssig.html#cb5-24"></a><span class="co">#&gt; 12:             total             1732 9.021      9</span></span></code></pre></div>
<p>The <code>laml</code> is a <code>MAF</code> object. The <code>MAF</code> class is exported from <strong>Maftools</strong> to <strong>sigminer</strong>. So <code>laml</code> can be directly use functions provided by <strong>Maftools</strong>.</p>
<p>As a <code>MAF</code> object, the mutation records are stored in slot <code>data</code> and <code>maf.silent</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="sbssig.html#cb6-1"></a><span class="kw">head</span>(laml<span class="op">@</span>data)</span>
<span id="cb6-2"><a href="sbssig.html#cb6-2"></a><span class="co">#&gt;    Hugo_Symbol Entrez_Gene_Id           Center NCBI_Build Chromosome</span></span>
<span id="cb6-3"><a href="sbssig.html#cb6-3"></a><span class="co">#&gt; 1:      ABCA10          10349 genome.wustl.edu         37         17</span></span>
<span id="cb6-4"><a href="sbssig.html#cb6-4"></a><span class="co">#&gt; 2:       ABCA4             24 genome.wustl.edu         37          1</span></span>
<span id="cb6-5"><a href="sbssig.html#cb6-5"></a><span class="co">#&gt; 3:      ABCB11           8647 genome.wustl.edu         37          2</span></span>
<span id="cb6-6"><a href="sbssig.html#cb6-6"></a><span class="co">#&gt; 4:       ABCC3           8714 genome.wustl.edu         37         17</span></span>
<span id="cb6-7"><a href="sbssig.html#cb6-7"></a><span class="co">#&gt; 5:       ABCF1             23 genome.wustl.edu         37          6</span></span>
<span id="cb6-8"><a href="sbssig.html#cb6-8"></a><span class="co">#&gt; 6:       ABCG4          64137 genome.wustl.edu         37         11</span></span>
<span id="cb6-9"><a href="sbssig.html#cb6-9"></a><span class="co">#&gt;    Start_Position End_Position Strand Variant_Classification Variant_Type</span></span>
<span id="cb6-10"><a href="sbssig.html#cb6-10"></a><span class="co">#&gt; 1:       67170917     67170917      +            Splice_Site          SNP</span></span>
<span id="cb6-11"><a href="sbssig.html#cb6-11"></a><span class="co">#&gt; 2:       94490594     94490594      +      Missense_Mutation          SNP</span></span>
<span id="cb6-12"><a href="sbssig.html#cb6-12"></a><span class="co">#&gt; 3:      169780250    169780250      +      Missense_Mutation          SNP</span></span>
<span id="cb6-13"><a href="sbssig.html#cb6-13"></a><span class="co">#&gt; 4:       48760974     48760974      +      Missense_Mutation          SNP</span></span>
<span id="cb6-14"><a href="sbssig.html#cb6-14"></a><span class="co">#&gt; 5:       30554429     30554429      +      Missense_Mutation          SNP</span></span>
<span id="cb6-15"><a href="sbssig.html#cb6-15"></a><span class="co">#&gt; 6:      119031351    119031351      +      Missense_Mutation          SNP</span></span>
<span id="cb6-16"><a href="sbssig.html#cb6-16"></a><span class="co">#&gt;    Reference_Allele Tumor_Seq_Allele1 Tumor_Seq_Allele2 Tumor_Sample_Barcode</span></span>
<span id="cb6-17"><a href="sbssig.html#cb6-17"></a><span class="co">#&gt; 1:                T                 T                 C         TCGA-AB-2988</span></span>
<span id="cb6-18"><a href="sbssig.html#cb6-18"></a><span class="co">#&gt; 2:                C                 C                 T         TCGA-AB-2869</span></span>
<span id="cb6-19"><a href="sbssig.html#cb6-19"></a><span class="co">#&gt; 3:                G                 G                 A         TCGA-AB-3009</span></span>
<span id="cb6-20"><a href="sbssig.html#cb6-20"></a><span class="co">#&gt; 4:                C                 C                 T         TCGA-AB-2887</span></span>
<span id="cb6-21"><a href="sbssig.html#cb6-21"></a><span class="co">#&gt; 5:                G                 G                 A         TCGA-AB-2920</span></span>
<span id="cb6-22"><a href="sbssig.html#cb6-22"></a><span class="co">#&gt; 6:                A                 A                 G         TCGA-AB-2934</span></span>
<span id="cb6-23"><a href="sbssig.html#cb6-23"></a><span class="co">#&gt;    Protein_Change i_TumorVAF_WU i_transcript_name</span></span>
<span id="cb6-24"><a href="sbssig.html#cb6-24"></a><span class="co">#&gt; 1:        p.K960R      45.66000       NM_080282.3</span></span>
<span id="cb6-25"><a href="sbssig.html#cb6-25"></a><span class="co">#&gt; 2:       p.R1517H      38.12000       NM_000350.2</span></span>
<span id="cb6-26"><a href="sbssig.html#cb6-26"></a><span class="co">#&gt; 3:       p.A1283V      46.97218       NM_003742.2</span></span>
<span id="cb6-27"><a href="sbssig.html#cb6-27"></a><span class="co">#&gt; 4:       p.P1271S      56.41000       NM_003786.1</span></span>
<span id="cb6-28"><a href="sbssig.html#cb6-28"></a><span class="co">#&gt; 5:        p.G658S      40.95000    NM_001025091.1</span></span>
<span id="cb6-29"><a href="sbssig.html#cb6-29"></a><span class="co">#&gt; 6:        p.Y567C      32.84000       NM_022169.1</span></span>
<span id="cb6-30"><a href="sbssig.html#cb6-30"></a><span class="kw">head</span>(laml<span class="op">@</span>maf.silent)</span>
<span id="cb6-31"><a href="sbssig.html#cb6-31"></a><span class="co">#&gt;    Hugo_Symbol Entrez_Gene_Id           Center NCBI_Build Chromosome</span></span>
<span id="cb6-32"><a href="sbssig.html#cb6-32"></a><span class="co">#&gt; 1:      ABCC11          85320 genome.wustl.edu         37         16</span></span>
<span id="cb6-33"><a href="sbssig.html#cb6-33"></a><span class="co">#&gt; 2:        ACAN            176 genome.wustl.edu         37         15</span></span>
<span id="cb6-34"><a href="sbssig.html#cb6-34"></a><span class="co">#&gt; 3:       ACAT1             38 genome.wustl.edu         37         11</span></span>
<span id="cb6-35"><a href="sbssig.html#cb6-35"></a><span class="co">#&gt; 4:       ACCN2             41 genome.wustl.edu         37         12</span></span>
<span id="cb6-36"><a href="sbssig.html#cb6-36"></a><span class="co">#&gt; 5:       ACTA2             59 genome.wustl.edu         37         10</span></span>
<span id="cb6-37"><a href="sbssig.html#cb6-37"></a><span class="co">#&gt; 6:       ACTL9         284382 genome.wustl.edu         37         19</span></span>
<span id="cb6-38"><a href="sbssig.html#cb6-38"></a><span class="co">#&gt;    Start_Position End_Position Strand Variant_Classification Variant_Type</span></span>
<span id="cb6-39"><a href="sbssig.html#cb6-39"></a><span class="co">#&gt; 1:       48244997     48244997      +                 Silent          SNP</span></span>
<span id="cb6-40"><a href="sbssig.html#cb6-40"></a><span class="co">#&gt; 2:       89401084     89401084      +                 Silent          SNP</span></span>
<span id="cb6-41"><a href="sbssig.html#cb6-41"></a><span class="co">#&gt; 3:      108009744    108009744      +                 Silent          SNP</span></span>
<span id="cb6-42"><a href="sbssig.html#cb6-42"></a><span class="co">#&gt; 4:       50452780     50452780      +                 Silent          SNP</span></span>
<span id="cb6-43"><a href="sbssig.html#cb6-43"></a><span class="co">#&gt; 5:       90695109     90695109      +                 Silent          SNP</span></span>
<span id="cb6-44"><a href="sbssig.html#cb6-44"></a><span class="co">#&gt; 6:        8808551      8808551      +                 Silent          SNP</span></span>
<span id="cb6-45"><a href="sbssig.html#cb6-45"></a><span class="co">#&gt;    Reference_Allele Tumor_Seq_Allele1 Tumor_Seq_Allele2 Tumor_Sample_Barcode</span></span>
<span id="cb6-46"><a href="sbssig.html#cb6-46"></a><span class="co">#&gt; 1:                G                 G                 A         TCGA-AB-2830</span></span>
<span id="cb6-47"><a href="sbssig.html#cb6-47"></a><span class="co">#&gt; 2:                C                 C                 T         TCGA-AB-2898</span></span>
<span id="cb6-48"><a href="sbssig.html#cb6-48"></a><span class="co">#&gt; 3:                T                 T                 G         TCGA-AB-2887</span></span>
<span id="cb6-49"><a href="sbssig.html#cb6-49"></a><span class="co">#&gt; 4:                C                 C                 G         TCGA-AB-3009</span></span>
<span id="cb6-50"><a href="sbssig.html#cb6-50"></a><span class="co">#&gt; 5:                C                 C                 T         TCGA-AB-2973</span></span>
<span id="cb6-51"><a href="sbssig.html#cb6-51"></a><span class="co">#&gt; 6:                G                 G                 A         TCGA-AB-2936</span></span>
<span id="cb6-52"><a href="sbssig.html#cb6-52"></a><span class="co">#&gt;    Protein_Change i_TumorVAF_WU i_transcript_name</span></span>
<span id="cb6-53"><a href="sbssig.html#cb6-53"></a><span class="co">#&gt; 1:        p.I490I    34.2700000       NM_032583.3</span></span>
<span id="cb6-54"><a href="sbssig.html#cb6-54"></a><span class="co">#&gt; 2:       p.S1756S    38.3000000       NM_013227.2</span></span>
<span id="cb6-55"><a href="sbssig.html#cb6-55"></a><span class="co">#&gt; 3:        p.T185T    49.0400000       NM_000019.3</span></span>
<span id="cb6-56"><a href="sbssig.html#cb6-56"></a><span class="co">#&gt; 4:         p.L77L    48.1000000       NM_020039.2</span></span>
<span id="cb6-57"><a href="sbssig.html#cb6-57"></a><span class="co">#&gt; 5:        p.P335P     0.2012072       NM_001613.1</span></span>
<span id="cb6-58"><a href="sbssig.html#cb6-58"></a><span class="co">#&gt; 6:        p.F167F    46.1500000       NM_178525.3</span></span></code></pre></div>
<p>The <code>data</code> slot contains non-silent variants, and the <code>maf.silent</code> slot contains silent variants.
Default uses Variant Classifications with High/Moderate variant consequences as non-silent variants. <a href="http://asia.ensembl.org/Help/Glossary?id=535" class="uri">http://asia.ensembl.org/Help/Glossary?id=535</a>: “Frame_Shift_Del”, “Frame_Shift_Ins”, “Splice_Site”, “Translation_Start_Site”,“Nonsense_Mutation”, “Nonstop_Mutation”, “In_Frame_Del”,“In_Frame_Ins”, “Missense_Mutation” (see <code>?read_maf</code>). If you want to change, please set <code>vc_nonSyn</code> option.</p>
<p>Other slots in <code>MAF</code> object are summary data either by sample or gene/variant type etc.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="sbssig.html#cb7-1"></a><span class="kw">slotNames</span>(laml)</span>
<span id="cb7-2"><a href="sbssig.html#cb7-2"></a><span class="co">#&gt; [1] &quot;data&quot;                           &quot;variants.per.sample&quot;           </span></span>
<span id="cb7-3"><a href="sbssig.html#cb7-3"></a><span class="co">#&gt; [3] &quot;variant.type.summary&quot;           &quot;variant.classification.summary&quot;</span></span>
<span id="cb7-4"><a href="sbssig.html#cb7-4"></a><span class="co">#&gt; [5] &quot;gene.summary&quot;                   &quot;summary&quot;                       </span></span>
<span id="cb7-5"><a href="sbssig.html#cb7-5"></a><span class="co">#&gt; [7] &quot;maf.silent&quot;                     &quot;clinical.data&quot;</span></span></code></pre></div>
</div>
<div id="tally-components" class="section level2">
<h2><span class="header-section-number">2.2</span> Tally Components</h2>
<p>According to 3-nucleotide context (mutated base, 5’ and 3’ adjacent bases) and base complementary pairing principle, we can divide all SBS mutations into 96 mutation types. We call each mutation type as a <em>component</em> here.</p>
<blockquote>
<p>This classification is based the six substitution subtypes: C&gt;A, C&gt;G, C&gt;T, T&gt;A, T&gt;C, and T&gt;G (all substitutions are referred to by the pyrimidine of the mutated Watson—Crick base pair). Further, each of the substitutions is examined by incorporating information on the bases immediately 5’ and 3’ to each mutated base generating 96 possible mutation types (6 types of substitution x 4 types of 5’ base x 4 types of 3’ base).</p>
</blockquote>
<p>We tally components in each sample, and generate a sample-by-component matrix.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="sbssig.html#cb8-1"></a><span class="cf">if</span> (<span class="kw">require</span>(<span class="st">&quot;BSgenome.Hsapiens.UCSC.hg19&quot;</span>)) {</span>
<span id="cb8-2"><a href="sbssig.html#cb8-2"></a>  mt_tally &lt;-<span class="st"> </span><span class="kw">sig_tally</span>(</span>
<span id="cb8-3"><a href="sbssig.html#cb8-3"></a>    laml,</span>
<span id="cb8-4"><a href="sbssig.html#cb8-4"></a>    <span class="dt">ref_genome =</span> <span class="st">&quot;BSgenome.Hsapiens.UCSC.hg19&quot;</span>,</span>
<span id="cb8-5"><a href="sbssig.html#cb8-5"></a>    <span class="dt">prefix =</span> <span class="st">&quot;chr&quot;</span>, <span class="dt">add =</span> <span class="ot">TRUE</span>, <span class="dt">useSyn =</span> <span class="ot">TRUE</span></span>
<span id="cb8-6"><a href="sbssig.html#cb8-6"></a>  )</span>
<span id="cb8-7"><a href="sbssig.html#cb8-7"></a>} <span class="cf">else</span> {</span>
<span id="cb8-8"><a href="sbssig.html#cb8-8"></a>  <span class="kw">message</span>(<span class="st">&quot;Please install package &#39;BSgenome.Hsapiens.UCSC.hg19&#39; firstly!&quot;</span>)</span>
<span id="cb8-9"><a href="sbssig.html#cb8-9"></a>}</span>
<span id="cb8-10"><a href="sbssig.html#cb8-10"></a><span class="co">#&gt; Loading required package: BSgenome.Hsapiens.UCSC.hg19</span></span>
<span id="cb8-11"><a href="sbssig.html#cb8-11"></a><span class="co">#&gt; Loading required package: BSgenome</span></span>
<span id="cb8-12"><a href="sbssig.html#cb8-12"></a><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span id="cb8-13"><a href="sbssig.html#cb8-13"></a><span class="co">#&gt; Loading required package: stats4</span></span>
<span id="cb8-14"><a href="sbssig.html#cb8-14"></a><span class="co">#&gt; </span></span>
<span id="cb8-15"><a href="sbssig.html#cb8-15"></a><span class="co">#&gt; Attaching package: &#39;S4Vectors&#39;</span></span>
<span id="cb8-16"><a href="sbssig.html#cb8-16"></a><span class="co">#&gt; The following object is masked from &#39;package:NMF&#39;:</span></span>
<span id="cb8-17"><a href="sbssig.html#cb8-17"></a><span class="co">#&gt; </span></span>
<span id="cb8-18"><a href="sbssig.html#cb8-18"></a><span class="co">#&gt;     nrun</span></span>
<span id="cb8-19"><a href="sbssig.html#cb8-19"></a><span class="co">#&gt; The following object is masked from &#39;package:pkgmaker&#39;:</span></span>
<span id="cb8-20"><a href="sbssig.html#cb8-20"></a><span class="co">#&gt; </span></span>
<span id="cb8-21"><a href="sbssig.html#cb8-21"></a><span class="co">#&gt;     new2</span></span>
<span id="cb8-22"><a href="sbssig.html#cb8-22"></a><span class="co">#&gt; The following object is masked from &#39;package:base&#39;:</span></span>
<span id="cb8-23"><a href="sbssig.html#cb8-23"></a><span class="co">#&gt; </span></span>
<span id="cb8-24"><a href="sbssig.html#cb8-24"></a><span class="co">#&gt;     expand.grid</span></span>
<span id="cb8-25"><a href="sbssig.html#cb8-25"></a><span class="co">#&gt; Loading required package: IRanges</span></span>
<span id="cb8-26"><a href="sbssig.html#cb8-26"></a><span class="co">#&gt; </span></span>
<span id="cb8-27"><a href="sbssig.html#cb8-27"></a><span class="co">#&gt; Attaching package: &#39;IRanges&#39;</span></span>
<span id="cb8-28"><a href="sbssig.html#cb8-28"></a><span class="co">#&gt; The following object is masked from &#39;package:grDevices&#39;:</span></span>
<span id="cb8-29"><a href="sbssig.html#cb8-29"></a><span class="co">#&gt; </span></span>
<span id="cb8-30"><a href="sbssig.html#cb8-30"></a><span class="co">#&gt;     windows</span></span>
<span id="cb8-31"><a href="sbssig.html#cb8-31"></a><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span id="cb8-32"><a href="sbssig.html#cb8-32"></a><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span id="cb8-33"><a href="sbssig.html#cb8-33"></a><span class="co">#&gt; Loading required package: Biostrings</span></span>
<span id="cb8-34"><a href="sbssig.html#cb8-34"></a><span class="co">#&gt; Loading required package: XVector</span></span>
<span id="cb8-35"><a href="sbssig.html#cb8-35"></a><span class="co">#&gt; </span></span>
<span id="cb8-36"><a href="sbssig.html#cb8-36"></a><span class="co">#&gt; Attaching package: &#39;Biostrings&#39;</span></span>
<span id="cb8-37"><a href="sbssig.html#cb8-37"></a><span class="co">#&gt; The following object is masked from &#39;package:base&#39;:</span></span>
<span id="cb8-38"><a href="sbssig.html#cb8-38"></a><span class="co">#&gt; </span></span>
<span id="cb8-39"><a href="sbssig.html#cb8-39"></a><span class="co">#&gt;     strsplit</span></span>
<span id="cb8-40"><a href="sbssig.html#cb8-40"></a><span class="co">#&gt; Loading required package: rtracklayer</span></span>
<span id="cb8-41"><a href="sbssig.html#cb8-41"></a><span class="co">#&gt; Warning in maftools::trinucleotideMatrix(object, ref_genome = ref_genome, : Chromosome names in MAF must match chromosome names in reference genome.</span></span>
<span id="cb8-42"><a href="sbssig.html#cb8-42"></a><span class="co">#&gt; Ignorinig 101 single nucleotide variants from missing chromosomes chr23</span></span>
<span id="cb8-43"><a href="sbssig.html#cb8-43"></a><span class="co">#&gt; -Extracting 5&#39; and 3&#39; adjacent bases</span></span>
<span id="cb8-44"><a href="sbssig.html#cb8-44"></a><span class="co">#&gt; -Extracting +/- 20bp around mutated bases for background C&gt;T estimation</span></span>
<span id="cb8-45"><a href="sbssig.html#cb8-45"></a><span class="co">#&gt; -Estimating APOBEC enrichment scores</span></span>
<span id="cb8-46"><a href="sbssig.html#cb8-46"></a><span class="co">#&gt; --Performing one-way Fisher&#39;s test for APOBEC enrichment</span></span>
<span id="cb8-47"><a href="sbssig.html#cb8-47"></a><span class="co">#&gt; ---APOBEC related mutations are enriched in  3.315 % of samples (APOBEC enrichment score &gt; 2 ;  6  of  181  samples)</span></span>
<span id="cb8-48"><a href="sbssig.html#cb8-48"></a><span class="co">#&gt; -Creating mutation matrix</span></span>
<span id="cb8-49"><a href="sbssig.html#cb8-49"></a><span class="co">#&gt; --matrix of dimension 188x96</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="sbssig.html#cb9-1"></a>mt_tally<span class="op">$</span>nmf_matrix[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</span>
<span id="cb9-2"><a href="sbssig.html#cb9-2"></a><span class="co">#&gt;              A[C&gt;A]A A[C&gt;A]C A[C&gt;A]G A[C&gt;A]T C[C&gt;A]A</span></span>
<span id="cb9-3"><a href="sbssig.html#cb9-3"></a><span class="co">#&gt; TCGA-AB-2802       0       0       0       0       0</span></span>
<span id="cb9-4"><a href="sbssig.html#cb9-4"></a><span class="co">#&gt; TCGA-AB-2803       0       2       0       0       0</span></span>
<span id="cb9-5"><a href="sbssig.html#cb9-5"></a><span class="co">#&gt; TCGA-AB-2804       0       0       0       0       0</span></span>
<span id="cb9-6"><a href="sbssig.html#cb9-6"></a><span class="co">#&gt; TCGA-AB-2805       0       0       0       0       0</span></span>
<span id="cb9-7"><a href="sbssig.html#cb9-7"></a><span class="co">#&gt; TCGA-AB-2806       0       0       0       0       0</span></span></code></pre></div>
<p>We use notion <code>left[ref&gt;mut]right</code> to mark each component, e.g. <code>C[T&gt;G]A</code> means a base T with 5’ adjacent base C and 3’ adjacent base A is mutated to base G.</p>
<blockquote>
<p>If you have used signature extraction feature in Maftools, you may feel familiar.
Currently, Maftools is the backend for <code>read_maf()</code> and <code>sig_tally</code> for SBS mutation.</p>
</blockquote>
</div>
<div id="extract-signatures" class="section level2">
<h2><span class="header-section-number">2.3</span> Extract Signatures</h2>
<p>There are two methods can be used to implement signature extraction.</p>
<ol style="list-style-type: decimal">
<li>Firstly estimate signature number and then extract signatures. This method is powered by <a href="https://github.com/renozao/NMF"><strong>NMF</strong></a> package <span class="citation">(Gaujoux and Seoighe <a href="#ref-gaujoux2010flexible" role="doc-biblioref">2010</a>)</span>.</li>
<li>Auto-extract signatures by automatic relevance determination technique in nonnegative matrix factorization <span class="citation">(Tan and Févotte <a href="#ref-tan2012automatic" role="doc-biblioref">2012</a>)</span>, the code is implemented by <strong>SignatureAnalyzer</strong> <span class="citation">(Kim et al. <a href="#ref-kim2016somatic" role="doc-biblioref">2016</a>)</span> and exported to <strong>sigminer</strong>.</li>
</ol>
<div id="method-1-estimate-signature-number" class="section level3">
<h3><span class="header-section-number">2.3.1</span> Method 1: Estimate Signature Number</h3>
<p>To estimate signature number, we firstly run NMF for multiple times and calculate some measures from the results, then select the best signature number based on the measure vs. signature number plot.</p>
<p>This is done by <code>sig_estimate()</code> function. Let’s try signature number 2-5. For simplicity, we just run NMF twice for each signature number. We use 4 cores to speed up the computation.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="sbssig.html#cb10-1"></a>mt_est &lt;-<span class="st"> </span><span class="kw">sig_estimate</span>(mt_tally<span class="op">$</span>nmf_matrix,</span>
<span id="cb10-2"><a href="sbssig.html#cb10-2"></a>  <span class="dt">range =</span> <span class="dv">2</span><span class="op">:</span><span class="dv">5</span>,</span>
<span id="cb10-3"><a href="sbssig.html#cb10-3"></a>  <span class="dt">nrun =</span> <span class="dv">2</span>,</span>
<span id="cb10-4"><a href="sbssig.html#cb10-4"></a>  <span class="dt">use_random =</span> <span class="ot">TRUE</span>,</span>
<span id="cb10-5"><a href="sbssig.html#cb10-5"></a>  <span class="dt">cores =</span> <span class="dv">4</span>,</span>
<span id="cb10-6"><a href="sbssig.html#cb10-6"></a>  <span class="dt">pConstant =</span> <span class="fl">1e-13</span>,</span>
<span id="cb10-7"><a href="sbssig.html#cb10-7"></a>  <span class="dt">verbose =</span> <span class="ot">TRUE</span></span>
<span id="cb10-8"><a href="sbssig.html#cb10-8"></a>)</span></code></pre></div>
<blockquote>
<p><code>pConstant</code> option is set to avoid errors raised by <strong>NMF</strong> package.</p>
</blockquote>
<p>We can show signature number survey for different measures by <code>show_sig_number_survey2()</code>.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="sbssig.html#cb11-1"></a><span class="co">## You can also select the measures to show</span></span>
<span id="cb11-2"><a href="sbssig.html#cb11-2"></a><span class="co">## by &#39;what&#39; option</span></span>
<span id="cb11-3"><a href="sbssig.html#cb11-3"></a><span class="kw">show_sig_number_survey2</span>(mt_est<span class="op">$</span>survey, mt_est<span class="op">$</span>survey.random)</span></code></pre></div>
<p><img src="sigminer-doc_files/figure-html/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>For the details of all the measures above, please read <span class="citation">Gaujoux and Seoighe (<a href="#ref-gaujoux2010flexible" role="doc-biblioref">2010</a>)</span> and <a href="https://cran.r-project.org/web/packages/NMF/vignettes/">vignette</a> of R package <strong>NMF</strong>.</p>
<p>Typically, measure <strong>cophenetic</strong> is used for determing the signature number. We can easily generate an elbow plot with <code>function</code>show_sig_number_survey()`.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="sbssig.html#cb12-1"></a><span class="kw">show_sig_number_survey</span>(mt_est<span class="op">$</span>survey, <span class="dt">right_y =</span> <span class="ot">NULL</span>)</span></code></pre></div>
<p><img src="sigminer-doc_files/figure-html/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Cophenetic indicates the robustness of consensus matrix clustering. In this situation, 2 or 3 is good.</p>
<blockquote>
<p>The most common approach is to use the cophenetic correlation coefficient. Brunet et al. suggested choosing the smallest value of r for which this coefficient starts decreasing. <span class="citation">(Gaujoux and Seoighe <a href="#ref-gaujoux2010flexible" role="doc-biblioref">2010</a>)</span></p>
</blockquote>
<p>However, there are no gold standard to determine the signature number. Sometimes, you should consider multiple measures. Remember, the most important thing is that you should have a good biological explanation for each signature.
The best solution in study may not be the best solution in math.</p>
</div>
<div id="method-1-extract-signatures" class="section level3">
<h3><span class="header-section-number">2.3.2</span> Method 1: Extract Signatures</h3>
<p>After selecting a proper signature, you can now extract signatures. In general, use 30~50 NMF runs will get a robust result.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="sbssig.html#cb13-1"></a>mt_sig &lt;-<span class="st"> </span><span class="kw">sig_extract</span>(mt_tally<span class="op">$</span>nmf_matrix,</span>
<span id="cb13-2"><a href="sbssig.html#cb13-2"></a>  <span class="dt">n_sig =</span> <span class="dv">3</span>,</span>
<span id="cb13-3"><a href="sbssig.html#cb13-3"></a>  <span class="dt">nrun =</span> <span class="dv">10</span>,</span>
<span id="cb13-4"><a href="sbssig.html#cb13-4"></a>  <span class="dt">cores =</span> <span class="dv">4</span>,</span>
<span id="cb13-5"><a href="sbssig.html#cb13-5"></a>  <span class="dt">pConstant =</span> <span class="fl">1e-13</span></span>
<span id="cb13-6"><a href="sbssig.html#cb13-6"></a>)</span></code></pre></div>
</div>
<div id="method-2-auto-extract" class="section level3">
<h3><span class="header-section-number">2.3.3</span> Method 2: Auto-extract</h3>
<p>If you have no idea to select a optimal signature number from technologies above, you can try auto-extract feature here. It uses a bayesian variant of NMF algorithm. You need to set a maximum signature number (default is <code>25</code>) and run times to get the result.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="sbssig.html#cb14-1"></a>mt_sig2 &lt;-<span class="st"> </span><span class="kw">sig_auto_extract</span>(mt_tally<span class="op">$</span>nmf_matrix, <span class="dt">K0 =</span> <span class="dv">10</span>, <span class="dt">nrun =</span> <span class="dv">10</span>)</span></code></pre></div>
<p>At default, the code will use ‘robust’ strategy to return the result (see <code>strategy</code> option). It means that if you run 10 times and 6 of them return <code>4</code> signatures, then the optimal result with <code>4</code> signatures will be returned.</p>
<p>The info of each run can be given as:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="sbssig.html#cb15-1"></a>mt_sig2<span class="op">$</span>Raw<span class="op">$</span>summary_run</span>
<span id="cb15-2"><a href="sbssig.html#cb15-2"></a><span class="co">#&gt; # A tibble: 10 x 4</span></span>
<span id="cb15-3"><a href="sbssig.html#cb15-3"></a><span class="co">#&gt;      Run     K posterior file                                      </span></span>
<span id="cb15-4"><a href="sbssig.html#cb15-4"></a><span class="co">#&gt;    &lt;int&gt; &lt;int&gt;     &lt;dbl&gt; &lt;chr&gt;                                     </span></span>
<span id="cb15-5"><a href="sbssig.html#cb15-5"></a><span class="co">#&gt;  1     9     2     1813. &quot;D:/Tool/Rtmp\\RtmpURv3Ba/BayesNMF.9.rds&quot; </span></span>
<span id="cb15-6"><a href="sbssig.html#cb15-6"></a><span class="co">#&gt;  2     6     3     1776. &quot;D:/Tool/Rtmp\\RtmpURv3Ba/BayesNMF.6.rds&quot; </span></span>
<span id="cb15-7"><a href="sbssig.html#cb15-7"></a><span class="co">#&gt;  3     3     2     1775. &quot;D:/Tool/Rtmp\\RtmpURv3Ba/BayesNMF.3.rds&quot; </span></span>
<span id="cb15-8"><a href="sbssig.html#cb15-8"></a><span class="co">#&gt;  4     5     3     1745. &quot;D:/Tool/Rtmp\\RtmpURv3Ba/BayesNMF.5.rds&quot; </span></span>
<span id="cb15-9"><a href="sbssig.html#cb15-9"></a><span class="co">#&gt;  5     8     4     1732. &quot;D:/Tool/Rtmp\\RtmpURv3Ba/BayesNMF.8.rds&quot; </span></span>
<span id="cb15-10"><a href="sbssig.html#cb15-10"></a><span class="co">#&gt;  6     7     4     1702. &quot;D:/Tool/Rtmp\\RtmpURv3Ba/BayesNMF.7.rds&quot; </span></span>
<span id="cb15-11"><a href="sbssig.html#cb15-11"></a><span class="co">#&gt;  7     1     4     1684. &quot;D:/Tool/Rtmp\\RtmpURv3Ba/BayesNMF.1.rds&quot; </span></span>
<span id="cb15-12"><a href="sbssig.html#cb15-12"></a><span class="co">#&gt;  8     4     4     1661. &quot;D:/Tool/Rtmp\\RtmpURv3Ba/BayesNMF.4.rds&quot; </span></span>
<span id="cb15-13"><a href="sbssig.html#cb15-13"></a><span class="co">#&gt;  9    10     4     1658. &quot;D:/Tool/Rtmp\\RtmpURv3Ba/BayesNMF.10.rds&quot;</span></span>
<span id="cb15-14"><a href="sbssig.html#cb15-14"></a><span class="co">#&gt; 10     2     5     1652. &quot;D:/Tool/Rtmp\\RtmpURv3Ba/BayesNMF.2.rds&quot;</span></span></code></pre></div>
</div>
</div>
<div id="match-signatures" class="section level2">
<h2><span class="header-section-number">2.4</span> Match Signatures</h2>
<p>After extracting signatures, we need to know their etiologies. This can be done by comparing the identified signatures and reference signatures from COSMIC database.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="sbssig.html#cb16-1"></a>sim &lt;-<span class="st"> </span><span class="kw">get_sig_similarity</span>(mt_sig)</span>
<span id="cb16-2"><a href="sbssig.html#cb16-2"></a><span class="co">#&gt; -Comparing against COSMIC signatures</span></span>
<span id="cb16-3"><a href="sbssig.html#cb16-3"></a><span class="co">#&gt; ------------------------------------</span></span>
<span id="cb16-4"><a href="sbssig.html#cb16-4"></a><span class="co">#&gt; --Found Sig1 most similar to COSMIC_6</span></span>
<span id="cb16-5"><a href="sbssig.html#cb16-5"></a><span class="co">#&gt;    Aetiology: defective DNA mismatch repair [similarity: 0.667]</span></span>
<span id="cb16-6"><a href="sbssig.html#cb16-6"></a><span class="co">#&gt; --Found Sig2 most similar to COSMIC_1</span></span>
<span id="cb16-7"><a href="sbssig.html#cb16-7"></a><span class="co">#&gt;    Aetiology: spontaneous deamination of 5-methylcytosine [similarity: 0.84]</span></span>
<span id="cb16-8"><a href="sbssig.html#cb16-8"></a><span class="co">#&gt; --Found Sig3 most similar to COSMIC_1</span></span>
<span id="cb16-9"><a href="sbssig.html#cb16-9"></a><span class="co">#&gt;    Aetiology: spontaneous deamination of 5-methylcytosine [similarity: 0.922]</span></span>
<span id="cb16-10"><a href="sbssig.html#cb16-10"></a><span class="co">#&gt; ------------------------------------</span></span>
<span id="cb16-11"><a href="sbssig.html#cb16-11"></a><span class="co">#&gt; Return result invisiblely.</span></span></code></pre></div>
<p>The result object <code>sim</code> is a list.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="sbssig.html#cb17-1"></a><span class="kw">str</span>(sim)</span>
<span id="cb17-2"><a href="sbssig.html#cb17-2"></a><span class="co">#&gt; List of 3</span></span>
<span id="cb17-3"><a href="sbssig.html#cb17-3"></a><span class="co">#&gt;  $ similarity  : num [1:3, 1:30] 0.612 0.84 0.922 0.212 0.172 0.105 0.362 0.327 0.236 0.295 ...</span></span>
<span id="cb17-4"><a href="sbssig.html#cb17-4"></a><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></span>
<span id="cb17-5"><a href="sbssig.html#cb17-5"></a><span class="co">#&gt;   .. ..$ : chr [1:3] &quot;Sig1&quot; &quot;Sig2&quot; &quot;Sig3&quot;</span></span>
<span id="cb17-6"><a href="sbssig.html#cb17-6"></a><span class="co">#&gt;   .. ..$ : chr [1:30] &quot;COSMIC_1&quot; &quot;COSMIC_2&quot; &quot;COSMIC_3&quot; &quot;COSMIC_4&quot; ...</span></span>
<span id="cb17-7"><a href="sbssig.html#cb17-7"></a><span class="co">#&gt;  $ aetiology_db:List of 1</span></span>
<span id="cb17-8"><a href="sbssig.html#cb17-8"></a><span class="co">#&gt;   ..$ : chr [1:30] &quot;spontaneous deamination of 5-methylcytosine&quot; &quot;APOBEC Cytidine Deaminase (C&gt;T)&quot; &quot;defects in DNA-DSB repair by HR&quot; &quot;exposure to tobacco (smoking) mutagens&quot; ...</span></span>
<span id="cb17-9"><a href="sbssig.html#cb17-9"></a><span class="co">#&gt;  $ best_match  :List of 3</span></span>
<span id="cb17-10"><a href="sbssig.html#cb17-10"></a><span class="co">#&gt;   ..$ Sig1:List of 2</span></span>
<span id="cb17-11"><a href="sbssig.html#cb17-11"></a><span class="co">#&gt;   .. ..$ aetiology : chr &quot;defective DNA mismatch repair&quot;</span></span>
<span id="cb17-12"><a href="sbssig.html#cb17-12"></a><span class="co">#&gt;   .. ..$ best_match: chr &quot;Best match: COSMIC_6 [similarity: 0.667]&quot;</span></span>
<span id="cb17-13"><a href="sbssig.html#cb17-13"></a><span class="co">#&gt;   ..$ Sig2:List of 2</span></span>
<span id="cb17-14"><a href="sbssig.html#cb17-14"></a><span class="co">#&gt;   .. ..$ aetiology : chr &quot;spontaneous deamination of 5-methylcytosine&quot;</span></span>
<span id="cb17-15"><a href="sbssig.html#cb17-15"></a><span class="co">#&gt;   .. ..$ best_match: chr &quot;Best match: COSMIC_1 [similarity: 0.84]&quot;</span></span>
<span id="cb17-16"><a href="sbssig.html#cb17-16"></a><span class="co">#&gt;   ..$ Sig3:List of 2</span></span>
<span id="cb17-17"><a href="sbssig.html#cb17-17"></a><span class="co">#&gt;   .. ..$ aetiology : chr &quot;spontaneous deamination of 5-methylcytosine&quot;</span></span>
<span id="cb17-18"><a href="sbssig.html#cb17-18"></a><span class="co">#&gt;   .. ..$ best_match: chr &quot;Best match: COSMIC_1 [similarity: 0.922]&quot;</span></span>
<span id="cb17-19"><a href="sbssig.html#cb17-19"></a><span class="co">#&gt;  - attr(*, &quot;class&quot;)= chr [1:2] &quot;similarity&quot; &quot;list&quot;</span></span></code></pre></div>
<p>From the result we can see that only one signature is clear. If you find unknow signatures in your study, you should explore the etiologies by other analyses and even experiments.</p>
<p>The similarity matrix can be plotted.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="sbssig.html#cb18-1"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(sim<span class="op">$</span>similarity)</span></code></pre></div>
<p><img src="sigminer-doc_files/figure-html/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>You can also try the COSMIC signature database V3 with:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="sbssig.html#cb19-1"></a>sim_v3 &lt;-<span class="st"> </span><span class="kw">get_sig_similarity</span>(mt_sig, <span class="dt">sig_db =</span> <span class="st">&quot;SBS&quot;</span>)</span>
<span id="cb19-2"><a href="sbssig.html#cb19-2"></a><span class="co">#&gt; -Comparing against COSMIC signatures</span></span>
<span id="cb19-3"><a href="sbssig.html#cb19-3"></a><span class="co">#&gt; ------------------------------------</span></span>
<span id="cb19-4"><a href="sbssig.html#cb19-4"></a><span class="co">#&gt; --Found Sig1 most similar to SBS6</span></span>
<span id="cb19-5"><a href="sbssig.html#cb19-5"></a><span class="co">#&gt;    Aetiology: defective DNA mismatch repair [similarity: 0.659]</span></span>
<span id="cb19-6"><a href="sbssig.html#cb19-6"></a><span class="co">#&gt; --Found Sig2 most similar to SBS1</span></span>
<span id="cb19-7"><a href="sbssig.html#cb19-7"></a><span class="co">#&gt;    Aetiology: spontaneous or enzymatic deamination of 5-methylcytosine [similarity: 0.796]</span></span>
<span id="cb19-8"><a href="sbssig.html#cb19-8"></a><span class="co">#&gt; --Found Sig3 most similar to SBS1</span></span>
<span id="cb19-9"><a href="sbssig.html#cb19-9"></a><span class="co">#&gt;    Aetiology: spontaneous or enzymatic deamination of 5-methylcytosine [similarity: 0.905]</span></span>
<span id="cb19-10"><a href="sbssig.html#cb19-10"></a><span class="co">#&gt; ------------------------------------</span></span>
<span id="cb19-11"><a href="sbssig.html#cb19-11"></a><span class="co">#&gt; Return result invisiblely.</span></span></code></pre></div>
</div>
<div id="operate-signature" class="section level2">
<h2><span class="header-section-number">2.5</span> Operate <code>Signature</code></h2>
<p>The result of <code>sig_extract()</code> or <code>sig_auto_extract()</code> is a <code>list</code> with <code>Signature</code> class. You can use <code>$</code> or use operation function to obtain the data stored in it.</p>
<p>To get the signature matrix:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="sbssig.html#cb20-1"></a><span class="co"># Or mt_sig$Signature</span></span>
<span id="cb20-2"><a href="sbssig.html#cb20-2"></a><span class="kw">sig_signature</span>(mt_sig)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, ]</span>
<span id="cb20-3"><a href="sbssig.html#cb20-3"></a><span class="co">#&gt;                 Sig1         Sig2         Sig3</span></span>
<span id="cb20-4"><a href="sbssig.html#cb20-4"></a><span class="co">#&gt; A[C&gt;A]A 2.573440e-03 5.328563e-03 1.452366e-02</span></span>
<span id="cb20-5"><a href="sbssig.html#cb20-5"></a><span class="co">#&gt; A[C&gt;A]C 5.615313e-19 4.431593e-02 2.663616e-14</span></span>
<span id="cb20-6"><a href="sbssig.html#cb20-6"></a><span class="co">#&gt; A[C&gt;A]G 1.456309e-15 6.330847e-03 1.252344e-14</span></span>
<span id="cb20-7"><a href="sbssig.html#cb20-7"></a><span class="co">#&gt; A[C&gt;A]T 9.651010e-15 4.254580e-15 1.142587e-02</span></span>
<span id="cb20-8"><a href="sbssig.html#cb20-8"></a><span class="co">#&gt; C[C&gt;A]A 6.955815e-03 2.937405e-14 2.005539e-02</span></span></code></pre></div>
<p>To get signature exposure matrix:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="sbssig.html#cb21-1"></a><span class="co"># Or mt_sig$Exposure</span></span>
<span id="cb21-2"><a href="sbssig.html#cb21-2"></a><span class="kw">sig_exposure</span>(mt_sig)[, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</span>
<span id="cb21-3"><a href="sbssig.html#cb21-3"></a><span class="co">#&gt;      TCGA-AB-2802 TCGA-AB-2803 TCGA-AB-2804 TCGA-AB-2805 TCGA-AB-2806</span></span>
<span id="cb21-4"><a href="sbssig.html#cb21-4"></a><span class="co">#&gt; Sig1 5.951574e+00 6.874475e+00 8.830048e-13 2.457190e-13 1.764339e-12</span></span>
<span id="cb21-5"><a href="sbssig.html#cb21-5"></a><span class="co">#&gt; Sig2 2.106827e-12 7.125541e+00 1.593764e+00 6.964313e+00 3.483714e-12</span></span>
<span id="cb21-6"><a href="sbssig.html#cb21-6"></a><span class="co">#&gt; Sig3 4.048419e+00 9.300760e-14 3.406233e+00 7.035688e+00 1.299997e+01</span></span></code></pre></div>
<p><code>get_sig_exposure()</code> may be more useful, it can be used to return a <code>data.frame</code> and set an exposure threshold.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="sbssig.html#cb22-1"></a><span class="kw">get_sig_exposure</span>(mt_sig)</span>
<span id="cb22-2"><a href="sbssig.html#cb22-2"></a><span class="co">#&gt;            sample         Sig1         Sig2         Sig3</span></span>
<span id="cb22-3"><a href="sbssig.html#cb22-3"></a><span class="co">#&gt;   1: TCGA-AB-2802 5.951574e+00 2.106827e-12 4.048419e+00</span></span>
<span id="cb22-4"><a href="sbssig.html#cb22-4"></a><span class="co">#&gt;   2: TCGA-AB-2803 6.874475e+00 7.125541e+00 9.300760e-14</span></span>
<span id="cb22-5"><a href="sbssig.html#cb22-5"></a><span class="co">#&gt;   3: TCGA-AB-2804 8.830048e-13 1.593764e+00 3.406233e+00</span></span>
<span id="cb22-6"><a href="sbssig.html#cb22-6"></a><span class="co">#&gt;   4: TCGA-AB-2805 2.457190e-13 6.964313e+00 7.035688e+00</span></span>
<span id="cb22-7"><a href="sbssig.html#cb22-7"></a><span class="co">#&gt;   5: TCGA-AB-2806 1.764339e-12 3.483714e-12 1.299997e+01</span></span>
<span id="cb22-8"><a href="sbssig.html#cb22-8"></a><span class="co">#&gt;  ---                                                    </span></span>
<span id="cb22-9"><a href="sbssig.html#cb22-9"></a><span class="co">#&gt; 184: TCGA-AB-3007 1.014093e-12 4.575247e+00 2.424758e+00</span></span>
<span id="cb22-10"><a href="sbssig.html#cb22-10"></a><span class="co">#&gt; 185: TCGA-AB-3008 1.294223e-11 4.000009e+00 3.528735e-12</span></span>
<span id="cb22-11"><a href="sbssig.html#cb22-11"></a><span class="co">#&gt; 186: TCGA-AB-3009 8.818428e+00 7.079667e+00 1.910188e+01</span></span>
<span id="cb22-12"><a href="sbssig.html#cb22-12"></a><span class="co">#&gt; 187: TCGA-AB-3011 5.000001e+00 4.441747e-11 1.666433e-11</span></span>
<span id="cb22-13"><a href="sbssig.html#cb22-13"></a><span class="co">#&gt; 188: TCGA-AB-3012 1.385718e-13 2.810821e+00 3.189179e+00</span></span></code></pre></div>
<p>For plotting signature profile and exposure profile, please go to chapter <a href="sigobject.html#sigobject">6</a>.</p>
<p>For optimizing signature exposure, please use <code>sig_fit()</code>.</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-gaujoux2010flexible">
<p>Gaujoux, Renaud, and Cathal Seoighe. 2010. “A Flexible R Package for Nonnegative Matrix Factorization.” <em>BMC Bioinformatics</em> 11 (1): 367.</p>
</div>
<div id="ref-kim2016somatic">
<p>Kim, Jaegil, Kent W Mouw, Paz Polak, Lior Z Braunstein, Atanas Kamburov, Grace Tiao, David J Kwiatkowski, et al. 2016. “Somatic Ercc2 Mutations Are Associated with a Distinct Genomic Signature in Urothelial Tumors.” <em>Nature Genetics</em> 48 (6): 600.</p>
</div>
<div id="ref-mayakonda2018maftools">
<p>Mayakonda, Anand, De-Chen Lin, Yassen Assenov, Christoph Plass, and H Phillip Koeffler. 2018. “Maftools: Efficient and Comprehensive Analysis of Somatic Variants in Cancer.” <em>Genome Research</em> 28 (11): 1747–56.</p>
</div>
<div id="ref-tan2012automatic">
<p>Tan, Vincent YF, and Cédric Févotte. 2012. “Automatic Relevance Determination in Nonnegative Matrix Factorization with the/Spl Beta/-Divergence.” <em>IEEE Transactions on Pattern Analysis and Machine Intelligence</em> 35 (7): 1592–1605.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="part-common-workflow.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="cnsig.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/ShixiangWang/sigminer-doc/edit/master/01-SBS-signature-identification.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["sigminer-doc.pdf", "sigminer-doc.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
