<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Sigflow Pipeline | Sigminer: A Scalable Toolkit to Extract, Analyze and Visualize Mutational Signatures</title>
  <meta name="description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Sigflow Pipeline | Sigminer: A Scalable Toolkit to Extract, Analyze and Visualize Mutational Signatures" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  <meta name="github-repo" content="ShixiangWang/sigminer-doc" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Sigflow Pipeline | Sigminer: A Scalable Toolkit to Extract, Analyze and Visualize Mutational Signatures" />
  
  <meta name="twitter:description" content="This package helps users to extract, analyze and visualize signatures from genomic variation records, thus providing new insight into cancer study." />
  

<meta name="author" content="Shixiang Wang, ShanghaiTech University" />
<meta name="author" content="Xue-Song Liu, ShanghaiTech University" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="subtyping.html"/>
<link rel="next" href="cnobject.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">sigminer: Signature Miner</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#biological-significance-of-mutational-signature"><i class="fa fa-check"></i><b>1.1</b> Biological Significance of Mutational Signature</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#sigminer"><i class="fa fa-check"></i><b>1.2</b> Sigminer</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.3</b> Installation</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#issues-or-suggestions"><i class="fa fa-check"></i><b>1.4</b> Issues or Suggestions</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#preparation"><i class="fa fa-check"></i><b>1.5</b> Preparation</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#overview-of-contents"><i class="fa fa-check"></i><b>1.6</b> Overview of Contents</a></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#citation-and-license"><i class="fa fa-check"></i><b>1.7</b> Citation and LICENSE</a></li>
</ul></li>
<li class="part"><span><b>I Common Workflow</b></span></li>
<li class="chapter" data-level="2" data-path="sbssig.html"><a href="sbssig.html"><i class="fa fa-check"></i><b>2</b> COSMIC Signature Identification</a><ul>
<li class="chapter" data-level="2.1" data-path="sbssig.html"><a href="sbssig.html#read-data"><i class="fa fa-check"></i><b>2.1</b> Read Data</a></li>
<li class="chapter" data-level="2.2" data-path="sbssig.html"><a href="sbssig.html#tally-components"><i class="fa fa-check"></i><b>2.2</b> Tally Components</a><ul>
<li class="chapter" data-level="2.2.1" data-path="sbssig.html"><a href="sbssig.html#the-most-common-96-components"><i class="fa fa-check"></i><b>2.2.1</b> The most common 96 components</a></li>
<li class="chapter" data-level="2.2.2" data-path="sbssig.html"><a href="sbssig.html#other-situations"><i class="fa fa-check"></i><b>2.2.2</b> Other Situations</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="sbssig.html"><a href="sbssig.html#extract-signatures"><i class="fa fa-check"></i><b>2.3</b> Extract Signatures</a><ul>
<li class="chapter" data-level="2.3.1" data-path="sbssig.html"><a href="sbssig.html#method-1-estimate-signature-number"><i class="fa fa-check"></i><b>2.3.1</b> Method 1: Estimate Signature Number</a></li>
<li class="chapter" data-level="2.3.2" data-path="sbssig.html"><a href="sbssig.html#method-1-extract-signatures"><i class="fa fa-check"></i><b>2.3.2</b> Method 1: Extract Signatures</a></li>
<li class="chapter" data-level="2.3.3" data-path="sbssig.html"><a href="sbssig.html#method-2-auto-extract"><i class="fa fa-check"></i><b>2.3.3</b> Method 2: Auto-extract</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="sbssig.html"><a href="sbssig.html#match-signatures"><i class="fa fa-check"></i><b>2.4</b> Match Signatures</a></li>
<li class="chapter" data-level="2.5" data-path="sbssig.html"><a href="sbssig.html#operate-signature"><i class="fa fa-check"></i><b>2.5</b> Operate <code>Signature</code></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cnsig.html"><a href="cnsig.html"><i class="fa fa-check"></i><b>3</b> Copy Number Signature Identification</a><ul>
<li class="chapter" data-level="3.1" data-path="cnsig.html"><a href="cnsig.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="cnsig.html"><a href="cnsig.html#read-data-1"><i class="fa fa-check"></i><b>3.2</b> Read Data</a></li>
<li class="chapter" data-level="3.3" data-path="cnsig.html"><a href="cnsig.html#tally-components-1"><i class="fa fa-check"></i><b>3.3</b> Tally Components</a></li>
<li class="chapter" data-level="3.4" data-path="cnsig.html"><a href="cnsig.html#extract-signatures-1"><i class="fa fa-check"></i><b>3.4</b> Extract Signatures</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sigfit.html"><a href="sigfit.html"><i class="fa fa-check"></i><b>4</b> Signature Fit: Sample Signature Exposure Quantification and Analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="sigfit.html"><a href="sigfit.html#fit-signatures-from-reference-databases"><i class="fa fa-check"></i><b>4.1</b> Fit Signatures from reference databases</a></li>
<li class="chapter" data-level="4.2" data-path="sigfit.html"><a href="sigfit.html#fit-custom-signatures"><i class="fa fa-check"></i><b>4.2</b> Fit Custom Signatures</a></li>
<li class="chapter" data-level="4.3" data-path="sigfit.html"><a href="sigfit.html#performance-comparison"><i class="fa fa-check"></i><b>4.3</b> Performance Comparison</a></li>
<li class="chapter" data-level="4.4" data-path="sigfit.html"><a href="sigfit.html#estimate-exposure-stability-by-bootstrap"><i class="fa fa-check"></i><b>4.4</b> Estimate Exposure Stability by Bootstrap</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="subtyping.html"><a href="subtyping.html"><i class="fa fa-check"></i><b>5</b> Subtype Prediction</a></li>
<li class="chapter" data-level="6" data-path="sigflow.html"><a href="sigflow.html"><i class="fa fa-check"></i><b>6</b> Sigflow Pipeline</a><ul>
<li class="chapter" data-level="6.1" data-path="sigflow.html"><a href="sigflow.html#installation-1"><i class="fa fa-check"></i><b>6.1</b> Installation</a><ul>
<li class="chapter" data-level="6.1.1" data-path="sigflow.html"><a href="sigflow.html#step-by-step-installation"><i class="fa fa-check"></i><b>6.1.1</b> Step-by-step installation</a></li>
<li class="chapter" data-level="6.1.2" data-path="sigflow.html"><a href="sigflow.html#install-by-docker"><i class="fa fa-check"></i><b>6.1.2</b> Install by Docker</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="sigflow.html"><a href="sigflow.html#usage"><i class="fa fa-check"></i><b>6.2</b> Usage</a></li>
<li class="chapter" data-level="6.3" data-path="sigflow.html"><a href="sigflow.html#suggestion-and-bug-report"><i class="fa fa-check"></i><b>6.3</b> Suggestion and bug report</a></li>
<li class="chapter" data-level="6.4" data-path="sigflow.html"><a href="sigflow.html#examples"><i class="fa fa-check"></i><b>6.4</b> Examples</a><ul>
<li class="chapter" data-level="6.4.1" data-path="sigflow.html"><a href="sigflow.html#extract-command"><i class="fa fa-check"></i><b>6.4.1</b> <code>extract</code> command</a></li>
<li class="chapter" data-level="6.4.2" data-path="sigflow.html"><a href="sigflow.html#fit-command"><i class="fa fa-check"></i><b>6.4.2</b> <code>fit</code> command</a></li>
<li class="chapter" data-level="6.4.3" data-path="sigflow.html"><a href="sigflow.html#bt-command"><i class="fa fa-check"></i><b>6.4.3</b> <code>bt</code> command</a></li>
<li class="chapter" data-level="6.4.4" data-path="sigflow.html"><a href="sigflow.html#how-to-use-docker-to-run-sigflow"><i class="fa fa-check"></i><b>6.4.4</b> How to use Docker to run Sigflow</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Target Visualization</b></span></li>
<li class="chapter" data-level="7" data-path="cnobject.html"><a href="cnobject.html"><i class="fa fa-check"></i><b>7</b> CopyNumber Object</a><ul>
<li class="chapter" data-level="7.1" data-path="cnobject.html"><a href="cnobject.html#profile"><i class="fa fa-check"></i><b>7.1</b> Profile</a></li>
<li class="chapter" data-level="7.2" data-path="cnobject.html"><a href="cnobject.html#distribution"><i class="fa fa-check"></i><b>7.2</b> Distribution</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="sigobject.html"><a href="sigobject.html"><i class="fa fa-check"></i><b>8</b> Signature Object</a><ul>
<li class="chapter" data-level="8.1" data-path="sigobject.html"><a href="sigobject.html#signature-profile"><i class="fa fa-check"></i><b>8.1</b> Signature Profile</a><ul>
<li class="chapter" data-level="8.1.1" data-path="sigobject.html"><a href="sigobject.html#sbs-signature-profile"><i class="fa fa-check"></i><b>8.1.1</b> SBS Signature profile</a></li>
<li class="chapter" data-level="8.1.2" data-path="sigobject.html"><a href="sigobject.html#copy-number-signature-porfile"><i class="fa fa-check"></i><b>8.1.2</b> Copy Number Signature Porfile</a></li>
<li class="chapter" data-level="8.1.3" data-path="sigobject.html"><a href="sigobject.html#cosmic-signature-profile"><i class="fa fa-check"></i><b>8.1.3</b> COSMIC Signature Profile</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="sigobject.html"><a href="sigobject.html#exposure-profile"><i class="fa fa-check"></i><b>8.2</b> Exposure Profile</a></li>
<li class="chapter" data-level="8.3" data-path="sigobject.html"><a href="sigobject.html#consensus-map"><i class="fa fa-check"></i><b>8.3</b> Consensus Map</a></li>
<li class="chapter" data-level="8.4" data-path="sigobject.html"><a href="sigobject.html#catalogue-profile"><i class="fa fa-check"></i><b>8.4</b> Catalogue Profile</a></li>
</ul></li>
<li class="part"><span><b>III Universal Analysis</b></span></li>
<li class="chapter" data-level="9" data-path="association-analysis.html"><a href="association-analysis.html"><i class="fa fa-check"></i><b>9</b> Association Analysis and Visualization</a></li>
<li class="chapter" data-level="10" data-path="group-analysis.html"><a href="group-analysis.html"><i class="fa fa-check"></i><b>10</b> Group Analysis and Visualization</a><ul>
<li class="chapter" data-level="10.1" data-path="group-analysis.html"><a href="group-analysis.html#group-generation"><i class="fa fa-check"></i><b>10.1</b> Group Generation</a></li>
<li class="chapter" data-level="10.2" data-path="group-analysis.html"><a href="group-analysis.html#group-comparison-analysis"><i class="fa fa-check"></i><b>10.2</b> Group Comparison Analysis</a></li>
<li class="chapter" data-level="10.3" data-path="group-analysis.html"><a href="group-analysis.html#group-visualization"><i class="fa fa-check"></i><b>10.3</b> Group Visualization</a></li>
</ul></li>
<li class="part"><span><b>IV Useful Utilities</b></span></li>
<li class="chapter" data-level="11" data-path="datasets.html"><a href="datasets.html"><i class="fa fa-check"></i><b>11</b> Datasets</a><ul>
<li class="chapter" data-level="11.1" data-path="datasets.html"><a href="datasets.html#reference-annotation"><i class="fa fa-check"></i><b>11.1</b> Reference Annotation</a></li>
<li class="chapter" data-level="11.2" data-path="datasets.html"><a href="datasets.html#copy-number-components-setting"><i class="fa fa-check"></i><b>11.2</b> Copy Number components setting</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="convert.html"><a href="convert.html"><i class="fa fa-check"></i><b>12</b> SBS Signature Conversion</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Sigminer: A Scalable Toolkit to Extract, Analyze and Visualize Mutational Signatures</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="sigflow" class="section level1">
<h1><span class="header-section-number">Chapter 6</span> Sigflow Pipeline</h1>
<p><a href="https://github.com/ShixiangWang/sigminer.workflow">Sigflow</a> provides useful mutational signature analysis workflows based on R package <strong>sigminer</strong>. It can auto-extract mutational signatures, fit mutation data to COSMIC reference signatures (SBS/DBS/INDEL) and run bootstrapping analysis for signature fitting.</p>
<div id="installation-1" class="section level2">
<h2><span class="header-section-number">6.1</span> Installation</h2>
<p>Sigflow can be installed in any operating system (Windows/MacOS/Linux). The user can choose one of the following two ways to install Sigflow.</p>
<ul>
<li>Step-by-step installation.</li>
<li>Install Sigflow Docker image.</li>
</ul>
<div id="step-by-step-installation" class="section level3">
<h3><span class="header-section-number">6.1.1</span> Step-by-step installation</h3>
<p>This process takes about 1 hour to finish (depends on your internet speed). This is recommended for R users.</p>
<ol style="list-style-type: decimal">
<li><p>Install R from <a href="https://cran.r-project.org/">CRAN</a> (R &gt;3.6 is required).</p></li>
<li><p>Install R packages by running the following commands in your R console.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1"><span class="kw">install.packages</span>(<span class="st">&quot;docopt&quot;</span>)</a>
<a class="sourceLine" id="cb43-2" data-line-number="2"><span class="kw">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</a>
<a class="sourceLine" id="cb43-3" data-line-number="3">BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&quot;sigminer&quot;</span>, <span class="dt">dependencies =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb43-4" data-line-number="4"><span class="co">## Install specific version of sigminer</span></a>
<a class="sourceLine" id="cb43-5" data-line-number="5"><span class="co">## e.g. sigminer v1.0.10</span></a>
<a class="sourceLine" id="cb43-6" data-line-number="6"><span class="co"># install.packages(&quot;remotes&quot;)</span></a>
<a class="sourceLine" id="cb43-7" data-line-number="7"><span class="co"># remotes::install_github(&quot;ShixiangWang/sigminer@v1.0.10&quot;)</span></a></code></pre></div></li>
<li><p>(Optional) Install required reference genome R packages. The hg19 genome build is installed along with package sigminer. To install hg38 or mm10 genome build, run the following R commands.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" data-line-number="1"><span class="co"># For hg38</span></a>
<a class="sourceLine" id="cb44-2" data-line-number="2">BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&quot;BSgenome.Hsapiens.UCSC.hg38&quot;</span>)</a>
<a class="sourceLine" id="cb44-3" data-line-number="3"><span class="co"># For mm10</span></a>
<a class="sourceLine" id="cb44-4" data-line-number="4">BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&quot;BSgenome.Mmusculus.UCSC.mm10&quot;</span>)</a></code></pre></div></li>
<li><p>Clone Sigflow GitHub repository.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode sh"><code class="sourceCode bash"><a class="sourceLine" id="cb45-1" data-line-number="1">$ <span class="fu">git</span> clone https://github.com/ShixiangWang/sigminer.workflow</a></code></pre></div></li>
<li><p>(Optional) Link the R script as a executable file.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode sh"><code class="sourceCode bash"><a class="sourceLine" id="cb46-1" data-line-number="1">$ <span class="bu">cd</span> sigminer.workflow</a>
<a class="sourceLine" id="cb46-2" data-line-number="2">$ <span class="fu">ln</span> -s <span class="va">$PWD</span>/sigflow.R /usr/bin/sigflow  # You can choose another place instead of /usr/bin/sigflow</a></code></pre></div></li>
</ol>
<p>If you have the the step 5, you can restart your terminal and check the Siglow command by running <code>siglow -h</code>, otherwise you can directly run this script by <code>./sigflow.R -h</code>.</p>
</div>
<div id="install-by-docker" class="section level3">
<h3><span class="header-section-number">6.1.2</span> Install by Docker</h3>
<p>This process takes about 10 minutes to finish (depends on your internet speed). This is recommended for non-R users or users have no programming experience.</p>
<ol style="list-style-type: decimal">
<li><p>Follow Docker <a href="https://docs.docker.com/">official documentation</a> to install Docker.</p></li>
<li><p>Install latest stable version of Sigflow docker image by running the following command in your terminal.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode sh"><code class="sourceCode bash"><a class="sourceLine" id="cb47-1" data-line-number="1">$ <span class="fu">sudo</span> docker pull shixiangwang/sigflow:1.0</a></code></pre></div>
<blockquote>
<p>Here <code>$</code> is the prompt.</p>
</blockquote></li>
</ol>
<p>Now you can run Siglow using the following command.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode sh"><code class="sourceCode bash"><a class="sourceLine" id="cb48-1" data-line-number="1">$ <span class="ex">docker</span> run shixiangwang/sigflow</a></code></pre></div>
</div>
</div>
<div id="usage" class="section level2">
<h2><span class="header-section-number">6.2</span> Usage</h2>
<p>The <strong>Installation</strong> section has described how to install Sigflow and call it.</p>
<p>Sigflow now provides 4 workflows which implemented in 3 commands:</p>
<ul>
<li><code>extract</code></li>
<li><code>fit</code></li>
<li><code>bt</code></li>
</ul>
<p>The details of each command and corresponding flags are documented as below.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode sh"><code class="sourceCode bash"><a class="sourceLine" id="cb49-1" data-line-number="1">=================================================================</a>
<a class="sourceLine" id="cb49-2" data-line-number="2"><span class="ex">sigflow</span>: Streamline Analysis Workflows for Mutational Signatures.</a>
<a class="sourceLine" id="cb49-3" data-line-number="3"></a>
<a class="sourceLine" id="cb49-4" data-line-number="4"><span class="ex">Author</span>: Shixiang Wang (wangshx@shanghaitech.edu.cn)</a>
<a class="sourceLine" id="cb49-5" data-line-number="5"><span class="ex">Author</span>: Xue-Song Liu (liuxs@shanghaitech.edu.cn)</a>
<a class="sourceLine" id="cb49-6" data-line-number="6"><span class="ex">Copyright</span>: AFL@2020 [https://opensource.org/licenses/AFL-3.0]</a>
<a class="sourceLine" id="cb49-7" data-line-number="7"></a>
<a class="sourceLine" id="cb49-8" data-line-number="8"><span class="ex">Desc</span>:</a>
<a class="sourceLine" id="cb49-9" data-line-number="9">  <span class="ex">There</span> are several subcommands.</a>
<a class="sourceLine" id="cb49-10" data-line-number="10">  ==</a>
<a class="sourceLine" id="cb49-11" data-line-number="11">  <span class="ex">extract</span> - extract signatures by either automated or manual way.</a>
<a class="sourceLine" id="cb49-12" data-line-number="12">            <span class="ex">Of</span> note, when you use manual way, you need to run 2 times, </a>
<a class="sourceLine" id="cb49-13" data-line-number="13">            <span class="ex">firstly</span> you should set --manual to get signature estimation results,</a>
<a class="sourceLine" id="cb49-14" data-line-number="14">            <span class="ex">and</span> secondly you should set --manual --number N to get N signatures.</a>
<a class="sourceLine" id="cb49-15" data-line-number="15">  ==</a>
<a class="sourceLine" id="cb49-16" data-line-number="16">  <span class="ex">fit</span>     - fit signatures in <span class="op">&gt;</span>=1 samples based on COSMIC reference signatures.</a>
<a class="sourceLine" id="cb49-17" data-line-number="17">  ==</a>
<a class="sourceLine" id="cb49-18" data-line-number="18">  <span class="ex">bt</span>      - run bootstrap signature fitting analysis in <span class="op">&gt;</span>=1 samples.</a>
<a class="sourceLine" id="cb49-19" data-line-number="19"></a>
<a class="sourceLine" id="cb49-20" data-line-number="20"><span class="ex">Usage</span>:</a>
<a class="sourceLine" id="cb49-21" data-line-number="21">  <span class="ex">sigflow</span> extract --input=<span class="op">&lt;</span>file<span class="op">&gt;</span> [--output=<span class="op">&lt;</span>outdir<span class="op">&gt;</span>] [--mode=<span class="op">&lt;</span>class<span class="op">&gt;</span>] [--manual --number <span class="op">&lt;</span>sigs<span class="op">&gt;</span>] [--max <span class="op">&lt;</span>max<span class="op">&gt;</span>] [--genome=<span class="op">&lt;</span>genome<span class="op">&gt;</span>] [--nrun=<span class="op">&lt;</span>runs<span class="op">&gt;</span>] [--cores=<span class="op">&lt;</span>cores<span class="op">&gt;</span>] [--sigprofiler] [--hyper] [--verbose]</a>
<a class="sourceLine" id="cb49-22" data-line-number="22">  <span class="ex">sigflow</span> fit --input=<span class="op">&lt;</span>file<span class="op">&gt;</span> [--output=<span class="op">&lt;</span>outdir<span class="op">&gt;</span>] [--mode=<span class="op">&lt;</span>class<span class="op">&gt;</span>] [--genome=<span class="op">&lt;</span>genome<span class="op">&gt;</span>] [--verbose]</a>
<a class="sourceLine" id="cb49-23" data-line-number="23">  <span class="ex">sigflow</span> bt  --input=<span class="op">&lt;</span>file<span class="op">&gt;</span> [--output=<span class="op">&lt;</span>outdir<span class="op">&gt;</span>] [--mode=<span class="op">&lt;</span>class<span class="op">&gt;</span>] [--genome=<span class="op">&lt;</span>genome<span class="op">&gt;</span>] [--nrun=<span class="op">&lt;</span>runs<span class="op">&gt;</span>] [--verbose]</a>
<a class="sourceLine" id="cb49-24" data-line-number="24">  <span class="ex">sigflow</span> (-h <span class="kw">|</span> <span class="ex">--help</span>)</a>
<a class="sourceLine" id="cb49-25" data-line-number="25">  <span class="ex">sigflow</span> --version</a>
<a class="sourceLine" id="cb49-26" data-line-number="26"></a>
<a class="sourceLine" id="cb49-27" data-line-number="27"><span class="ex">Options</span>:</a>
<a class="sourceLine" id="cb49-28" data-line-number="28">  <span class="ex">-h</span> --help     Show help message.</a>
<a class="sourceLine" id="cb49-29" data-line-number="29">  <span class="ex">--version</span>     Show version.</a>
<a class="sourceLine" id="cb49-30" data-line-number="30">  <span class="ex">-i</span> <span class="op">&lt;</span>file<span class="op">&gt;</span>, --input <span class="op">&lt;</span>file<span class="op">&gt;</span>       input file/directory path.</a>
<a class="sourceLine" id="cb49-31" data-line-number="31">  <span class="ex">-o</span> <span class="op">&lt;</span>outdir<span class="op">&gt;</span>, --output <span class="op">&lt;</span>outdir<span class="op">&gt;</span>  output directory path [default: ./sigflow_result/].</a>
<a class="sourceLine" id="cb49-32" data-line-number="32">  <span class="ex">-m</span> <span class="op">&lt;</span>class<span class="op">&gt;</span>, --mode <span class="op">&lt;</span>class<span class="op">&gt;</span>      extract/fit mode, can be one of SBS, DBS, ID, MAF (for three types), <span class="ex">CN</span> (not supported in fit subcommand) [<span class="ex">default</span>: SBS].</a>
<a class="sourceLine" id="cb49-33" data-line-number="33">  <span class="ex">--manual</span>                        enable manual extraction, set -N=0 for outputing signature estimation firstly.</a>
<a class="sourceLine" id="cb49-34" data-line-number="34">  <span class="ex">-N</span> <span class="op">&lt;</span>sigs<span class="op">&gt;</span>, --number <span class="op">&lt;</span>sigs<span class="op">&gt;</span>      extract specified number of signatures [default: 0].</a>
<a class="sourceLine" id="cb49-35" data-line-number="35">  <span class="ex">--max</span> <span class="op">&lt;</span>max<span class="op">&gt;</span>                     maximum signature number, default is auto-configured, should <span class="op">&gt;</span>2 [default: -1].</a>
<a class="sourceLine" id="cb49-36" data-line-number="36">  <span class="ex">-g</span> <span class="op">&lt;</span>genome<span class="op">&gt;</span>, --genome <span class="op">&lt;</span>genome<span class="op">&gt;</span>  genome build, can be hg19, hg38 or mm10, [default: hg19].</a>
<a class="sourceLine" id="cb49-37" data-line-number="37">  <span class="ex">-r</span> <span class="op">&lt;</span>runs<span class="op">&gt;</span>, --nrun <span class="op">&lt;</span>runs<span class="op">&gt;</span>        run times of NMF (extract) <span class="ex">or</span> bootstrapping (bt) <span class="ex">to</span> get results [default: 30].</a>
<a class="sourceLine" id="cb49-38" data-line-number="38">  <span class="ex">-T</span> <span class="op">&lt;</span>cores<span class="op">&gt;</span>, --cores <span class="op">&lt;</span>cores<span class="op">&gt;</span>     cores to run the program, large dataset will benefit from it [default: 1].</a>
<a class="sourceLine" id="cb49-39" data-line-number="39">  <span class="ex">--hyper</span>                         enable hyper mutation handling in COSMIC signatures (not used by SigProfiler approach)<span class="ex">.</span></a>
<a class="sourceLine" id="cb49-40" data-line-number="40">  <span class="ex">--sigprofiler</span>                   enable auto-extraction by SigProfiler software.</a>
<a class="sourceLine" id="cb49-41" data-line-number="41">  <span class="ex">-v</span>, --verbose                   print verbose message.</a>
<a class="sourceLine" id="cb49-42" data-line-number="42"></a>
<a class="sourceLine" id="cb49-43" data-line-number="43">=================================================================</a></code></pre></div>
<p>Sigflow supports input data in VCF/MAF/CSV/EXCEL format. The file format is auto-detected by Sigflow.</p>
<p>For SBS/DBS/INDEL data in CSV (including TSV) or EXCEL format, the following columns typically described in MAF format are necessary:</p>
<ul>
<li>‘Hugo_Symbol’: gene symbol</li>
<li>‘Chromosome’: chromosome name, e.g. “chr1”</li>
<li>‘Start_Position’: start positionof the variant (1-based)</li>
<li>‘End_Position’: end position of the variant (1-based)</li>
<li>‘Reference_Allele’: reference allele of the variant, e.g. “C”</li>
<li>‘Tumor_Seq_Allele2’: tumor sequence allele, e.g. “T”</li>
<li>‘Variant_Classification’: variant classification, e.g. “Missense_Mutation”</li>
<li>‘Variant_Type’: variant type, e.g. “SNP”</li>
<li>‘Tumor_Sample_Barcode’: sample identifier</li>
</ul>
<p>For copy number segment data in in CSV (including TSV) or EXCEL format, the following columns are necessary:</p>
<ul>
<li>‘Chromosome’: chromosome name, e.g. “chr1”</li>
<li>‘Start.bp’: start breakpoint position of segment</li>
<li>‘End.bp’: end breakpoint position of segment</li>
<li>‘modal_cn’: integer copy number value</li>
<li>‘sample’: sample identifier</li>
</ul>
</div>
<div id="suggestion-and-bug-report" class="section level2">
<h2><span class="header-section-number">6.3</span> Suggestion and bug report</h2>
<p>Any suggestion, feature request or bug report is welcome at <a href="https://github.com/ShixiangWang/sigminer.workflow/issues">GitHub repository issue</a>.</p>
</div>
<div id="examples" class="section level2">
<h2><span class="header-section-number">6.4</span> Examples</h2>
<p>Example datasets along with many example code are available in clone repository above (you can read it online at <a href="https://github.com/ShixiangWang/sigminer.workflow/tree/master/test">here</a>).</p>
<p>The following parts give an example for each command.</p>
<p>Result directory of any command has the following structure.</p>
<ul>
<li>Files with extension<code>.RData</code> and <code>.rds</code> are R related files to reproduce the results, and can be imported into R for further analysis and visualization.</li>
<li>Files with extension<code>.pdf</code> are common visualization results used for communication.</li>
<li>Files with extension <code>.csv</code> are formated data tables used for inspection, communication or further analysis.</li>
</ul>
<div id="extract-command" class="section level3">
<h3><span class="header-section-number">6.4.1</span> <code>extract</code> command</h3>
<div class="sourceCode" id="cb50"><pre class="sourceCode sh"><code class="sourceCode bash"><a class="sourceLine" id="cb50-1" data-line-number="1">$ <span class="co"># Assume you have done the clone step</span></a>
<a class="sourceLine" id="cb50-2" data-line-number="2">$ <span class="co"># git clone https://github.com/ShixiangWang/sigminer.workflow</span></a>
<a class="sourceLine" id="cb50-3" data-line-number="3">$ <span class="bu">cd</span> sigminer.workflow/test</a>
<a class="sourceLine" id="cb50-4" data-line-number="4">$ <span class="ex">sigflow</span> extract -i tcga_laml.maf.gz -o test_results/test_maf -m MAF -r 10 -T 4 --max 10</a></code></pre></div>
<p>This will auto-extract SBS/DBS/INDEL signatures from data <code>toga_laml.maf.gz</code> by 10 Bayesian NMF runs with 4 computer cores (4 threads) from signature number ranges from 1 to 10, output results to directory <code>test_results/test_maf</code>.</p>
</div>
<div id="fit-command" class="section level3">
<h3><span class="header-section-number">6.4.2</span> <code>fit</code> command</h3>
<div class="sourceCode" id="cb51"><pre class="sourceCode sh"><code class="sourceCode bash"><a class="sourceLine" id="cb51-1" data-line-number="1">$ <span class="co"># Assume you have done the clone step</span></a>
<a class="sourceLine" id="cb51-2" data-line-number="2">$ <span class="co"># git clone https://github.com/ShixiangWang/sigminer.workflow</span></a>
<a class="sourceLine" id="cb51-3" data-line-number="3">$ <span class="bu">cd</span> sigminer.workflow/test</a>
<a class="sourceLine" id="cb51-4" data-line-number="4">$ <span class="ex">sigflow</span> fit -i tcga_laml.maf.gz -o test_results/test_fitting -m MAF</a></code></pre></div>
<p>This will auto-fit input data <code>tcga_laml.maf.gz</code> to COSMIC SBS/DBS/INDEL signatures. Signature exposure data tables and plots are outputed.</p>
</div>
<div id="bt-command" class="section level3">
<h3><span class="header-section-number">6.4.3</span> <code>bt</code> command</h3>
<div class="sourceCode" id="cb52"><pre class="sourceCode sh"><code class="sourceCode bash"><a class="sourceLine" id="cb52-1" data-line-number="1">$ <span class="co"># Assume you have done the clone step</span></a>
<a class="sourceLine" id="cb52-2" data-line-number="2">$ <span class="co"># git clone https://github.com/ShixiangWang/sigminer.workflow</span></a>
<a class="sourceLine" id="cb52-3" data-line-number="3">$ <span class="bu">cd</span> sigminer.workflow/test</a>
<a class="sourceLine" id="cb52-4" data-line-number="4">$ <span class="ex">sigflow</span> bt -i tcga_laml.maf.gz -o test_results/test_bt -m SBS -r 5</a></code></pre></div>
<p>This will auto-fit the random resample of input mutation profile to COSMIC SBS/DBS/INDEL signatures for specified times (here is 5). Data tables and plots of bootstrap signature exposures, errors and p values under different exposure cutoff are outputed.</p>
<blockquote>
<p>NOTE, in practice, set <code>-r</code> to a value <code>&gt;=100</code> is recommended.</p>
</blockquote>
</div>
<div id="how-to-use-docker-to-run-sigflow" class="section level3">
<h3><span class="header-section-number">6.4.4</span> How to use Docker to run Sigflow</h3>
<p>If you use Docker to run Sigflow, you cannot directly call <code>sigflow</code> command. Instead, you should use <code>sudo docker run --rm -v /your_local_path:/docker_path shixiangwang/sigflow</code> to start a Docker container.</p>
<p>For example, if you want to accomplish the same task shown in <code>extract</code> command above, you need to run:</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode sh"><code class="sourceCode bash"><a class="sourceLine" id="cb53-1" data-line-number="1">$ <span class="fu">sudo</span> docker run --rm -v /your_local_path:/docker_path shixiangwang/sigflow extract -i /docker_path/tcga_laml.maf.gz -o /docker_path/test_maf -m MAF -r 10 -T 4 --max 10</a></code></pre></div>
<p>Here,</p>
<ul>
<li><p><code>--rm</code> will delete this container when this task is finished.</p></li>
<li><p><code>-v</code> is used for mounting your local directory <code>/your_local_path</code> as <code>/docker_path</code> in Docker image. <strong>This is important</strong>. You need to use the Docker container path in Sigflow arguments. So there must be a file called <code>/your_local_path/tcga_laml.maf.gz</code> exists in your computer, it will be treated as <code>/docker_path/test_maf</code> in the container.</p></li>
</ul>

</div>
</div>
</div>



            </section>

          </div>
        </div>
      </div>
<a href="subtyping.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="cnobject.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/ShixiangWang/sigminer-doc/edit/master/03-quantify-sample-exposure.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["sigminer-doc.pdf", "sigminer-doc.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
